From 7621c71d3fc9abaf90f220a5a2d68c1bc25ee543 Mon Sep 17 00:00:00 2001
From: Alexander Khryukin <alexander@mezon.ru>
Date: Sat, 14 Dec 2013 17:07:55 +0000
Subject: [PATCH] fix aarch64 rpm5 multiarch-headers scripting

---
 scripts/multiarch-dispatch.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/scripts/multiarch-dispatch.h b/scripts/multiarch-dispatch.h
index f6c2cef..31aa860 100644
--- a/scripts/multiarch-dispatch.h
+++ b/scripts/multiarch-dispatch.h
@@ -30,6 +30,10 @@
 #define _multiarch_defined_ppc64
 #undef ppc64
 #endif
+#if aarch64 == 1
+#define _multiarch_defined_aarch64
+#undef aarch64
+#endif
 #if ppc == 1
 #define _multiarch_defined_ppc
 #undef ppc
@@ -74,6 +78,8 @@
 #include _MULTIARCH_MAKE_HEADER(i386,_MULTIARCH_HEADER)
 #elif defined(__powerpc64__)
 #include _MULTIARCH_MAKE_HEADER(ppc64,_MULTIARCH_HEADER)
+#elif defined(__aarch64__)
+#include _MULTIARCH_MAKE_HEADER(aarch64,_MULTIARCH_HEADER)
 #elif defined(__powerpc__)
 #include _MULTIARCH_MAKE_HEADER(ppc,_MULTIARCH_HEADER)
 #elif defined(__sparc__) && defined(__arch64__)
@@ -138,6 +144,10 @@
 #undef  _multiarch_defined_ppc64
 #define ppc64 1
 #endif
+#ifdef  _multiarch_defined_aarch64
+#undef  _multiarch_defined_aarch64
+#define aarch64 1
+#endif
 #ifdef  _multiarch_defined_ppc
 #undef  _multiarch_defined_ppc
 #define ppc 1
-- 
1.8.3.2

