From ff7ae173bd80424c0e5af63c3b844a7bcfb15604 Mon Sep 17 00:00:00 2001
From: jbj <jbj>
Date: Fri, 5 Sep 2014 16:04:51 +0000
Subject: [PATCH] - rpmds: fix: off-by-1 comparison check parsing N.A compound.

---
 lib/rpmds.c | 4 ++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/lib/rpmds.c b/lib/rpmds.c
index 30dc093..16aee35 100644
--- a/lib/rpmds.c
+++ b/lib/rpmds.c
@@ -3991,11 +3991,11 @@ static int rpmdsNAcmp(rpmds A, rpmds B)
     if (!AA && !BA) {
 	rc = strcmp(AN, BN);
     } else if (AA && !BA) {
-	rc = strncmp(AN, BN, (AA - AN)) || BN[AA - AN];
+	rc = strncmp(AN, BN, (AA - AN)) || BN[AA - AN - 1];
 	if (!rc)
 	    rc = strcmp(AA, B->A);
     } else if (!AA && BA) {
-	rc = strncmp(AN, BN, (BA - BN)) || AN[BA - BN];
+	rc = strncmp(AN, BN, (BA - BN)) || AN[BA - BN - 1];
 	if (!rc)
 	    rc = strcmp(BA, A->A);
     } else {
-- 
1.9.0

