diff -up rpm-4.19.1/fileattrs/CMakeLists.txt.8~ rpm-4.19.1/fileattrs/CMakeLists.txt
--- rpm-4.19.1/fileattrs/CMakeLists.txt.8~	2023-12-13 11:03:36.355404257 +0100
+++ rpm-4.19.1/fileattrs/CMakeLists.txt	2023-12-13 11:10:13.156259282 +0100
@@ -2,5 +2,6 @@ install(FILES
 	debuginfo.attr desktop.attr elf.attr font.attr metainfo.attr
 	perl.attr perllib.attr pkgconfig.attr ocaml.attr
 	rpm_macro.attr rpm_lua.attr script.attr sysusers.attr usergroup.attr
+	elflib.attr
 	DESTINATION ${RPM_CONFIGDIR}/fileattrs
 )
diff -up rpm-4.19.1/fileattrs/elf.attr.8~ rpm-4.19.1/fileattrs/elf.attr
--- rpm-4.19.1/fileattrs/elf.attr.8~	2023-12-12 11:57:31.000000000 +0100
+++ rpm-4.19.1/fileattrs/elf.attr	2023-12-13 11:03:36.355404257 +0100
@@ -1,4 +1,5 @@
 %__elf_provides		%{_rpmconfigdir}/elfdeps --provides
 %__elf_requires		%{_rpmconfigdir}/elfdeps --requires
-%__elf_magic		^(setuid,? )?(setgid,? )?(sticky )?ELF (32|64)-bit.*$
+%__elf_magic		^(setuid,? )?(setgid,? )?(sticky )?ELF (32|64)-bit.*executable
+%__elf_exclude_path	^/usr/lib/debug/
 %__elf_exclude_path	^/lib/modules/.*\.ko?(\.[[:alnum:]]*)$
diff -up rpm-4.19.1/fileattrs/elflib.attr.8~ rpm-4.19.1/fileattrs/elflib.attr
--- rpm-4.19.1/fileattrs/elflib.attr.8~	2023-12-13 11:03:36.355404257 +0100
+++ rpm-4.19.1/fileattrs/elflib.attr	2023-12-13 11:03:36.355404257 +0100
@@ -0,0 +1,4 @@
+%__elflib_provides	%{_rpmconfigdir}/elfdeps --assume-exec --provides %{?__filter_GLIBC_PRIVATE:--filter-private}
+%__elflib_requires	%{_rpmconfigdir}/elfdeps --assume-exec --requires %{?__filter_GLIBC_PRIVATE:--filter-private}
+%__elflib_magic		^(setuid )?(setgid )?(sticky )?ELF (32|64)-bit.*shared object
+%__elflib_exclude_path	^/usr/lib/debug/
