diff --git a/macros/mandriva.in b/macros/mandriva.in
index 731d460..6ced5f4 100644
--- a/macros/mandriva.in
+++ b/macros/mandriva.in
@@ -355,6 +355,9 @@ GCONF_CONFIG_SOURCE=`%{_gconftool_bin} --get-default-source` %{_gconftool_bin} -
 
 %before_configure \
   %setup_compile_flags \
+  for i in $(find . -name config.guess -o -name config.sub) ; do \
+           [ -f /usr/share/libtool/config/$(basename $i) ] && /bin/rm -f $i && /bin/cp -fv /usr/share/libtool/config//$(basename $i) $i ; \
+  done ; \
   CONFIGURE_TOP="${CONFIGURE_TOP:-.}"
 
 # This is an improved version of %configure (from PLD team).
