diff --git a/macros/mandriva.in b/macros/mandriva.in
index 8aa16a5..9e42b1d 100644
--- a/macros/mandriva.in
+++ b/macros/mandriva.in
@@ -348,7 +348,13 @@ GCONF_CONFIG_SOURCE=`%{_gconftool_bin} --get-default-source` %{_gconftool_bin} -
   export CROSSCOMPILE ; \
   %{nil}
 
+%config_update \
+  for i in $(find . -name config.guess -o -name config.sub) ; do \
+           [ -f /usr/share/libtool/config/$(basename $i) ] && /bin/rm -f $i && /bin/cp -fv /usr/share/libtool/config//$(basename $i) $i ; \
+  done ; \
+
 %before_configure \
+  %config_update \
   %setup_compile_flags \
   CONFIGURE_TOP="${CONFIGURE_TOP:-.}"
 
