diff -up perl-rpm-packaging-master/scripts/perl.prov.omv~ perl-rpm-packaging-master/scripts/perl.prov
--- perl-rpm-packaging-master/scripts/perl.prov.omv~	2024-10-12 00:20:24.144045876 +0200
+++ perl-rpm-packaging-master/scripts/perl.prov	2024-10-12 00:20:44.834281627 +0200
@@ -78,7 +78,7 @@ if ("@ARGV") {
 foreach $module (sort keys %require) {
   if (length($require{$module}) == 0) {
     print "perl($module)\n";
-    print "perln($module)\n" if !$perln;
+    print "perln($module)\n" if $perln;
   } else {
 
     # I am not using rpm3.0 so I do not want spaces around my
diff -up perl-rpm-packaging-master/scripts/perl.req.omv~ perl-rpm-packaging-master/scripts/perl.req
--- perl-rpm-packaging-master/scripts/perl.req.omv~	2024-10-12 00:20:04.003816331 +0200
+++ perl-rpm-packaging-master/scripts/perl.req	2024-10-12 00:20:22.264024452 +0200
@@ -77,7 +77,7 @@ foreach $perlver (sort keys %perlreq) {
 foreach $module (sort keys %require) {
   if (length($require{$module}) == 0) {
     print "perl($module)\n";
-    print "perln($module)\n" if !$perln;
+    print "perln($module)\n" if $perln;
   } else {
 
     # I am not using rpm3.0 so I do not want spaces around my
