diff -up perl-rpm-packaging-master/scripts/perl.req.1~ perl-rpm-packaging-master/scripts/perl.req
--- perl-rpm-packaging-master/scripts/perl.req.1~	2025-02-21 00:35:00.000000000 +0100
+++ perl-rpm-packaging-master/scripts/perl.req	2025-04-10 18:18:05.167301388 +0200
@@ -86,15 +86,15 @@ foreach $module (sort keys %require) {
 
     if ($normalversion) {
       my $normv = normalversion($require{$module});
-      print "perl($module) = $normv\n";
+      print "perl($module) >= $normv\n";
     }
     else
     {
-      print "perl($module) = $require{$module}\n";
+      print "perl($module) >= $require{$module}\n";
     }
     if ($perln) {
       my $normv = normalversion($require{$module});
-      print "perln($module) = $normv\n";
+      print "perln($module) >= $normv\n";
     }
   }
 }
