diff --git a/perl/Makefile.PL.in b/perl/Makefile.PL.in
index f6059c5..8c6b172 100644
--- a/perl/Makefile.PL.in
+++ b/perl/Makefile.PL.in
@@ -66,6 +66,7 @@ WriteMakefile(
 	      'MAKEFILE'=> 'Makefile.perl',
 	      'CC'	=> '@CC@',
 	      'CXX'	=> '@CXX@',
+	      'LD'	=> '@CC@',
 	      'CCFLAGS' => join(' ', '@CFLAGS@', '-Wno-unused-variable', '-Wno-missing-prototypes', '-Wno-implicit', '-Wno-strict-prototypes'),
 	      'dynamic_lib'	=> { OTHERLDFLAGS => join(' ', @ldaddp, @ldadd, @ldlibs) },
 	      'OPTIMIZE'=> '-g',
