--- rpm-5.4.9/rpmio/rpmpython.c.py_input~	2012-05-15 03:40:55.316698083 +0200
+++ rpm-5.4.9/rpmio/rpmpython.c	2012-05-15 03:40:59.799722384 +0200
@@ -215,7 +215,7 @@ fprintf(stderr, "==> %s(%p,%s,%p)\n", __
 	PyCompilerFlags cf = { 0 };
 	PyObject * m = PyImport_AddModule("__main__");
 	PyObject * d = (m ? PyModule_GetDict(m) : NULL);
-	PyObject * v = (m ? PyRun_StringFlags(val, Py_file_input, d, d, &cf) : NULL);
+	PyObject * v = (m ? PyRun_StringFlags(val, Py_single_input, d, d, &cf) : NULL);
 
         if (v == NULL) {
 	    PyErr_Print();
