diff --git a/platform/armv7hl-linux.macros b/platform/armv7hl-linux.macros
index 326d1aa..8b9edde 100644
--- a/platform/armv7hl-linux.macros
+++ b/platform/armv7hl-linux.macros
@@ -6,7 +6,7 @@
 %_arch			arm
 %_os			linux
 %_gnu			-gnueabihf
-%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fPIC -g -march=armv7-a -mfloat-abi=hard -mfpu=neon}%{!?__common_cflags_with_ssp:-O2} %{?!_disable_lto:-flto}
+%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fPIC -g -march=armv7-a -mfloat-abi=hard}%{!?__common_cflags_with_ssp:-O2} %{?!_disable_neon:-mfpu=neon} %{?!_disable_lto:-flto}
 
 #==============================================================================
 # ---- configure macros.
