--- rpm-5.4.10/scripts/pythoneggs.py.python3~	2012-05-08 00:34:19.000000000 +0200
+++ rpm-5.4.10/scripts/pythoneggs.py	2014-06-22 15:37:31.571539741 +0200
@@ -125,7 +125,10 @@ for f in files:
                 if not name in py_deps:
                     py_deps[name] = []
                 py_deps[name].append(('==', dist.py_version))
-            name = 'pythonegg(%s)' % dist.key
+            if f.find('python3') > 0:
+                name = 'python3egg(%s)' % dist.key
+            else:
+                name = 'pythonegg(%s)' % dist.key
             if not name in py_deps:
                 py_deps[name] = []
             if dist.version:
@@ -155,7 +158,10 @@ for f in files:
                 deps = depsextras
             # add requires/suggests based on egg metadata
             for dep in deps:
-                name = 'pythonegg(%s)' % dep.key
+                if f.find('python3') > 0:
+                    name = 'python3egg(%s)' % dep.key
+                else:
+                    name = 'pythonegg(%s)' % dep.key
                 for spec in dep.specs:
                     if spec[0] != '!=':
                         if not name in py_deps:
