--- rpm-5.4.9/lib/rpmfc.c.no_verdepfeat~	2012-05-15 03:02:42.578268986 +0200
+++ rpm-5.4.9/lib/rpmfc.c	2012-05-15 03:03:46.035612993 +0200
@@ -504,19 +504,6 @@ assert(EVR != NULL);
 	    if (_filter_values && rpmfcMatchRegexps(mire, nmire, N, deptype))
 		continue;
 
-	    /* Add tracking dependency for versioned Provides: */
-	    if (!fc->tracked && deptype == 'P' && *EVR != '\0') {
-		static evrFlags _Flags = (evrFlags)
-			(RPMSENSE_RPMLIB|(RPMSENSE_LESS|RPMSENSE_EQUAL));
-		ds = rpmdsSingle(RPMTAG_REQUIRENAME,
-			"rpmlib(VersionedDependencies)", "3.0.3-1",
-			_Flags);
-		xx = rpmdsMerge(&fc->requires, ds);
-		(void)rpmdsFree(ds);
-		ds = NULL;
-		fc->tracked = 1;
-	    }
-
 	    ds = rpmdsSingle(tagN, N, EVR, Flags);
 
 #if defined(RPM_VENDOR_MANDRIVA) /* filter-overlapping-dependencies */
@@ -2022,7 +2009,6 @@ rpmRC rpmfcGenerateDepends(void * _spec,
     fc = rpmfcNew();
     fc->skipProv = !pkg->autoProv;
     fc->skipReq = !pkg->autoReq;
-    fc->tracked = 0;
 
     {	const char * buildRootURL;
 	const char * buildRoot;
--- rpm-5.4.9/lib/rpmfc.h.no_verdepfeat~	2012-04-22 19:07:07.000000000 +0200
+++ rpm-5.4.9/lib/rpmfc.h	2012-05-15 03:02:42.609269153 +0200
@@ -76,7 +76,6 @@ struct rpmfc_s {
     size_t ix;		/*!< current file index */
     int skipProv;	/*!< Don't auto-generate Provides:? */
     int skipReq;	/*!< Don't auto-generate Requires:? */
-    int tracked;	/*!< Versioned Provides: tracking dependency added? */
     size_t brlen;	/*!< strlen(spec->buildRoot) */
 
     ARGV_t fn;		/*!< (no. files) file names */
