--- rpm-5.4.4/build/reqprov.c.drop_basedeps~	2011-12-11 03:55:58.958164657 +0100
+++ rpm-5.4.4/build/reqprov.c	2011-12-11 03:56:04.794169328 +0100
@@ -48,6 +48,12 @@ int addReqProv(/*@unused@*/ Spec spec, H
 	indextag = RPMTAG_TRIGGERINDEX;
 	extra = Flags & RPMSENSE_TRIGGER;
     } else {
+#if defined(RPM_VENDOR_MANDRIVA)
+	if (!strcmp(N, "/bin/sh") || !strcmp(N, "/sbin/ldconfig")) {
+	    rpmlog(RPMLOG_DEBUG, "%s will be satisfied by glibc or it's dependencies, won't add dependency to package\n");
+	    return 0;
+	}
+#endif
 	nametag = RPMTAG_REQUIRENAME;
 	versiontag = RPMTAG_REQUIREVERSION;
 	flagtag = RPMTAG_REQUIREFLAGS;
