--- rpm-5.4.10/tools/debugedit.c.debugedit_errmsg~	2013-01-03 09:24:01.764153125 +0100
+++ rpm-5.4.10/tools/debugedit.c	2013-01-03 09:31:33.485896571 +0100
@@ -696,7 +696,7 @@ edit_dwarf2_line (DSO *dso, rpmuint32_t
 	{
 	  if (--shrank == 0)
 	    error (EXIT_FAILURE, 0,
-		   "canonicalization unexpectedly shrank by one character");
+		   "canonicalization unexpectedly shrank by one character ('%s' vs '%s')", base_dir, dest_dir);
 	  else
 	    {
 	      memset (ptr, 'X', shrank);
