--- rpm-5.4.5/lib/rpmds.c.uclibc_nolib64~	2012-03-07 16:45:18.844313010 +0100
+++ rpm-5.4.5/lib/rpmds.c	2012-03-07 16:45:26.711321576 +0100
@@ -3131,7 +3131,7 @@ static int checkuClibc(GElf_Ehdr *ehdr,
 	if (tmp[0] == 'l' && tmp[1] == 'd') {
 	    tmp += ((tmp[2] == '3' && tmp[3] == '2') ||
 		    (tmp[2] == '6' && tmp[3] == '4')) ?
-		5 : 7;
+		5 : 3;
 	    if (!strncasecmp(tmp, "uClibc.so", sizeof("uClibc.so")-1))
 		ret = 1;
 	}
