--- rpm-5.4.12/configure.ac.db52~	2013-07-30 21:07:55.479389231 +0200
+++ rpm-5.4.12/configure.ac	2013-07-30 21:09:26.087462718 +0200
@@ -544,7 +544,7 @@ else
     MYPATH=$PATH
 fi
 
-DBXY=db60
+DBXY=db52
 
 AC_PATH_PROG(__BASH, bash, %{_bindir}/bash, $MYPATH)
 AC_PATH_PROG(__BZIP2, bzip2, %{_bindir}/bzip2, $MYPATH)
@@ -1644,7 +1644,7 @@ DBLIBSRCS=""
 CPPFLAGS="${CPPFLAGS} -I${prefix}/include/${DBXY}"
 RPM_CHECK_LIB(
     [Berkeley-DB], [db],
-    [db-6.0], [db_create], [db.h],
+    [db-5.2], [db_create], [db.h],
     [yes,external], [db3],
     [ DBLIBSRCS="$DBLIBSRCS db3.c"
       AM_CONDITIONAL(WITH_DB, [ true ])
@@ -1671,7 +1671,7 @@ dnl # Sqlite 3.7.0.1 from db-5.1.19
 dnl XXX error: `db3' is already registered with AC_CONFIG_SUBDIRS.
 RPM_CHECK_LIB(
     [Berkeley-DB (+SQLite3)], [dbsql],
-    [db_sql-6.0], [sqlite3_open], [dbsql.h],
+    [db_sql-5.2], [sqlite3_open], [dbsql.h],
     [yes,external], [db3/sql],
     [
       AM_CONDITIONAL(WITH_DBSQL, [ true ])
