--- rpm-5.4.12/rpmio/rpmsquirrel.c.squir_ver~	2013-07-30 18:01:16.027090121 +0200
+++ rpm-5.4.12/rpmio/rpmsquirrel.c	2013-07-30 18:00:53.800778490 +0200
@@ -162,7 +162,7 @@ rpmsquirrel rpmsquirrelNew(char ** av, u
 
     squirrel->I = v;
     sq_setforeignptr(v, squirrel);
-#if defined(SQUIRREL_VERSION_NUMBER)
+#if defined(SQUIRREL_VERSION_NUMBER) && SQUIRREL_VERSION_NUMBER >= 300
     sq_setprintfunc(v, rpmsquirrelPrint, rpmsquirrelStderr);
 #else
     sq_setprintfunc(v, rpmsquirrelPrint);
