--- rpm-5.4.15/build/files.c.return_type~	2015-04-14 10:08:37.936921891 +0200
+++ rpm-5.4.15/build/files.c	2015-04-14 10:08:08.486024424 +0200
@@ -1691,7 +1691,7 @@ if (_rpmbuildFlags & 4) {
     rpmfi fi = rpmfiNew(ts, h, RPMTAG_BASENAMES, scareMem);
     char * a, * d;
 
-    if (fi == NULL) return;		/* XXX can't happen */
+    if (fi == NULL) return RPMRC_FAIL;		/* XXX can't happen */
 
 /*@-onlytrans@*/
     fi->te = xcalloc(1, sizeof(*((rpmte)fi->te)));
