--- rpm-5.4.14/rpmio/rpmdav.c.ne_redirect~	2014-05-17 21:24:08.547223302 +0200
+++ rpm-5.4.14/rpmio/rpmdav.c	2014-05-17 21:29:36.877646356 +0200
@@ -12,6 +12,7 @@
 #include "ne_basic.h"
 #include "ne_dates.h"
 #include "ne_locks.h"
+#include "ne_redirect.h"
 
 #define	NEONBLOWSCHUNKS
 #ifndef	NEONBLOWSCHUNKS
@@ -506,10 +507,11 @@ static int davConnect(urlinfo u)
 	    rc = NE_OK;
 	    break;
 	}
+    case NE_REDIRECT:
 	/* HACK: "301 Moved Permanently" on empty subdir. */
 	if (!strncmp("301 ", ne_get_error((ne_session *)u->sess), sizeof("301 ")-1))
 	    break;
-#ifdef	HACK	/* XXX need davHEAD changes here? */
+	/* XXX need davHEAD changes here? */
 	/* HACK: "302 Found" if URI is missing pesky trailing '/'. */
 	if (!strncmp("302 ", ne_get_error((ne_session *)u->sess), sizeof("302 ")-1)) {
 	    char * t;
@@ -517,7 +519,6 @@ static int davConnect(urlinfo u)
 		*t = '/';
 	    break;
 	}
-#endif
 	errno = EIO;		/* HACK: more precise errno. */
 	goto bottom;
     case NE_LOOKUP:
@@ -543,6 +544,8 @@ static int davInit(const char * url, url
 	/*@modifies *uret, internalState @*/
 {
     urlinfo u = NULL;
+    const ne_uri *newui = NULL;;
+    char *newurl = NULL;
     int rc = 0;
 
 /*@-globs@*/	/* FIX: h_errno annoyance. */
@@ -571,6 +574,7 @@ static int davInit(const char * url, url
 
 	u->capabilities = capabilities = (ne_server_capabilities *) xcalloc(1, sizeof(*capabilities));
 	u->sess = ne_session_create(u->scheme, u->host, u->port);
+ne_redirect_register(u->sess);
 
 	ne_lockstore_register((ne_lock_store *)u->lockstore, (ne_session *)u->sess);
 
@@ -616,6 +620,15 @@ static int davInit(const char * url, url
 
 	/* HACK: where should server capabilities be read? */
 	rc = davConnect(u);
+
+	if (rc == NE_REDIRECT) {
+	    if (!strncmp("301 ", ne_get_error((ne_session *)u->sess), sizeof("301 ")-1)) {
+		loc = ne_redirect_location(u->sess);
+		newurl = ne_uri_unparse(loc);
+	    }
+	    if (!strncmp("302 ", ne_get_error((ne_session *)u->sess), sizeof("302 ")-1))
+		rc = NE_OK;
+	}
 	if (rc)
 	    goto exit;
       }	break;
@@ -626,6 +639,11 @@ DAVDEBUG(-1, (stderr, "<-- %s(%s) u->url
     if (uret != NULL)
 	*uret = urlLink(u, "davInit");
     u = urlFree(u, "urlSplit (davInit)");
+    if (rc == NE_REDIRECT) {
+	rc = davInit(newurl, uret);
+	_free(newurl);
+	ne_uri_free(newuri);
+    }
 
     return rc;
 }
@@ -1550,6 +1568,7 @@ u->url = avx->uri;
 	/* XXX #206066 OPTIONS is ok, but PROPFIND from Stat() fails. */
 	/* rpm -qp --rpmiodebug --davdebug http://people.freedesktop.org/~sandmann/metacity-2.16.0-2.fc6/i386/metacity-2.16.0-2.fc6.i386.rpm */
 
+    case NE_REDIRECT:
 	/* HACK: "301 Moved Permanently" on empty subdir. */
 	if (!strncmp("301 ", ne_get_error((ne_session *)u->sess), sizeof("301 ")-1))
 	    break;
@@ -1743,6 +1762,7 @@ assert(ctrl->req == NULL);
 /*@=nullderef@*/
 /*@-nullpass@*/
     ctrl->req = ne_request_create((ne_session *)u->sess, httpCmd, httpArg);
+    ne_redirect_register(u->sess);
 /*@=nullpass@*/
 assert(ctrl->req != NULL);
 
@@ -1806,8 +1826,6 @@ assert(ctrl->req != NULL);
 	    rc = FTPERR_FILE_NOT_FOUND;
 	    break;
     }
-    if (rc || _dav_debug)
-	fprintf(stderr, "HTTP request sent, awaiting response... %d %s\n", status->code, status->reason_phrase);
 
     if (rc)
 	goto errxit;
