--- rpm-5.4.15/macros/macros.rpmbuild.in.omv~	2015-07-11 15:04:43.750862762 +0200
+++ rpm-5.4.15/macros/macros.rpmbuild.in	2015-07-11 15:06:04.123753033 +0200
@@ -336,7 +336,7 @@ echo "Patch #%{__patch_number} (%{basena
 
 #
 # Define a generic value for optflags. Normally overridden by per-target macros.
-%optflags		-Os
+%optflags		-Os %{?!_disable_lto:-flto}
 
 #
 # Define per-arch and per-os defaults. Normally overridden by per-target macros.
@@ -641,7 +641,7 @@ echo "Patch #%{__patch_number} (%{basena
 #
 %GNUconfigure(MCs:)      \
   CFLAGS="${CFLAGS:-%optflags}" ; export CFLAGS; \
-  LDFLAGS="${LDFLAGS:-%{-s:-s}}"  ; export LDFLAGS; \
+  LDFLAGS="${LDFLAGS:-%{optflags} %{?!_disable_lto:-flto}}"  ; export LDFLAGS; \
   %{-C:_mydir="`pwd`"; %{-M: %{__mkdir} -p "%{-C*}";} cd "%{-C*}"} \
   dirs="`find ${_mydir} -name 'configure.[ai][cn]' -print`"; export dirs; \
     for coin in `echo ${dirs}` \
--- rpm-5.4.15/macros/mandriva.in.omv~	2015-07-11 15:00:59.409157794 +0200
+++ rpm-5.4.15/macros/mandriva.in	2015-07-11 15:04:33.291876906 +0200
@@ -331,7 +331,7 @@ GCONF_CONFIG_SOURCE=`%{_gconftool_bin} -
 
 %__libtoolize_configure %{?__libtoolize:(cd $CONFIGURE_TOP; [ ! -f configure.in -a ! -f configure.ac ] || %{__libtoolize} --copy --force)}
 
-%ldflags %{?!_disable_ld_no_undefined: -Wl,--no-undefined} %{?_hardened_flags}
+%ldflags %{optflags} -Wl,-O2 %{?!_disable_ld_no_undefined: -Wl,--no-undefined} %{?!_disable_lto:-flto} %{?_hardened_flags}
 
 %setup_compile_flags \
   CC="${CC:-%__cc}" ; export CC ; \
--- rpm-5.4.15/platform/aarch64-linux.macros.omv~	2015-07-11 15:06:58.783677454 +0200
+++ rpm-5.4.15/platform/aarch64-linux.macros	2015-07-11 15:07:07.316665591 +0200
@@ -6,7 +6,7 @@
 %_arch                  aarch64
 %_os                    linux
 %_gnu                   -gnu
-%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fPIC}%{!?__common_cflags_with_ssp:-O2 -g}
+%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fPIC}%{!?__common_cflags_with_ssp:-O2 -g} %{?!_disable_lto:-flto}
 #==============================================================================
 # ---- configure macros.
 #
--- rpm-5.4.15/platform/amd64-linux.macros.omv~	2015-07-11 15:07:19.926648032 +0200
+++ rpm-5.4.15/platform/amd64-linux.macros	2015-07-11 15:07:27.895636917 +0200
@@ -6,7 +6,7 @@
 %_arch			x86_64
 %_os			linux
 %_gnu			-gnu
-%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fPIC}%{!?__common_cflags_with_ssp:-O2 -g -m64 -mtune=generic}
+%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fPIC}%{!?__common_cflags_with_ssp:-O2 -g -m64 -mtune=generic} %{?!_disable_lto:-flto}
 
 #==============================================================================
 # ---- configure macros.
--- rpm-5.4.15/platform/armv6j-linux.macros.omv~	2015-07-11 15:07:27.925636875 +0200
+++ rpm-5.4.15/platform/armv6j-linux.macros	2015-07-11 15:07:36.474624936 +0200
@@ -6,7 +6,7 @@
 %_arch			arm
 %_os			linux
 %_gnu			-gnueabihf
-%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fPIC -g -march=armv6j -mfpu=vfp -mfloat-abi=hard}%{!?__common_cflags_with_ssp:-O2 -g}
+%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fPIC -g -march=armv6j -mfpu=vfp -mfloat-abi=hard}%{!?__common_cflags_with_ssp:-O2 -g} %{?!_disable_lto:-flto}
 
 #==============================================================================
 # ---- configure macros.
--- rpm-5.4.15/platform/armv7hl-linux.macros.omv~	2015-07-11 15:07:36.504624894 +0200
+++ rpm-5.4.15/platform/armv7hl-linux.macros	2015-07-11 15:07:42.625616336 +0200
@@ -6,7 +6,7 @@
 %_arch			arm
 %_os			linux
 %_gnu			-gnueabihf
-%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fPIC -g -march=armv7-a -mfloat-abi=hard -mfpu=neon}%{!?__common_cflags_with_ssp:-O2}
+%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fPIC -g -march=armv7-a -mfloat-abi=hard -mfpu=neon}%{!?__common_cflags_with_ssp:-O2} %{?!_disable_lto:-flto}
 
 #==============================================================================
 # ---- configure macros.
--- rpm-5.4.15/platform/armv7hnl-linux.macros.omv~	2015-07-11 15:07:42.655616294 +0200
+++ rpm-5.4.15/platform/armv7hnl-linux.macros	2015-07-11 15:07:47.390609668 +0200
@@ -6,7 +6,7 @@
 %_arch			arm
 %_os			linux
 %_gnu			-gnueabihf
-%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fPIC -g -march=armv7-a -mfloat-abi=hard -mfpu=neon}%{!?__common_cflags_with_ssp:-O2}
+%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fPIC -g -march=armv7-a -mfloat-abi=hard -mfpu=neon}%{!?__common_cflags_with_ssp:-O2} %{?!_disable_lto:-flto}
 
 #==============================================================================
 # ---- configure macros.
--- rpm-5.4.15/platform/armv7l-linux.macros.omv~	2015-07-11 15:07:47.420609626 +0200
+++ rpm-5.4.15/platform/armv7l-linux.macros	2015-07-11 15:07:52.136603023 +0200
@@ -6,7 +6,7 @@
 %_arch			arm
 %_os			linux
 %_gnu			-gnueabi
-%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fPIC -g -march=armv7-a -mfloat-abi=softfp -mfpu=vfpv3-d16}%{!?__common_cflags_with_ssp:-O2}
+%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fPIC -g -march=armv7-a -mfloat-abi=softfp -mfpu=vfpv3-d16}%{!?__common_cflags_with_ssp:-O2} %{?!_disable_lto:-flto}
 
 #==============================================================================
 # ---- configure macros.
--- rpm-5.4.15/platform/armv7nhl-linux.macros.omv~	2015-07-11 15:07:52.167602979 +0200
+++ rpm-5.4.15/platform/armv7nhl-linux.macros	2015-07-11 15:07:56.477596940 +0200
@@ -6,7 +6,7 @@
 %_arch			arm
 %_os			linux
 %_gnu			-gnueabihf
-%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fPIC -g -march=armv7-a -mfloat-abi=hard -mfpu=neon}%{!?__common_cflags_with_ssp:-O2}
+%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fPIC -g -march=armv7-a -mfloat-abi=hard -mfpu=neon}%{!?__common_cflags_with_ssp:-O2} %{?!_disable_lto:-flto}
 
 #==============================================================================
 # ---- configure macros.
--- rpm-5.4.15/platform/athlon-linux.macros.omv~	2015-07-11 15:07:56.513596890 +0200
+++ rpm-5.4.15/platform/athlon-linux.macros	2015-07-11 15:08:02.545588431 +0200
@@ -6,7 +6,7 @@
 %_arch			i386
 %_os			linux
 %_gnu			-gnu
-%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fomit-frame-pointer}%{!?__common_cflags_with_ssp:-O2 -g -m32} -march=athlon -fasynchronous-unwind-tables
+%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fomit-frame-pointer}%{!?__common_cflags_with_ssp:-O2 -g -m32} -march=athlon -fasynchronous-unwind-tables %{?!_disable_lto:-flto}
 
 #==============================================================================
 # ---- configure macros.
--- rpm-5.4.15/platform/i386-linux.macros.omv~	2015-07-11 15:08:02.574588391 +0200
+++ rpm-5.4.15/platform/i386-linux.macros	2015-07-11 15:08:07.874580952 +0200
@@ -6,7 +6,7 @@
 %_arch			i386
 %_os			linux
 %_gnu			-gnu
-%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fomit-frame-pointer}%{!?__common_cflags_with_ssp:-O2 -g -m32} -march=i386 -mtune=generic -fasynchronous-unwind-tables
+%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fomit-frame-pointer}%{!?__common_cflags_with_ssp:-O2 -g -m32} -march=i386 -mtune=generic -fasynchronous-unwind-tables %{?!_disable_lto:-flto}
 
 #==============================================================================
 # ---- configure macros.
--- rpm-5.4.15/platform/i486-linux.macros.omv~	2015-07-11 15:08:07.909580903 +0200
+++ rpm-5.4.15/platform/i486-linux.macros	2015-07-11 15:08:12.387574614 +0200
@@ -6,7 +6,7 @@
 %_arch			i386
 %_os			linux
 %_gnu			-gnu
-%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fomit-frame-pointer}%{!?__common_cflags_with_ssp:-O2 -g -m32} -march=i486 -fasynchronous-unwind-tables
+%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fomit-frame-pointer}%{!?__common_cflags_with_ssp:-O2 -g -m32} -march=i486 -fasynchronous-unwind-tables %{?!_disable_lto:-flto}
 
 #==============================================================================
 # ---- configure macros.
--- rpm-5.4.15/platform/i586-linux.macros.omv~	2015-07-11 15:08:12.416574574 +0200
+++ rpm-5.4.15/platform/i586-linux.macros	2015-07-11 15:08:17.465567478 +0200
@@ -6,7 +6,7 @@
 %_arch			i386
 %_os			linux
 %_gnu			-gnu
-%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fomit-frame-pointer -mtune=atom}%{!?__common_cflags_with_ssp:-O2 -g -m32} -march=i586 -fasynchronous-unwind-tables
+%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fomit-frame-pointer -mtune=atom}%{!?__common_cflags_with_ssp:-O2 -g -m32} -march=i586 -fasynchronous-unwind-tables %{?!_disable_lto:-flto}
 
 #==============================================================================
 # ---- configure macros.
--- rpm-5.4.15/platform/i686-linux.macros.omv~	2015-07-11 15:08:17.501567428 +0200
+++ rpm-5.4.15/platform/i686-linux.macros	2015-07-11 15:08:23.176559446 +0200
@@ -6,7 +6,7 @@
 %_arch			i386
 %_os			linux
 %_gnu			-gnu
-%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fomit-frame-pointer}%{!?__common_cflags_with_ssp:-O2 -g -m32 -mtune=generic} -march=i686 -fasynchronous-unwind-tables
+%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fomit-frame-pointer}%{!?__common_cflags_with_ssp:-O2 -g -m32 -mtune=generic} -march=i686 -fasynchronous-unwind-tables %{?!_disable_lto:-flto}
 
 #==============================================================================
 # ---- configure macros.
--- rpm-5.4.15/platform/ia32e-linux.macros.omv~	2015-07-11 15:08:23.206559404 +0200
+++ rpm-5.4.15/platform/ia32e-linux.macros	2015-07-11 15:08:28.828551491 +0200
@@ -6,7 +6,7 @@
 %_arch			x86_64
 %_os			linux
 %_gnu			-gnu
-%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fPIC}%{!?__common_cflags_with_ssp:-O2 -g -m64 -mtune=generic}
+%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fPIC}%{!?__common_cflags_with_ssp:-O2 -g -m64 -mtune=generic} %{?!_disable_lto:-flto}
 
 #==============================================================================
 # ---- configure macros.
--- rpm-5.4.15/platform/k6-linux.macros.omv~	2015-07-11 15:08:31.959547082 +0200
+++ rpm-5.4.15/platform/k6-linux.macros	2015-07-11 15:08:37.484539296 +0200
@@ -6,7 +6,7 @@
 %_arch			i386
 %_os			linux
 %_gnu			-gnu
-%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fomit-frame-pointer}%{!?__common_cflags_with_ssp:-O2 -g -m32} -march=k6 -fasynchronous-unwind-tables
+%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fomit-frame-pointer}%{!?__common_cflags_with_ssp:-O2 -g -m32} -march=k6 -fasynchronous-unwind-tables %{?!_disable_lto:-flto}
 
 #==============================================================================
 # ---- configure macros.
--- rpm-5.4.15/platform/mipsel-linux.macros.omv~	2015-07-11 15:08:37.514539254 +0200
+++ rpm-5.4.15/platform/mipsel-linux.macros	2015-07-11 15:08:43.302531091 +0200
@@ -6,7 +6,7 @@
 %_arch			mipsel
 %_os			linux
 %_gnu			-gnu
-%optflags		%{?__common_cflags:%{__common_cflags} -march=mips3 -mtune=loongson2f}%{!?__common_cflags:-O2 -g -m32}
+%optflags		%{?__common_cflags:%{__common_cflags} -march=mips3 -mtune=loongson2f}%{!?__common_cflags:-O2 -g -m32} %{?!_disable_lto:-flto}
 
 #==============================================================================
 # ---- configure macros.
--- rpm-5.4.15/platform/mips-linux.macros.omv~	2015-07-11 15:08:43.332531049 +0200
+++ rpm-5.4.15/platform/mips-linux.macros	2015-07-11 15:08:49.393522495 +0200
@@ -6,7 +6,7 @@
 %_arch			mips
 %_os			linux
 %_gnu			-gnu
-%optflags		%{?__common_cflags:%{__common_cflags} -march=mips3}%{!?__common_cflags:-O2 -g -m32}
+%optflags		%{?__common_cflags:%{__common_cflags} -march=mips3}%{!?__common_cflags:-O2 -g -m32} %{?!_disable_lto:-flto}
 
 #==============================================================================
 # ---- configure macros.
--- rpm-5.4.15/platform/pentium3-linux.macros.omv~	2015-07-11 15:08:53.588516571 +0200
+++ rpm-5.4.15/platform/pentium3-linux.macros	2015-07-11 15:08:58.678509378 +0200
@@ -6,7 +6,7 @@
 %_arch			i386
 %_os			linux
 %_gnu			-gnu
-%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fomit-frame-pointer}%{!?__common_cflags_with_ssp:-O2 -g -m32} -march=pentium3 -fasynchronous-unwind-tables
+%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fomit-frame-pointer}%{!?__common_cflags_with_ssp:-O2 -g -m32} -march=pentium3 -fasynchronous-unwind-tables %{?!_disable_lto:-flto}
 
 #==============================================================================
 # ---- configure macros.
--- rpm-5.4.15/platform/pentium4-linux.macros.omv~	2015-07-11 15:08:58.707509337 +0200
+++ rpm-5.4.15/platform/pentium4-linux.macros	2015-07-11 15:09:03.410502687 +0200
@@ -6,7 +6,7 @@
 %_arch			i386
 %_os			linux
 %_gnu			-gnu
-%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fomit-frame-pointer}%{!?__common_cflags_with_ssp:-O2 -g -m32} -march=pentium4 -fasynchronous-unwind-tables
+%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fomit-frame-pointer}%{!?__common_cflags_with_ssp:-O2 -g -m32} -march=pentium4 -fasynchronous-unwind-tables %{?!_disable_lto:-flto}
 
 #==============================================================================
 # ---- configure macros.
--- rpm-5.4.15/platform/ppc64-linux.macros.omv~	2015-07-11 15:09:16.846483667 +0200
+++ rpm-5.4.15/platform/ppc64-linux.macros	2015-07-11 15:09:23.596474099 +0200
@@ -6,7 +6,7 @@
 %_arch			ppc64
 %_os			linux
 %_gnu			-gnu
-%optflags		-O2 -g -m64 -mminimal-toc
+%optflags		-O2 -g -m64 -mminimal-toc %{?!_disable_lto:-flto}
 
 #==============================================================================
 # ---- configure macros.
--- rpm-5.4.15/platform/ppc-linux.macros.omv~	2015-07-11 15:09:29.539465669 +0200
+++ rpm-5.4.15/platform/ppc-linux.macros	2015-07-11 15:09:34.939458004 +0200
@@ -6,7 +6,7 @@
 %_arch			ppc
 %_os			linux
 %_gnu			-gnu
-%optflags		-O2 -g -m32
+%optflags		-O2 -g -m32 %{?!_disable_lto:-flto}
 
 #==============================================================================
 # ---- configure macros.
--- rpm-5.4.15/platform/x32-linux.macros.omv~	2015-07-11 15:09:54.748429845 +0200
+++ rpm-5.4.15/platform/x32-linux.macros	2015-07-11 15:10:06.301413392 +0200
@@ -6,7 +6,7 @@
 %_arch			x86_64
 %_os			linux
 %_gnu			-gnux32
-%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fPIC}%{!?__common_cflags_with_ssp:-Os -g} -mx32 -march=atom -mtune=core-avx-i
+%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fPIC}%{!?__common_cflags_with_ssp:-Os -g} -mx32 -march=atom -mtune=core-avx-i %{?!_disable_lto:-flto}
 
 #==============================================================================
 # ---- configure macros.
--- rpm-5.4.15/platform/x86_64-linux.macros.omv~	2015-07-11 15:10:06.344413331 +0200
+++ rpm-5.4.15/platform/x86_64-linux.macros	2015-07-11 15:10:12.476404589 +0200
@@ -6,7 +6,7 @@
 %_arch			x86_64
 %_os			linux
 %_gnu			-gnu
-%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fPIC}%{!?__common_cflags_with_ssp:-O2 -g -m64 -mtune=generic}
+%optflags		%{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fPIC}%{!?__common_cflags_with_ssp:-O2 -g -m64 -mtune=generic} %{?!_disable_lto:-flto}
 
 #==============================================================================
 # ---- configure macros.
