--- rpm-5.4.14/macros/mandriva.in.ro~	2013-12-05 23:44:16.023499495 +0100
+++ rpm-5.4.14/macros/mandriva.in	2013-12-05 23:48:34.997058263 +0100
@@ -1,7 +1,9 @@
-%distribution		Moondrake GNU/Linux
-%vendor			Moondrake
+%.distribution		Moondrake GNU/Linux
+%.vendor		Moondrake
 %bugurl			http://qa.mandriva.com
-%disturl		http://moondrake.net
+%.disturl		http://moondrake.org
+%._distribution         moondrake_gnu_linux
+%._vendor		moondrake
 
 # This macro will disable the transaction lock on /var/lib/rpm/__db.*.
 # This lock is of no use to us and will also result in errors when trying to
@@ -39,7 +41,7 @@
 # %release with such.
 %evr_tuple_select	1234
 %evr_tuple_order	EVRD
-%disttag		%{!?distsuffix:mdk}%{?distsuffix}
+%.disttag		mdk
 %distepoch		%(sed -e 's#.*release\\ \\(\\S*\\).*#\\1#' /etc/release)
 
 # Dynamic EVRD tags macro for use with versioned provides/requires in place of
