--- rpm-5.4.14/build/files.c.nodefaultdocdirs~	2014-06-16 01:51:22.769932024 +0200
+++ rpm-5.4.14/build/files.c	2014-06-16 01:51:27.776643751 +0200
@@ -2378,6 +2378,7 @@ static rpmRC processPackageFiles(Spec sp
 #if defined(RPM_VENDOR_OPENPKG) /* no-default-doc-files */
     /* do not declare any files as %doc files by default. */
 #else
+    if (!rpmExpandNumeric("%{_no_default_doc_files}")) {
     fl.docDirs[fl.docDirCount++] = xstrdup("/usr/doc");
     fl.docDirs[fl.docDirCount++] = xstrdup("/usr/man");
     fl.docDirs[fl.docDirCount++] = xstrdup("/usr/info");
@@ -2391,6 +2392,7 @@ static rpmRC processPackageFiles(Spec sp
     fl.docDirs[fl.docDirCount++] = rpmGetPath("%{_infodir}", NULL);
     fl.docDirs[fl.docDirCount++] = rpmGetPath("%{_javadocdir}", NULL);
     fl.docDirs[fl.docDirCount++] = rpmGetPath("%{_examplesdir}", NULL);
+    }
 #endif
     
     fl.fileList = NULL;
