--- rpm-5.4.14/lib/librpm.vers.etcrpm~	2013-12-04 08:53:04.947014978 +0000
+++ rpm-5.4.14/lib/librpm.vers	2013-12-04 08:53:40.304168486 +0000
@@ -401,6 +401,7 @@ LIBRPM_0
     rpmVerifyPoptTable;
     rpmVerifySignatures;
     RPMVERSION;
+    setRuntimeRelocPaths;
     showQueryPackage;
     showVerifyPackage;
     specedit;
--- rpm-5.4.14/lib/poptALL.c.etcrpm~	2013-12-04 07:53:21.391012686 +0000
+++ rpm-5.4.14/lib/poptALL.c	2013-12-04 07:54:50.995578804 +0000
@@ -6,7 +6,7 @@
 #include "system.h"
 extern const char *__progname;
 
-#if defined(RPM_VENDOR_WINDRIVER)
+#if defined(RPM_VENDOR_WINDRIVER) || defined(RPM_VENDOR_MANDRIVA)
 const char *__usrlibrpm = USRLIBRPM;
 const char *__etcrpm = SYSCONFIGDIR;
 #endif
@@ -625,7 +625,7 @@ static inline int checkfd(const char * d
     return ret;
 }
 
-#if defined(RPM_VENDOR_WINDRIVER)
+#if defined(RPM_VENDOR_WINDRIVER) || defined(RPM_VENDOR_MANDRIVA)
 void setRuntimeRelocPaths(void)
 {
     /* 
@@ -699,7 +699,7 @@ rpmcliInit(int argc, char *const argv[],
 #endif
    }
 
-#if defined(RPM_VENDOR_WINDRIVER)
+#if defined(RPM_VENDOR_WINDRIVER) || defined(RPM_VENDOR_MANDRIVA)
     (void) setRuntimeRelocPaths();
 #endif
 
@@ -785,7 +785,7 @@ rpmcliInit(int argc, char *const argv[],
 	rpmlog(RPMLOG_WARNING, "existing POPT configuration file \"%s\" considered INSECURE -- not loaded\n", rpmpoptfiles);
 #endif
 
-#if defined(RPM_VENDOR_WINDRIVER)
+#if defined(RPM_VENDOR_WINDRIVER) || defined(RPM_VENDOR_MANDRIVA)
     {	const char * poptAliasFn = rpmGetPath(__usrlibrpm, "/rpmpopt", NULL);
 	(void) poptReadConfigFile(optCon, poptAliasFn);
 	poptAliasFn = _free(poptAliasFn);
@@ -796,7 +796,7 @@ rpmcliInit(int argc, char *const argv[],
     /* XXX FIXME: the 2nd arg useEnv flag is UNUSED. */
     (void) poptReadDefaultConfig(optCon, 1);
 
-#if defined(RPM_VENDOR_WINDRIVER)
+#if defined(RPM_VENDOR_WINDRIVER) || defined(RPM_VENDOR_MANDRIVA)
     {	const char * poptExecPath = rpmGetPath(__usrlibrpm, NULL);
 	poptSetExecPath(optCon, poptExecPath, 1);
 	poptExecPath = _free(poptExecPath);
--- rpm-5.4.14/lib/rpmrc.c.etcrpm~	2013-07-29 05:11:18.000000000 +0000
+++ rpm-5.4.14/lib/rpmrc.c	2013-12-04 09:16:55.916737716 +0000
@@ -310,7 +310,7 @@ export RPM_PACKAGE_NAME RPM_PACKAGE_VERS
 export RPM_BUILD_ROOT\n}\
 ";
 
-#if defined(RPM_VENDOR_WINDRIVER)
+#if defined(RPM_VENDOR_WINDRIVER) || defined(RPM_VENDOR_MANDRIVA)
 /*@unchecked@*/
 extern const char * __usrlibrpm;
 /*@unchecked@*/
@@ -322,7 +322,7 @@ static void setDefaults(void)
 	/*@modifies rpmGlobalMacroContext, internalState @*/
 {
 
-#if defined(RPM_VENDOR_WINDRIVER)
+#if defined(RPM_VENDOR_WINDRIVER) || defined(RPM_VENDOR_MANDRIVA)
     addMacro(NULL, "_usrlibrpm", NULL, __usrlibrpm, RMIL_DEFAULT);
     addMacro(NULL, "_etcrpm", NULL, __etcrpm, RMIL_DEFAULT);
     addMacro(NULL, "_vendor", NULL, "%{?_host_vendor}%{!?_host_vendor:wrs}", RMIL_DEFAULT);
@@ -694,7 +694,7 @@ static void defaultMachine(/*@out@*/ con
     int rc;
 
     while (!gotDefaults) {
-#if defined(RPM_VENDOR_WINDRIVER)
+#if defined(RPM_VENDOR_WINDRIVER) || defined(RPM_VENDOR_MANDRIVA)
 	const char * _platform = rpmGetPath(__etcrpm, "/platform", NULL);
 #else
 	const char * _platform = platform;
@@ -780,7 +780,7 @@ static void defaultMachine(/*@out@*/ con
 	if (cp != NULL && cp != _platform)
 	    cp = _free(cp);
 #endif
-#if defined(RPM_VENDOR_WINDRIVER)
+#if defined(RPM_VENDOR_WINDRIVER) || defined(RPM_VENDOR_MANDRIVA)
 	_platform = _free(_platform);
 #endif
 
