--- rpm-5.4.14/scripts/rubygems.rb.rubyabiver~	2014-03-16 08:07:27.652980213 +0100
+++ rpm-5.4.14/scripts/rubygems.rb	2014-03-16 08:19:23.104361204 +0100
@@ -41,6 +41,7 @@ for path in $stdin.readlines
   # therefore provide ruby(abi) = version
   if provides and path.match(RbConfig::CONFIG["archdir"] + "/rbconfig.rb")
      abi_provide = true
+     ruby_versioned = true
   elsif path.match(specpatt)
     ruby_versioned = true
     gems.push(path.chomp)
