--- rpm-5.4.14/build/parseSpec.c.srcfilename~	2014-06-03 16:25:21.634980410 +0200
+++ rpm-5.4.14/build/parseSpec.c	2014-06-03 16:25:28.286049682 +0200
@@ -511,17 +511,9 @@ static inline int genSourceRpmName(Spec
 		internalState @*/
 {
     if (spec->sourceRpmName == NULL) {
-	const char *N, *V, *R;
-	char fileName[BUFSIZ];
-
-	(void) headerNEVRA(spec->packages->header, &N, NULL, &V, &R, NULL);
-	(void) snprintf(fileName, sizeof(fileName), "%s-%s-%s.%ssrc.rpm",
-			N, V, R, spec->noSource ? "no" : "");
-	fileName[sizeof(fileName)-1] = '\0';
-	N = _free(N);
-	V = _free(V);
-	R = _free(R);
-	spec->sourceRpmName = xstrdup(fileName);
+	const char *srcFormat = rpmExpand("%{_srcrpmfilename}",  NULL);
+        spec->sourceRpmName = headerSprintf(spec->packages->header, srcFormat, NULL,
+			       rpmHeaderFormats, NULL);
     }
 
     return 0;
--- rpm-5.4.14/lib/rpmrc.c.srcfilename~	2014-06-03 16:25:00.618760343 +0200
+++ rpm-5.4.14/lib/rpmrc.c	2014-06-03 16:25:03.349789045 +0200
@@ -345,6 +345,8 @@ static void setDefaults(void)
 
     addMacroDefault("_rpmfilename",
 	"%%{ARCH}/%%{NAME}-%%{VERSION}-%%{RELEASE}.%%{ARCH}.rpm",NULL);
+    addMacroDefault("_srcrpmfilename",
+	"%%{NAME}-%%{VERSION}-%%{RELEASE}.%%|NOSOURCE?{no}|src.rpm", NULL);
 
     addMacroDefault("optflags",
 		"-O2 -g",			NULL);
