--- rpm-5.4.15/macros/mono.in.orig	2015-08-06 20:19:22.065936749 +0100
+++ rpm-5.4.15/macros/mono.in	2015-08-06 20:19:37.401936164 +0100
@@ -6,5 +6,5 @@
 #
 # Note: Used iff _use_internal_dependency_generator is non-zero. The
 # helpers are also used by %{_rpmhome}/rpmdeps {--provides|--requires}.
-%__mono_provides	%{_rpmhome}/mono-find-provides
-%__mono_requires	%{_rpmhome}/mono-find-requires
+%__mono_provides	/usr/bin/mono-find-provides
+%__mono_requires	/usr/bin/mono-find-requires
