From e0abf810a0733cb9ce9ad66cdbfeaf8e8a2cbb71 Mon Sep 17 00:00:00 2001
From: Ruediger Oertel <ro@suse.de>
Date: Mon, 9 Feb 2009 14:16:52 +0100
Subject: [PATCH] Define DistTag as optional tag with macro just like DistURL

---
 build/parsePreamble.c |  1 +
 macros.in             | 10 ++++++++++
 2 files changed, 11 insertions(+)

diff --git a/build/parsePreamble.c b/build/parsePreamble.c
index 7d89617c2..6ae39ada6 100644
--- a/build/parsePreamble.c
+++ b/build/parsePreamble.c
@@ -519,6 +519,7 @@ static struct optionalTag {
     { RPMTAG_PACKAGER,		"%{packager}" },
     { RPMTAG_DISTRIBUTION,	"%{distribution}" },
     { RPMTAG_DISTURL,		"%{disturl}" },
+    { RPMTAG_DISTTAG,		"%{disttag}" },
     { RPMTAG_BUGURL,		"%{bugurl}" },
     { -1, NULL }
 };
diff --git a/macros.in b/macros.in
index b686701b2..5f20f4680 100644
--- a/macros.in
+++ b/macros.in
@@ -309,6 +309,16 @@ package or when debugging this package.\
 #
 #%disturl
 
+#	Configurable distribution tag, same as DistTag: tag in a specfile.
+#	The tag will be used to supply reliable information to tools like
+#	rpmfind.
+#
+# Note: You should not configure with disttag (or build packages with
+# the DistTag: tag) unless you are willing to supply content in a
+# yet-to-be-determined format at the tag specified.
+#
+#%disttag
+
 #	Configurable bug URL, same as BugURL: tag in a specfile.
 #	The URL will be used to supply reliable information to where
 #	to file bugs.
-- 
2.15.1

