diff -Naur rpm-5.4.14/build/parsePrep.c rpm-5.4.14.tpg/build/parsePrep.c
--- rpm-5.4.14/build/parsePrep.c	2013-07-26 08:19:37.000000000 +0000
+++ rpm-5.4.14.tpg/build/parsePrep.c	2014-11-11 15:38:46.766767930 +0000
@@ -300,7 +300,7 @@
         tar = xstrdup("tar");
     }
 
-#if defined(RPM_VENDOR_ARK) /* use-gnu-tar-compression-detection */
+#if defined(RPM_VENDOR_ARK) || definied(RPM_VENDOR_MANDRIVA) /* use-gnu-tar-compression-detection */
 /* We leave compression handling for all tar based files up to GNU tar */
     if (compressed == COMPRESSED_ZIP)
 #else
