Index: rpm/lib/transaction.c
RCS File: /v/rpm/cvs/rpm/lib/transaction.c,v
rcsdiff -q -kk '-r1.429.2.12' '-r1.429.2.13' -u '/v/rpm/cvs/rpm/lib/transaction.c,v' 2>/dev/null
--- rpm/lib/transaction.c	2012/06/04 15:10:11	1.429.2.12
+++ rpm/lib/transaction.c	2012/12/04 18:48:32	1.429.2.13
@@ -2244,9 +2244,11 @@
     }
 
     /* ===============================================
-     * Save removed files before erasing.
+     * Save removed files before erasing (w/o --justdb).
      */
-    if (TSF_ISSET(tsflags, DIRSTASH) || TSF_ISSET(tsflags, REPACKAGE)) {
+    if ((TSF_ISSET(tsflags, DIRSTASH) || TSF_ISSET(tsflags, REPACKAGE))
+     && !TSF_ISSET(tsflags, JUSTDB))
+    {
 	xx = rpmtsRepackage(ts, numRemoved);
     }
 
