--- rpm-5.4.15/python/rpmmodule.c.py_confread~	2015-01-03 15:07:28.392144226 +0100
+++ rpm-5.4.15/python/rpmmodule.c	2015-01-03 15:10:01.515880948 +0100
@@ -391,7 +391,15 @@ void init_rpm(void)
     if (Py_AtExit(rpm_exithook) == -1)
 	return;
 
-    rpmReadConfigFiles(NULL, NULL);
+    /* a bit hackishly way to determine whether configuration has been read
+     * already or not, as unconditionally (re-)reading configuration files
+     * will redefine all macros already defined when using ie. embedded
+     * interpreter, for where it would've redefined all macros at first
+     * invocation in spec files. Since %_target_os should always be set
+     * by rpmReadConfigFiles(), we'll check for whether it's been defined.
+     */
+    if (rpmExpandNumeric("%{?!_target_os:1}"))
+	rpmReadConfigFiles(NULL, NULL);
 
     d = PyModule_GetDict(m);
 
