--- rpm-5.4.15/configure.ac.strip_gnu~	2015-01-03 23:46:32.135381088 +0100
+++ rpm-5.4.15/configure.ac	2015-01-03 23:49:33.666068447 +0100
@@ -2270,8 +2270,8 @@ if echo "$target_os" | grep '.*-gnulibc1
 	target_os_gnu="-gnulibc1"
 fi
 if echo "$target_os" | grep '.*-gnu' > /dev/null ; then
-	target_os=`echo "${target_os}" | sed 's/-gnu$//'`
-	target_os_gnu="-gnu"
+	target_os_gnu="`echo ${target_os} | sed 's/.*-gnu/-gnu/'`"
+	target_os=`echo "${target_os}" | sed 's/-gnu.*//'`
 fi
 changequote(<, >)
 target_os_exact="${target_os}"
@@ -2303,6 +2303,22 @@ else
     ln -s ${srcdir}/autodeps/none ./find-requires
 fi
 
+dnl # get rid of -gnu from host_os as well...
+host_os_gnu="%{nil}"
+if echo "$host_os" | grep '.*-gnulibc1' > /dev/null ; then
+	host_os=`echo "${host_os}" | sed 's/-gnulibc1$//'`
+	host_os_gnu="-gnulibc1"
+fi
+if echo "$host_os" | grep '.*-gnu' > /dev/null ; then
+	host_os_gnu="`echo ${host_os} | sed 's/.*-gnu/-gnu/'`"
+	host_os=`echo "${host_os}" | sed 's/-gnu.*//'`
+fi
+changequote(<, >)
+host_os_exact="${host_os}"
+host_os_major=`echo "${host_os}" | sed 's/\..*$//'`
+host_os_noversion=`echo "${host_os}" | sed 's/[0-9]*\..*$//'`
+changequote([, ])
+
 dnl # determine the canonical arch-vendor-os for the build machine
 autorelocate_path='%{nil}'
 autorelocate_dcolor='0'
