--- rpm-5.4.10/macros/mandriva.in.nosilentrules~	2012-10-24 07:24:59.333164843 +0200
+++ rpm-5.4.10/macros/mandriva.in	2012-10-24 07:25:00.871187345 +0200
@@ -367,6 +367,7 @@ GCONF_CONFIG_SOURCE=`%{_gconftool_bin} -
   [ -f $CONFIGURE_TOP/configure.in -o -f $CONFIGURE_TOP/configure.ac ] && \
   CONFIGURE_XPATH="--x-includes=%{_prefix}/include --x-libraries=%{_prefix}/%{_lib}" \
   $CONFIGURE_TOP/configure %{_target_platform} \\\
+  	--disable-silent-rules \\\
 	--program-prefix=%{?_program_prefix} \\\
  	--prefix=%{_prefix} \\\
 	--exec-prefix=%{_exec_prefix} \\\
@@ -389,6 +390,7 @@ GCONF_CONFIG_SOURCE=`%{_gconftool_bin} -
   [ -f $CONFIGURE_TOP/configure.in -o -f $CONFIGURE_TOP/configure.ac ] && \
   CONFIGURE_XPATH="--x-includes=%{_prefix}/include --x-libraries=%{_prefix}/%{_lib}" \
   $CONFIGURE_TOP/configure --build=%{_target_platform} \\\
+  	--disable-silent-rules \\\
  	--prefix=%{_prefix} \\\
 	--exec-prefix=%{_exec_prefix} \\\
 	--bindir=%{_bindir} \\\
