--- rpm-5.4.15/macros/mandriva.in.omv~	2016-05-17 00:55:55.631204147 +0200
+++ rpm-5.4.15/macros/mandriva.in	2016-05-17 00:56:37.296975447 +0200
@@ -306,7 +306,7 @@ GCONF_CONFIG_SOURCE=`%{_gconftool_bin} -
 # - "9.1"    =>  910
 # - "10.2.2" => 1022
 # (user may copy the following line in specfile)
-%mdkversion		%(perl -pe '/(\\d+)\\.(\\d)\\.?(\\d)?/; $_="$1$2".($3||0)' /etc/mandriva-release)
+%mdkversion		%(cat /etc/os-release|grep VERSION_ID= |sed -e 's,.*=,,;s,\",,g;s,\.,0,g')
 
 #==============================================================================
 # ---- Build policy macros.
