--- rpm-5.4.14/macros/macros.rpmbuild.in.armx~	2014-02-10 17:06:27.037770022 +0100
+++ rpm-5.4.14/macros/macros.rpmbuild.in	2014-02-10 17:07:49.077770359 +0100
@@ -680,7 +680,10 @@ done \
 %efi %{ix86} x86_64 ia64
 
 # same thing for arm
-%arm    arm armv3 armv3l armv3b armv4 armv4l armv4b armv5 armv5l armv5b armv5te armv5tel armv5teb xscale armv6 armv6l armv7l armv7hl armv7lh armv7hln armv7hnl armv7lhn armv7lnh armv7nhl armv7nlh
+%arm    arm armv3 armv3l armv3b armv4 armv4l armv4b armv5 armv5l armv5b armv5te armv5tel armv5teb xscale armv6 armv6l armv6j armv7l armv7hl armv7lh armv7hln armv7hnl armv7lhn armv7lnh armv7nhl armv7nlh
+
+# macro that contains all arm archs, even 64 bit ones
+%armx %arm aarch64
 
 # same thing for leg
 %leg    leg legv3 legv3l legv3b legv4 legv4l legv4b legv5 legv5l legv5b legv5te legv5tel legv5teb xscale legv6 legv6l legv7l legv7hl legv7lh legv7hln legv7hnl legv7lhn legv7lnh legv7nhl legv7nlh
