--- rpm-5.4.4/build/parseSpec.c.specfile~	2011-11-11 13:41:29.137827186 +0100
+++ rpm-5.4.4/build/parseSpec.c	2011-11-11 13:42:12.938874762 +0100
@@ -542,6 +542,7 @@ int parseSpec(rpmts ts, const char *spec
      *          /.././../usr/../bin//./sh (XXX FIXME: dots not handled yet)
      */
     spec->specFile = rpmGetPath(specFile, NULL);
+    addMacro(spec->macros, "_specfile", NULL, spec->specFile, RMIL_SPEC);
     spec->fileStack = newOpenFileInfo();
     spec->fileStack->fileName = xstrdup(spec->specFile);
 
