--- rpm-5.4.15/scripts/gem_helper.rb.pld_gemhelper_ruby20~	2014-12-19 18:27:41.581603525 +0100
+++ rpm-5.4.15/scripts/gem_helper.rb	2015-01-03 15:19:12.838933038 +0100
@@ -7,7 +7,6 @@
 
 require 'optparse'
 require 'rubygems'
-require 'rubygems/package'
 
 if ARGV[0] == "build" or ARGV[0] == "install" or ARGV[0] == "spec"
   require 'yaml'
@@ -97,20 +96,21 @@ if ARGV[0] == "build" or ARGV[0] == "ins
     require 'rubygems/ext'
     module Gem::Ext
       class Builder
-	def build_extensions
+	def self.make(dest_path, results)
 	  make_program = ENV['make']
 	  unless make_program then
 	    make_program = (/mswin/ =~ RUBY_PLATFORM) ? 'nmake' : 'make'
 	  end
 	  cmd = make_program
 	  if ENV['GEM_MODE'] == "build"
-	    cmd += " %s" % ENV['jobs']
-	    system("ruby %s" % @spec.extensions)
+  	    cmd += " %s" % ENV['jobs']
 	  elsif ENV['GEM_MODE'] == "install"
 	    cmd += " DESTDIR='%s' install" % ENV['DESTDIR']
 	  end
-	  system(cmd)
-	  raise Gem::ExtensionBuildError, "Make failed!\n" unless
+	  results << cmd
+	  results << `#{cmd} #{redirector}`
+
+	  raise Gem::ExtensionBuildError, "make failed:\n\n#{results}" unless
 	  $?.success?
 	end
       end
@@ -122,9 +122,6 @@ if ARGV[0] == "build" or ARGV[0] == "ins
       	def initialize(spec, options={})
 	  @gem_dir = Dir.pwd
       	  @spec = spec
-	  @options = options
-
-	  process_options
 	end
       end
       class ConfigFile
@@ -180,7 +177,7 @@ if ARGV[0] == "build" or ARGV[0] == "ins
 
     spec = Gem::Specification.from_yaml(YAML.dump(header))
     unless dry_run
-      Gem::Package.build(spec)
+      Gem::Builder.new(spec).build
     else
       files.concat(spec.files)
       print "%s\n" % files.join("\n")
