--- rpm-5.4.14/lib/rpmfc.c.orig	2014-10-03 12:49:21.083158448 +0100
+++ rpm-5.4.14/lib/rpmfc.c	2014-10-03 12:53:34.575139772 +0100
@@ -1253,6 +1253,16 @@
 		    if (mireRegexec(mire, fc->fn[fc->ix], (size_t) 0) >= 0)
 			fc->fcolor->vals[fc->ix] |= RPMFC_SCRIPT;
 		mire = mireFree(mire);
+                mire = mireNew(RPMMIRE_REGEX, RPMTAG_FILEPATHS);
+                if (!mireRegcomp(mire, ".*/usr/share/gems/specifications/.*\\.gemspec$")) {
+                    if (mireRegexec(mire, fc->fn[fc->ix], (size_t) 0) >= 0)
+                    {
+			fc->fcolor->vals[fc->ix] |= RPMFC_RUBY;
+                        fc->fcolor->vals[fc->ix] |= RPMFC_MODULE;
+                    }
+                }
+                mire = mireFree(mire);
+
 #endif
 	    }
 	}
--- rpm-5.4.14/macros/ruby.in	2014-10-03 10:29:42.886463215 +0100
+++ rpm-5.4.14/macros/ruby.in.orig	2014-10-03 10:34:35.560441652 +0100
@@ -24,6 +24,18 @@
 %ruby_gemdir		%(%{__ruby} -rrbconfig -e 'print RbConfig::CONFIG["rubygemsdir"]')
 %ruby_ridir		%(%{__ruby} -rrbconfig -e 'print RbConfig::CONFIG["ridir"])')
 
+# The RubyGems root folder.
+%gem_dir %{_datadir}/gems
+%gem_archdir %{_libdir}/gems
+
+# Common gem locations and files.
+%gem_instdir %{gem_dir}/gems/%{gem_name}-%{version}
+%gem_extdir_mri %{gem_archdir}/%{name}/%{gem_name}-%{version}
+%gem_libdir %{gem_instdir}/lib
+%gem_cache %{gem_dir}/cache/%{gem_name}-%{version}.gem
+%gem_spec %{gem_dir}/specifications/%{gem_name}-%{version}.gemspec
+%gem_docdir %{gem_dir}/doc/%{gem_name}-%{version}
+
 %__gem_helper %{_usrlibrpm}/gem_helper.rb
 
 %gem_build(f:j:) \
@@ -37,6 +49,6 @@
 	DESTDIR=${DESTDIR:-%{buildroot}} \\\
 	%__gem_helper install \\\
 	--env-shebang --rdoc --ri --force --ignore-dependencies \\\
-	%{!-i:--install-dir %{buildroot}%{ruby_gemdir}}%{-i:--install-dir %{-i*}} \\\
+	%{!-i:--install-dir %{buildroot}%{gem_dir}}%{-i:--install-dir %{-i*}} \\\
 	%{!-n:--bindir %{buildroot}%{_bindir}}%{-n:--bindir%{-n*}} \\\
 	%{!-C:--fix-permissions}
--- rpm-5.4.14/scripts/rubygems.rb.org	2014-10-03 10:35:18.196438511 +0100
+++ rpm-5.4.14/scripts/rubygems.rb	2014-10-03 12:57:36.535121946 +0100
@@ -30,7 +30,7 @@
   exit(1)
 end
 
-specpatt = RbConfig::CONFIG["rubylibdir"].sub(RbConfig::CONFIG["ruby_version"], ".*/specifications/.*\.gemspec$")
+specpatt = "/usr/share/gems/specifications/.*\.gemspec$"
 gems = []
 ruby_versioned = false
 abi_provide = false
