--- rpm-5.4.5/scripts/rubygems.rb.rb_newline~	2012-02-29 17:25:34.815069060 +0100
+++ rpm-5.4.5/scripts/rubygems.rb	2012-02-29 18:45:41.587464927 +0100
@@ -61,10 +61,11 @@ for path in $stdin.readlines
 end
 
 if requires or abi_provide
-  print "ruby(abi)"
+  abidep = "ruby(abi)"
   if ruby_versioned
-    print " = %s\n" % RbConfig::CONFIG["ruby_version"]
+    abidep += " = %s" % RbConfig::CONFIG["ruby_version"]
   end
+  print abidep + "\n"
 end
 
 if gems.length > 0
