--- rpm-5.4.15/rpmio/rpmpython.c.system_header~	2014-12-05 20:29:25.896709266 +0100
+++ rpm-5.4.15/rpmio/rpmpython.c	2014-12-05 20:29:30.376697886 +0100
@@ -1,4 +1,5 @@
-#include "config.h"
+#include "system.h"
+
 #if defined(MODULE_EMBED)
 #include <Python.h>
 #include <cStringIO.h>
@@ -10,7 +11,6 @@
 #include <dlfcn.h>
 #include <rpmlog.h>
 #endif
-#include "system.h"
 
 #define	_RPMIOB_INTERNAL	/* XXX necessary? */
 #include <rpmiotypes.h>
