--- rpm-5.4.10/macros/mandriva.in.ldflags~	2012-10-30 09:49:34.586248202 +0100
+++ rpm-5.4.10/macros/mandriva.in	2012-10-31 13:12:19.439849728 +0100
@@ -349,7 +349,8 @@ GCONF_CONFIG_SOURCE=`%{_gconftool_bin} -
   CFLAGS="${CFLAGS:-%optflags}" ; export CFLAGS ; \
   CXXFLAGS="${CXXFLAGS:-%optflags}" ; export CXXFLAGS ; \
   FFLAGS="${FFLAGS:-%optflags}" ; export FFLAGS ; \
-  %(if [ -n '%ldflags' ]; then echo 'LDFLAGS="$LDFLAGS%ldflags"; export LDFLAGS ;'; fi)
+  %{?ldflags:LDFLAGS="${LDFLAGS:-%{ldflags}}"; export LDFLAGS ;} \
+  %{nil}
 
 %before_configure \
   %setup_compile_flags \
