--- rpm-5.4.14/build/parseSpec.c.noarb~	2014-03-15 04:28:49.960823250 +0100
+++ rpm-5.4.14/build/parseSpec.c	2014-03-15 04:30:12.757136461 +0100
@@ -96,6 +96,8 @@ rpmParseState isPart(Spec spec)
         if (aTags != NULL && aTags[0] != NULL) {
             ARGV_t av;
             s = tagCanonicalize(line+1);	/* XXX +1 to skip leading '%' */
+            if (!*s) /* not a tag, just unknown macro */
+                goto exit;
 #if defined(RPM_VENDOR_OPENPKG) /* wildcard-matching-arbitrary-tagnames */
             av = argvSearchLinear(aTags, s, argvFnmatchCasefold);
 #else
