--- rpm-5.4.4/build/parseChangelog.c.clean~	2011-08-15 00:43:47.000000000 +0200
+++ rpm-5.4.4/build/parseChangelog.c	2011-11-12 17:52:00.756971839 +0100
@@ -269,11 +269,7 @@ int parseChangelog(Spec spec)
     /* There are no options to %changelog */
     if ((rc = readLine(spec, STRIP_COMMENTS)) > 0) {
 	iob = rpmiobFree(iob);
-#if defined(RPM_VENDOR_MANDRIVA)
-	return (spec->clean == NULL) ? PART_CLEAN : PART_NONE;
-#else
 	return PART_NONE;
-#endif
     }
     if (rc != RPMRC_OK)
 	return rc;
@@ -285,11 +281,7 @@ int parseChangelog(Spec spec)
 	iob = rpmiobAppend(iob, spec->line, 0);
 	line = _free(line);
 	if ((rc = readLine(spec, STRIP_COMMENTS | STRIP_NOEXPAND)) > 0) {
-#if defined(RPM_VENDOR_MANDRIVA)
-	    nextPart = (spec->clean == NULL) ? PART_CLEAN : PART_NONE;
-#else
 	    nextPart = PART_NONE;
-#endif
 	    break;
 	}
 	if (rc != RPMRC_OK)
--- rpm-5.4.4/build/parseFiles.c.clean~	2011-03-25 18:45:46.000000000 +0100
+++ rpm-5.4.4/build/parseFiles.c	2011-11-12 17:52:00.758971903 +0100
@@ -114,12 +114,7 @@ int parseFiles(Spec spec)
 	}
     }
 
-#if defined(RPM_VENDOR_MANDRIVA)
-    /* if no %clean section, add implicit */
-    rc = (nextPart == PART_NONE && spec->clean == NULL) ? PART_CLEAN : nextPart;
-#else
     rc = nextPart;
-#endif
 
 exit:
     argv = _free(argv);
