diff -up rpm-4.19.90/lib/rpmrc.c.5~ rpm-4.19.90/lib/rpmrc.c
--- rpm-4.19.90/lib/rpmrc.c.5~	2024-04-07 13:15:21.614441840 +0200
+++ rpm-4.19.90/lib/rpmrc.c	2024-04-07 13:17:35.571824891 +0200
@@ -492,10 +492,12 @@ static void setDefaults(void)
     /* macrofiles may be pre-set from --macros */
     if (!macrofiles) {
 	macrofiles = rstrscat(NULL, confdir, "/macros", ":",
+				confdir, "/macros.d/*.macros", ":",
 				confdir, "/macros.d/macros.*", ":",
 				confdir, "/platform/%{_target}/macros", ":",
 				confdir, "/fileattrs/*.attr", ":",
 				confdir, "/" RPM_VENDOR "/macros", ":",
+				SYSCONFDIR "/rpm/macros.d/*.macros", ":",
 				SYSCONFDIR "/rpm/macros.*", ":",
 				SYSCONFDIR "/rpm/macros", ":",
 				SYSCONFDIR "/rpm/%{_target}/macros", ":",
