--- rpm-5.4.10/macros/mandriva.in~	2013-04-02 08:31:46.123220885 +0200
+++ rpm-5.4.10/macros/mandriva.in	2013-04-02 08:33:23.468152733 +0200
@@ -187,6 +187,8 @@ export CFLAGS="%{optflags} -fPIE"; expor
 #	0x00f00 (_RPMVSF_NOHEADER)     --nohdrchk      if set, don't check rpmdb headers
 %_vsflags_query		0xc0c00
 
+# enable nofsync when rebuilding rpmdb
+%__dbi_rebuild nofsync
 
 #	Open all indices before doing chroot(2).
 #
