--- rpm-5.4.10/macros/macros.rpmbuild.in.helper_order~	2012-10-04 16:36:15.673593212 +0200
+++ rpm-5.4.10/macros/macros.rpmbuild.in	2012-10-30 08:59:38.687200958 +0100
@@ -399,9 +399,9 @@ echo "Patch #%{__patch_number} (%{basena
 %{?buildroot:%{__rm} -rf '%{buildroot}'; %{__mkdir_p} '%{buildroot}'}
 %__spec_install_body	%{___build_body}
 %__spec_install_post\
-%{?__debug_package:%{__debug_install_post}}\
 %{__arch_install_post}\
 %{__os_install_post}\
+%{?__debug_package:%{__debug_install_post}}\
 %{nil}
 %__spec_install_template	#!%{__spec_install_shell}\
 %{__spec_install_pre}\
