--- rpm-5.4.14/scripts/rubygems.rb.rubyarchdirs~	2014-03-16 06:14:50.373536514 +0100
+++ rpm-5.4.14/scripts/rubygems.rb	2014-03-16 08:07:27.652980213 +0100
@@ -51,10 +51,16 @@ for path in $stdin.readlines
   elsif not ruby_versioned
     if path.match(RbConfig::CONFIG["rubylibdir"])
       ruby_versioned = true
+    elsif path.match(RbConfig::CONFIG["archdir"])
+      ruby_versioned = true
     elsif path.match(RbConfig::CONFIG["sitelibdir"])
       ruby_versioned = true
+    elsif path.match(RbConfig::CONFIG["sitearchdir"])
+      ruby_versioned = true
     elsif path.match(RbConfig::CONFIG["vendorlibdir"])
       ruby_versioned = true
+    elsif path.match(RbConfig::CONFIG["vendorarchdir"])
+      ruby_versioned = true
     end
   end
 end
