--- rpm-5.4.15/scripts/find-debuginfo.sh.debuginfo_too_many_notes~	2015-02-20 11:12:33.301913402 +0100
+++ rpm-5.4.15/scripts/find-debuginfo.sh	2015-02-20 11:13:21.826814636 +0100
@@ -193,7 +193,8 @@ echo $EXCLUDE_REGEXP
 find "$RPM_BUILD_ROOT" ! -path "${debugdir}/*.debug" -type f \
      		     \( -perm -0100 -or -perm -0010 -or -perm -0001 \) \
 		     -print | LANG=C sort |
-file -N -f - | sed -n -e 's/^\(.*\):[ 	]*.*ELF.*, not stripped/\1/p' |
+file -N -f - | sed -n -e 's/^\(.*\):[ 	]*.*ELF.*, not stripped.*/\1/p' |
+
 xargs --no-run-if-empty stat -c '%h %D_%i %n' |
 while read nlinks inum f; do
   [ -n "$EXCLUDE_REGEXP" ] && grep -E -q "$EXCLUDE_REGEXP" <<< "$f" && \
