--- rpm-5.4.9/lib/rpmgi.c.pattern~	2012-04-15 23:20:58.000000000 +0200
+++ rpm-5.4.9/lib/rpmgi.c	2012-05-17 01:56:21.018481577 +0200
@@ -473,7 +473,7 @@ fprintf(stderr, "*** gi %p key %p[%d]\tm
 		pat = ae;
 		got++;
 	    }
-	    if (got) {
+	    if (got >= 0) {
 if (_rpmgi_debug  < 0)
 fprintf(stderr, "\tav %p[%d]: \"%s\" -> %s ~= \"%s\"\n", gi->argv, (int)(av - gi->argv), *av, tagName((rpmTag)tag), pat);
 		got = rpmmiAddPattern(gi->mi, (rpmTag)tag, RPMMIRE_DEFAULT, pat);
