--- rpm-5.4.10/scripts/perl.req.perl_abireq~	2012-12-19 04:47:18.928428133 +0100
+++ rpm-5.4.10/scripts/perl.req	2012-12-19 04:45:40.698118700 +0100
@@ -59,15 +59,18 @@ if ("@ARGV") {
 
 foreach $module (sort keys %require) {
   if ($module =~ m/^abi$/) {
-      print "perl($module) = $require{$module}\n";
-      next;
+#   in Mandriva Linux the perl library now has a versioned soname, so no need
+#   for explicit version dependency for modules anymore
+#   print "perl($module) = $require{$module}\n";
+    print "perl($module) >= $require{$module}\n";
+    next;
   } elsif ($module =~ m/^api$/) {
-      print "perl(abi) >= $require{$module}\n";
-      next;
+    print "perl(abi) >= $require{$module}\n";
+    next;
   }
   # XXX: skip any modules not starting with upper case letters
   if ($module =~ m/^[^A-Z]/) {
-      next;
+    next;
   }
   if (length($require{$module}) == 0) {
     print "perl($module)\n";
@@ -78,9 +81,9 @@ foreach $module (sort keys %require) {
     # $RPM_* variable when I upgrade.
 
     my $v = qx{ rpm --eval '%perl_convert_version $require{$module}' }; #' workaround vim highlighting issue.. ;)
-    chomp($v);
-    print "perl($module) >= $v\n";
-  }
+  chomp($v);
+  print "perl($module) >= $v\n";
+}
 }
 
 exit 0;
