--- rpm-5.4.15/libtpm/configure.ac.tpm_ac~	2014-12-10 16:28:51.431407206 +0000
+++ rpm-5.4.15/libtpm/configure.ac	2014-12-10 16:29:01.498055226 +0000
@@ -330,7 +330,7 @@ AC_SUBST(LIBTPMS_AVAILABLE)
 
 AC_CONFIG_FILES([
     Makefile
-    libtpm/Makefile libtpm/lib/Makefile libtpm/utils/Makefile
+    libtpm/Makefile libtpm/lib/Makefile
     tpm/Makefile
     tpm_proxy/Makefile tpm_proxy/linux/Makefile
 ])
