--- rpm-5.4.10/lib/psm.c.syslog~	2013-05-15 07:51:35.724425404 +0200
+++ rpm-5.4.10/lib/psm.c	2013-05-15 08:01:05.572296077 +0200
@@ -68,6 +68,10 @@
 
 #include "debug.h"
 
+#if defined(HAVE_SYSLOG)
+#include <syslog.h>
+#endif
+
 #define PATT_ISDIR(patt, len)	((patt[0] == '/' && patt[len-1] == '/') || \
 		(patt[0] == '^' && patt[len-1] == '$' && patt[len-2] == '/'))
 
@@ -3498,13 +3502,13 @@ assert(psm->mi == NULL);
 	    else
 		rc = (rpmRC) rpmdbAdd(rpmtsGetRdb(ts), tid, fi->h, NULL);
 	    (void) rpmswExit(rpmtsOp(ts, RPMTS_OP_DBADD), 0);
-#if defined(HAVE_SYSLOG_H) && defined(RPM_VENDOR_MANDRIVA) /* log-install-remove-to-syslog */
+#if defined(HAVE_SYSLOG) && defined(RPM_VENDOR_MANDRIVA) /* log-install-remove-to-syslog */
     	    {
-		char *s, fmt;
+		char *s, *fmt;
 		fmt = rpmExpand("%{___NVRA}", NULL);
 		s = headerSprintf(fi->h, fmt,
 			rpmTagTable, rpmHeaderFormats, NULL);
-		syslog(LOG_NOTICE, "[RPM] %s installed\n", s);
+		syslog(RPMLOG_NOTICE, "[RPM] %s installed\n", s);
 		fmt = _free(fmt);
 		s = _free(s);
     	    }
@@ -3538,13 +3542,13 @@ assert(psm->te != NULL);
 	(void) rpmswEnter(rpmtsOp(ts, RPMTS_OP_DBREMOVE), 0);
 	rc = (rpmRC) rpmdbRemove(rpmtsGetRdb(ts), rpmtsGetTid(ts), fi->record, NULL);
 	(void) rpmswExit(rpmtsOp(ts, RPMTS_OP_DBREMOVE), 0);
-#if defined(HAVE_SYSLOG_H) && defined(RPM_VENDOR_MANDRIVA) /* log-install-remove-to-syslog */
+#if defined(HAVE_SYSLOG) && defined(RPM_VENDOR_MANDRIVA) /* log-install-remove-to-syslog */
         {
-	    char *s, fmt;
+	    char *s, *fmt;
 	    fmt = rpmExpand("%{___NVRA}", NULL);
 	    s = headerSprintf(fi->h, fmt,
 		    rpmTagTable, rpmHeaderFormats, NULL);
-	    syslog(LOG_NOTICE, "[RPM] %s removed\n", s);
+	    syslog(RPMLOG_NOTICE, "[RPM] %s removed\n", s);
 	    fmt = _free(fmt);
 	    s = _free(s);
 	}
