diff -up rpm-4.14.1/macros.in.bero rpm-4.14.1/macros.in
--- rpm-4.14.1/macros.in.bero	2018-03-24 19:56:49.191713667 +0000
+++ rpm-4.14.1/macros.in	2018-03-24 19:59:12.666217396 +0000
@@ -1179,7 +1179,7 @@ package or when debugging this package.\
 # Macros for further automated spec %setup and patch application
 
 # default to plain patch
-%__scm patch
+%__scm gendiff
 # meh, figure something saner
 %__scm_username rpm-build
 %__scm_usermail <rpm-build>
@@ -1193,7 +1193,7 @@ package or when debugging this package.\
 # Plain patch with backups for gendiff
 %__scm_setup_gendiff(q) %{nil}
 %__scm_apply_gendiff(qp:m:)\
-%{__patch} %{-p:-p%{-p*}} %{-q:-s} --fuzz=%{_default_patch_fuzz} %{_default_patch_flags} -b --suffix ".%{2}"
+%{__patch} %{-p:-p%{-p*}} %{-q:-s} --fuzz=%{_default_patch_fuzz} %{_default_patch_flags} -b --suffix ".%{2}~"
 
 # Mercurial (aka hg)
 %__scm_setup_hg(q)\
