--- rpm-5.4.14/scripts/gem_helper.rb.rubygems2~	2014-03-16 11:17:38.008717554 +0100
+++ rpm-5.4.14/scripts/gem_helper.rb	2014-03-16 11:17:26.239185656 +0100
@@ -7,6 +7,7 @@
 
 require 'optparse'
 require 'rubygems'
+require 'rubygems/package'
 
 if ARGV[0] == "build" or ARGV[0] == "install" or ARGV[0] == "spec"
   require 'yaml'
@@ -177,7 +178,7 @@ if ARGV[0] == "build" or ARGV[0] == "ins
 
     spec = Gem::Specification.from_yaml(YAML.dump(header))
     unless dry_run
-      Gem::Builder.new(spec).build
+      Gem::Package.build(spec)
     else
       files.concat(spec.files)
       print "%s\n" % files.join("\n")
