--- rpm-5.4.9/build/parseDescription.c.mdvbz62979~	2012-05-03 20:39:51.000000000 +0200
+++ rpm-5.4.9/build/parseDescription.c	2012-05-15 01:55:31.332318888 +0200
@@ -92,6 +92,7 @@ int parseDescription(Spec spec)
     }
 
     /* Lose the inheirited %description (if present). */
+#if !defined(RPM_VENDOR_MANDRIVA) /* mdvbz#62979 */
     if (spec->packages->header != pkg->header) {
 	he->tag = RPMTAG_DESCRIPTION;
 	xx = headerGet(pkg->header, he, 0);
@@ -99,6 +100,7 @@ int parseDescription(Spec spec)
 	if (xx && he->t == RPM_STRING_TYPE)
 	    xx = headerDel(pkg->header, he, 0);
     }
+#endif
     
     t = stashSt(spec, pkg->header, RPMTAG_DESCRIPTION, lang);
     
