--- rpm-5.4.15/configure.ac.perlembed_typo~	2014-10-23 11:17:57.243681910 +0200
+++ rpm-5.4.15/configure.ac	2014-10-23 10:06:10.850185308 +0200
@@ -1776,7 +1776,7 @@ RPM_CHECK_LIB(
     [no,external:none], [],
     [ if test ".$RPM_CHECK_LIB_LOCATION" != .internal; then
          LDFLAGS="$LDFLAGS -Wl,-rpath,$RPM_CHECK_LIB_LOCATION"
-      fi
+      fi ]
     [ AM_CONDITIONAL(
       [WITH_PERLEMBED],
       [true])
