diff -up rpm-4.14.0/build/files.c.omv~ rpm-4.14.0/build/files.c
--- rpm-4.14.0/build/files.c.omv~	2018-03-14 21:51:32.078929855 +0100
+++ rpm-4.14.0/build/files.c	2018-03-14 21:54:42.998798137 +0100
@@ -857,6 +857,7 @@ static VFA_t const virtualAttrs[] = {
     { "%license",	RPMFILE_LICENSE },
     { "%pubkey",	RPMFILE_PUBKEY },
     { "%missingok",	RPMFILE_MISSINGOK },
+    { "%optional",	RPMFILE_OPTIONAL },
     { NULL, 0 }
 };
 
@@ -1384,7 +1385,7 @@ static rpmRC addFile(FileList fl, const
 	    int lvl = RPMLOG_ERR;
 	    const char *msg = fl->cur.isDir ? _("Directory not found: %s\n") :
 					      _("File not found: %s\n");
-	    if (fl->cur.attrFlags & RPMFILE_EXCLUDE) {
+	    if (fl->cur.attrFlags & RPMFILE_EXCLUDE || fl->cur.attrFlags & RPMFILE_OPTIONAL) {
 		lvl = RPMLOG_WARNING;
 		rc = RPMRC_OK;
 	    }
@@ -2335,6 +2336,8 @@ static void processSpecialDir(rpmSpec sp
 		free(newfile);
 	    }
 	    argvFree(globFiles);
+	} else if (fl->cur.attrFlags & RPMFILE_OPTIONAL) {
+	    rpmlog(RPMLOG_WARNING, _("Optional file not found by glob: %s\n"), eorigfile);
 	} else {
 	    rpmlog(RPMLOG_ERR, _("File not found by glob: %s\n"), eorigfile);
 	    fl->processingFailed = 1;
diff -up rpm-4.14.0/lib/rpmfiles.h.omv~ rpm-4.14.0/lib/rpmfiles.h
--- rpm-4.14.0/lib/rpmfiles.h.omv~	2018-03-14 21:55:08.980192257 +0100
+++ rpm-4.14.0/lib/rpmfiles.h	2018-03-14 21:55:33.315562108 +0100
@@ -60,6 +60,7 @@ enum rpmfileAttrs_e {
     RPMFILE_README	= (1 <<  8),	/*!< from %%readme */
     /* bits 9-10 unused */
     RPMFILE_PUBKEY	= (1 << 11),	/*!< from %%pubkey */
+    RPMFILE_OPTIONAL	= (1 << 12),	/*!< from %%optional */
 };
 
 typedef rpmFlags rpmfileAttrs;
diff -up rpm-4.14.0/lib/rpmlib.h.omv~ rpm-4.14.0/lib/rpmlib.h
