--- rpm-5.4.10/scripts/perl.prov.perl_abiver~	2012-12-19 01:42:07.219674388 +0100
+++ rpm-5.4.10/scripts/perl.prov	2012-12-19 01:42:13.285755596 +0100
@@ -97,11 +97,15 @@ sub process_file {
   my ($file) = @_;
   
   if ($file =~ m/$Config{archname}\/Config.pm$/) {
-      my $vercmd = "perl -I". File::Basename::dirname($file) . " -MConfig -e 'print \"\$Config{version}\"'";
-      my $v = `$vercmd`;
-      if ($v) {
-        $require{"abi"} = $v;
+    my $vercmd = "perl -I". File::Basename::dirname($file) . " -MConfig -e 'print \"VERSION=\$Config{version}\"'";
+    my $v = `$vercmd 2>&1`;
+    if ($v) {
+      if ($? && $v =~ m!^Perl lib version \((\d*.\d*.\d*)\) doesn't match executable!) {
+	$require{"abi"} = $1;
+      } elsif($v =~ m!^VERSION=(\d*.\d*.\d*)!) {
+	$require{"abi"} = $1;
       }
+    }
   }
 
   if (!open(FILE, $file)) {
