--- rpm-5.4.10/scripts/gstreamer.sh.gsboth~	2011-10-12 18:51:52.000000000 +0200
+++ rpm-5.4.10/scripts/gstreamer.sh	2013-04-06 10:34:56.452965672 +0200
@@ -52,15 +52,20 @@ while [ "$#" -ne 0 ]; do
     shift
 done
 
-if [ -z "$gst_inspect" ]; then
-    exit 1
-fi
+#if [ -z "$gst_inspect" ]; then
+#    exit 1
+#fi
 
 if [ $provides -eq 1 ]; then
     for so in $solist ; do
 	getmark $so
 	libdir $so
-	LD_LIBRARY_PATH=$buildlibdir $gst_inspect --print-plugin-auto-install-info --rpm $so 2> /dev/null | while read line ; do
+	LD_LIBRARY_PATH=$buildlibdir gst-inspect-0.10 --print-plugin-auto-install-info --rpm $so 2> /dev/null | while read line ; do
+	    echo -n "$line";
+	    echo -n "$lib64"
+	    echo
+	done
+	LD_LIBRARY_PATH=$buildlibdir gst-inspect-1.0 --print-plugin-auto-install-info --rpm $so 2> /dev/null | while read line ; do
 	    echo -n "$line";
 	    echo -n "$lib64"
 	    echo
