--- rpm-5.4.10/build/build.c.doc_post~	2011-01-01 15:39:54.000000000 +0100
+++ rpm-5.4.10/build/build.c	2013-02-25 01:22:28.407418823 +0100
@@ -181,9 +181,9 @@ rpmRC doScript(Spec spec, int what, cons
 	break;
     case RPMBUILD_STRINGBUF:
     default:
-	mTemplate = "%{___build_template}";
-	mPost = "%{___build_post}";
-	mCmd = "%{___build_cmd}";
+	mTemplate = "%{__spec_stringbuf_template}";
+	mPost = "%{__spec_stringbuf_post}";
+	mCmd = "%{__spec_stringbuf_cmd}";
 	break;
     }
 
--- rpm-5.4.10/macros/macros.rpmbuild.in.doc_post~	2013-02-25 01:22:28.400418746 +0100
+++ rpm-5.4.10/macros/macros.rpmbuild.in	2013-02-25 01:22:28.410418856 +0100
@@ -411,6 +411,22 @@ echo "Patch #%{__patch_number} (%{basena
 #%{__spec_install_post}\
 #%{nil}
 
+%__spec_stringbuf_shell	%{___build_shell}
+%__spec_stringbuf_args	%{___build_args}
+%__spec_stringbuf_cmd	%{___build_cmd}
+%__spec_stringbuf_pre	%{___build_pre}
+%__spec_stringbuf_body	%{___build_body}
+%__spec_stringbuf_post\
+%{?__os_stringbuf_post}\
+%{nil}
+%__spec_stringbuf_template	#!%{__spec_stringbuf_shell}\
+%{__spec_stringbuf_pre}\
+%{nil}
+
+#%{__spec_stringbuf_body}\
+#%{__spec_stringbuf_post}\
+#%{nil}
+
 %__spec_check_shell	%{___build_shell}
 %__spec_check_args	%{___build_args}
 %__spec_check_cmd	%{___build_cmd}
--- rpm-5.4.10/macros/mandriva.in.doc_post~	2013-02-25 01:22:28.395418692 +0100
+++ rpm-5.4.10/macros/mandriva.in	2013-02-25 01:22:45.531606854 +0100
@@ -332,6 +332,10 @@ GCONF_CONFIG_SOURCE=`%{_gconftool_bin} -
     %{?__spec_helper_post}%{?!__spec_helper_post:/usr/share/spec-helper/spec-helper} \
 %{nil}
 
+%__os_stringbuf_post    \
+    %{?__spec_helper_stringbuf_post} \
+%{nil}
+
 #==============================================================================
 # ---- specfile macros.
 #	Macro(s) here can be used reliably for reproducible builds.
