--- rpm-5.4.14/scripts/find-debuginfo.sh.exitifnone~	2014-03-12 20:09:51.849500748 +0100
+++ rpm-5.4.14/scripts/find-debuginfo.sh	2014-03-12 20:09:59.739500835 +0100
@@ -252,6 +252,10 @@ while read nlinks inum f; do
   fi
 done || exit
 
+if [[ ! -s "$SOURCEFILE" && ! -d "${debugdir}" ]]; then
+  exit 0
+fi
+
 # For each symlink whose target has a .debug file,
 # make a .debug symlink to that file.
 find "$RPM_BUILD_ROOT" ! -path "${debugdir}/*" -type l -print | LANG=C sort |
