--- rpm-5.4.10/macros/macros.in.omv~	2014-06-24 21:34:51.582495024 +0200
+++ rpm-5.4.10/macros/macros.in	2014-06-24 21:35:47.012494042 +0200
@@ -926,6 +926,7 @@ $_arbitrary_tags_tests	Foo:Bar
 %_mandir		%{_datarootdir}/man
 %_localedir		%{_datarootdir}/locale
 %_lispdir		%{_datarootdir}/emacs/site-lisp
+%_rundir		/run
 %_initddir              %{_sysconfdir}/rc.d/init.d
 # This is a deprecated misspelling, present for backwards compatibility.
 %_initrddir             %{_initddir}
