--- rpm-5.4.10/macros/mandriva.in	2013-08-13 09:06:20.183532957 +0000
+++ rpm-5.4.10/macros/mandriva.in.disablestatic	2013-08-13 09:10:28.969370710 +0000
@@ -364,6 +364,7 @@
   [ -f $CONFIGURE_TOP/configure.in -o -f $CONFIGURE_TOP/configure.ac ] && \
   CONFIGURE_XPATH="--x-includes=%{_prefix}/include --x-libraries=%{_prefix}/%{_lib}" \
   $CONFIGURE_TOP/configure $CROSSCOMPILE \\\
+	--disable-static \\\
   	--disable-silent-rules \\\
 	--disable-dependency-tracking \\\
 	--disable-rpath \\\
@@ -389,6 +390,7 @@
   [ -f $CONFIGURE_TOP/configure.in -o -f $CONFIGURE_TOP/configure.ac ] && \
   CONFIGURE_XPATH="--x-includes=%{_prefix}/include --x-libraries=%{_prefix}/%{_lib}" \
   $CONFIGURE_TOP/configure $CROSSCOMPILE \\\
+	--disable-static \\\
   	--disable-silent-rules \\\
 	--disable-dependency-tracking \\\
 	--disable-rpath \\\
