--- rpm-5.4.10/rpmdb/header.c.ui64p~	2013-03-18 12:43:09.043950947 +0100
+++ rpm-5.4.10/rpmdb/header.c	2013-03-18 12:44:35.059800829 +0100
@@ -398,7 +398,7 @@ static unsigned char * tagSwab(/*@out@*/
     switch (he->t) {
     case RPM_UINT64_TYPE:
     {	rpmuint32_t * tt = (rpmuint32_t *)t;
-assert(nb == (he->c * sizeof(*tt)));
+assert(nb == (he->c * sizeof(rpmuint64_t)));
 	for (i = 0; i < he->c; i++) {
 	    rpmuint32_t j = 2 * i;
 	    rpmuint32_t b = (rpmuint32_t) htonl(he->p.ui32p[j]);
