--- rpm-5.4.15/Makefile.am.nocmake~	2015-07-11 15:29:22.564244403 +0200
+++ rpm-5.4.15/Makefile.am	2015-07-11 15:29:30.228224006 +0200
@@ -187,7 +187,7 @@ pkgbin_SCRIPTS = install-sh mkinstalldir
 
 pkgcfgdir =	$(pkgdatadir)/macros.d
 pkgcfg_DATA = \
-	macros/cmake macros/cmakedeps macros/haskell macros/gstreamer macros/java macros/kernel \
+	macros/cmakedeps macros/haskell macros/gstreamer macros/java macros/kernel \
 	macros/libtool macros/mandriva macros/mono macros/perl macros/pkgconfig \
 	macros/php macros/python macros/ruby macros/selinux macros/tcl
 
