diff -urN rpm-5.4.10/scripts/gem_helper.rb rpm-5.4.10-patched/scripts/gem_helper.rb
--- rpm-5.4.10/scripts/gem_helper.rb	2014-08-13 03:32:31.448925946 +1100
+++ rpm-5.4.10-patched/scripts/gem_helper.rb	2014-08-13 03:35:00.000000000 +1100
@@ -97,21 +97,20 @@
     require 'rubygems/ext'
     module Gem::Ext
       class Builder
-	def self.make(dest_path, results)
+	def build_extensions
 	  make_program = ENV['make']
 	  unless make_program then
 	    make_program = (/mswin/ =~ RUBY_PLATFORM) ? 'nmake' : 'make'
 	  end
 	  cmd = make_program
 	  if ENV['GEM_MODE'] == "build"
-  	    cmd += " %s" % ENV['jobs']
+	    cmd += " %s" % ENV['jobs']
+	    system("ruby %s" % @spec.extensions)
 	  elsif ENV['GEM_MODE'] == "install"
 	    cmd += " DESTDIR='%s' install" % ENV['DESTDIR']
 	  end
-	  results << cmd
-	  results << `#{cmd} #{redirector}`
-
-	  raise Gem::ExtensionBuildError, "make failed:\n\n#{results}" unless
+	  system(cmd)
+	  raise Gem::ExtensionBuildError, "Make failed!\n" unless
 	  $?.success?
 	end
       end
@@ -123,6 +122,9 @@
       	def initialize(spec, options={})
 	  @gem_dir = Dir.pwd
       	  @spec = spec
+	  @options = options
+
+	  process_options
 	end
       end
       class ConfigFile
