--- rpm-5.4.14/lua/linit.c.lua_posix~	2013-06-29 23:03:22.000000000 +0200
+++ rpm-5.4.14/lua/linit.c	2014-03-16 05:36:28.325628126 +0100
@@ -20,6 +20,8 @@
 
 #include "lualib.h"
 #include "lauxlib.h"
+#include "lposix.h"
+#include "lrexlib.h"
 
 
 /*
@@ -37,6 +39,9 @@ static const luaL_Reg loadedlibs[] = {
   {LUA_BITLIBNAME, luaopen_bit32},
   {LUA_MATHLIBNAME, luaopen_math},
   {LUA_DBLIBNAME, luaopen_debug},
+  {"rex_posix", luaopen_rex_posix},
+  {"rex_pcre", luaopen_rex_pcre},
+  {"posix", luaopen_posix},
   {NULL, NULL}
 };
 
