--- rpm-5.4.14/rpmio/rpmsquirrel.c.omv~	2014-07-27 12:48:04.515137643 +0200
+++ rpm-5.4.14/rpmio/rpmsquirrel.c	2014-07-27 12:48:39.702060923 +0200
@@ -161,7 +161,7 @@ rpmsquirrel rpmsquirrelNew(char ** av, u
 
     squirrel->I = v;
     sq_setforeignptr(v, squirrel);
-#if defined(SQUIRREL_VERSION_NUMBER)
+#if defined(SQUIRREL_VERSION_NUMBER) && SQUIRREL_VERSION_NUMBER >= 300
     sq_setprintfunc(v, rpmsquirrelPrint, rpmsquirrelStderr);
 #else
     sq_setprintfunc(v, rpmsquirrelPrint);
