--- rpm-5.4.10/macros/mandriva.in.libpackage~	2013-11-21 21:27:22.780323257 +0100
+++ rpm-5.4.10/macros/mandriva.in	2013-11-21 21:28:01.494839258 +0100
@@ -496,4 +496,25 @@ Internationalization and locale data for
 [ %{_with check} -eq 1 ] || exit 0 \
 %{nil}
 
+%libpackage()\
+%%package -n %{expand:%%mklibname %{1} %{2} %{?3:%{3}}}\
+Summary: The %{1} library, a part of %{name}\
+Group: System/Libraries\
+%%description -n %{expand:%%mklibname %{1} %{2} %{?3:%{3}}}\
+The %{1} library, a part of %{name}\
+%%files -n %{expand:%%mklibname %{1} %{2} %{?3:%{3}}}\
+%{_libdir}/lib%{1}%{?3:-%{2}}.so.%{?3:%{3}}%{?!3:%{2}}*\
+%{nil}
+
+%dependinglibpackage()\
+%%package -n %{expand:%%mklibname %{1} %{2} %{?3:%{3}}}\
+Summary: The %{1} library, a part of %{name}\
+Group: System/Libraries\
+Requires: %{name} = %{EVRD}\
+%%description -n %{expand:%%mklibname %{1} %{2} %{?3:%{3}}}\
+The %{1} library, a part of %{name}\
+%%files -n %{expand:%%mklibname %{1} %{2} %{?3:%{3}}}\
+%{_libdir}/lib%{1}%{?3:-%{2}}.so.%{?3:%{3}}%{?!3:%{2}}*\
+%{nil}
+
+### Attempt at creating automatic perl pkgs
+%perlvendorpkg()\
+%{?2:%define ARCH %{2}}%{!?2:%define ARCH arch}\
+%{expand:%(\
+ARCH=arch;\
+perlname=$(echo %{1} | sed 's,-,/,g');\
+manname=$(echo %{1} | sed 's,-,::,g');\
+perllib=$(echo %{1} | awk -F- '{print $NF}');\
+echo "%package -n perl-%{1}\
+Summary:	Perl Module for %{1}\
+Group:		Development/Perl";\
+if [[ %{ARCH} == "noarch" ]]; then\
+	ARCH=lib;\
+	echo "BuildArch: 	noarch"\
+fi;\
+echo "\
+%description -n perl-%{1}\
+Perl module for %{1}.\
+\
+%files -n perl-%{1}\
+%doc README Changes\
+%{perl_vendor${ARCH}}/$perlname.pm\
+%optional %{perl_vendor${ARCH}}/$perlname.pod"\
+dirs=($(echo %{1} | sed 's,-, ,g'));\
+dirnum=$(echo %{1} | awk -F- '{print NF-1}');\
+perldir=$(echo %{1} | awk -F- '{print $1}');\
+if [[ $dirnum -gt 0 ]]; then\
+	count=0;\
+	echo "%dir %{perl_vendor${ARCH}}/$perldir";\
+	if [[ %{ARCH} == "arch" ]]; then\
+		echo "%dir %{perl_vendor${ARCH}}/auto/$perldir";\
+	fi;\
+	while [[ $count -lt $dirnum ]]; do\
+		perldir="$perldir/${dirs[$count+1]}";\
+		if [[ $count+1 -eq $dirnum ]]; then\
+			echo "%optional %dir %{perl_vendor${ARCH}}/$perldir";\
+		else\
+			echo "%dir %{perl_vendor${ARCH}}/$perldir";\
+		fi;\
+		if [[ %{ARCH} == "arch" ]]; then\
+			echo "%dir %{perl_vendor${ARCH}}/auto/$perldir";\
+		fi;\
+		count=$[$count+1];\
+	done;\
+fi;\
+echo "%optional %{perl_vendor${ARCH}}/$perlname/*";\
+if [[ %{ARCH} == "arch" ]]; then\
+	echo "%{perl_vendor${ARCH}}/auto/$perlname/$perllib.so";\
+	echo "%optional %{perl_vendor${ARCH}}/auto/$perlname/*.itx";\
+fi;\
+echo "%{_mandir}/man3/$manname.3pm.*\
+%optional %{_mandir}/man3/$manname::*.3pm.*"\
+)}\
+%{nil}
+
 %{load:/etc/rpm/macros.d/*.macros}
