diff -Nurp rpm-4.18.0.orig/scripts/brp-strip rpm-4.18.0/scripts/brp-strip
--- rpm-4.18.0.orig/scripts/brp-strip	2022-04-07 14:13:19.072518377 +0300
+++ rpm-4.18.0/scripts/brp-strip	2022-11-29 15:01:19.705111799 +0200
@@ -13,5 +13,5 @@ Darwin*) exit 0 ;;
 esac
 
 # Strip ELF binaries
-find "$RPM_BUILD_ROOT" -type f \! -regex "${RPM_BUILD_ROOT}/*usr/lib/debug.*" \! -name "*.go" -print0 | \
+find "$RPM_BUILD_ROOT" -type f \! -regex "${RPM_BUILD_ROOT}/*usr/lib/debug.*" \! -regex "${RPM_BUILD_ROOT}/lib/firmware.*" \! -name "*.go" -print0 | \
     xargs -0 -r -P$NCPUS -n32 sh -c "file \"\$@\" | sed -n -e 's/^\(.*\):[ 	]*ELF.*, not stripped.*/\1/p' | grep -v 'no machine' | xargs -I\{\} $STRIP -g \{\}" ARG0
