--- rpm-5.4.7/rpmdb/db3.c.db_rdonly~	2012-03-24 04:16:30.305949579 +0100
+++ rpm-5.4.7/rpmdb/db3.c	2012-03-24 04:16:39.649962411 +0100
@@ -2700,7 +2700,7 @@ assert(Pdbi != NULL);
 	    if (oflags & (DB_CREATE|DB_TRUNCATE)) _flags |= DB_CREATE;
 	    xx = db3associate(Pdbi, dbi, _callback, _flags);
 	}
-	if (dbi->dbi_seq_id) {
+	if (dbi->dbi_seq_id && !(oflags & DB_RDONLY)) {
 	    char * end = NULL;
 	    uint32_t u = (uint32_t) strtoll(dbi->dbi_seq_id, &end, 0);
 
