--- rpm-5.4.14/lib/rpminstall.c.closefds~	2014-03-15 05:26:49.117966453 +0100
+++ rpm-5.4.14/lib/rpminstall.c	2014-03-16 05:06:41.537055894 +0100
@@ -168,8 +168,15 @@ void * rpmShowProgress(/*@null@*/ const
 		xx = Fclose(fd);
 		fd = NULL;
 	    }
-	} else
+	} else {
+ 	    long oldfd;
 	    fd = fdLink(fd, "persist (showProgress)");
+ 	    oldfd=Fcntl(fd, F_GETFD, 0);
+ 	    if(oldfd >= 0) {
+ 		oldfd |= FD_CLOEXEC; /* scripts shouldn't inherit rpm file descriptor */
+ 		Fcntl(fd, F_SETFD, (void*)oldfd);
+ 	    }
+ 	}
 
 #if defined(POSIX_FADV_WILLNEED)
 	(void) Fadvise(fd, 0, 0, POSIX_FADV_WILLNEED);
