diff -up rpm-4.16.0-beta1/macros.in.omv~ rpm-4.16.0-beta1/macros.in
--- rpm-4.16.0-beta1/macros.in.omv~	2020-06-20 18:38:43.584043149 +0200
+++ rpm-4.16.0-beta1/macros.in	2020-06-20 18:38:56.964201029 +0200
@@ -620,7 +620,7 @@ package or when debugging this package.\
 # sqlite Sqlite database
 # dummy dummy backend (no actual functionality)
 #
-%_db_backend	      bdb
+%_db_backend	      sqlite
 
 #
 #	Macros used to configure Berkley db parameters.
