--- rpm-5.4.12/lib/rpmfc.c.sizeof~	2013-07-30 16:22:53.291676301 +0200
+++ rpm-5.4.12/lib/rpmfc.c	2013-07-30 16:24:12.978800861 +0200
@@ -924,7 +924,7 @@ static int rpmfcSCRIPT(rpmfc fc)
             defaultdocdir = xstrdup("/usr/share/doc");
 	}
 
-	if (strncmp(fn, defaultdocdir, sizeof(defaultdocdir)-1)) {
+	if (strncmp(fn, defaultdocdir, strlen(defaultdocdir))) {
 	    if (fc->fcolor->vals[fc->ix] & RPMFC_MODULE)
 		xx = rpmfcHelper(fc, 'P', "perl");
 	    if (is_executable || (fc->fcolor->vals[fc->ix] & RPMFC_MODULE))
