--- rpm-5.4.10/macros/mandriva.in.dlopen_req~	2014-07-27 10:19:15.499266230 +0200
+++ rpm-5.4.10/macros/mandriva.in	2014-07-27 10:22:04.073754462 +0200
@@ -499,4 +499,15 @@ The %{1} library, a part of %{name}\
 %{_libdir}/lib%{1}%{?3:-%{2}}.so.%{?3:%{3}}%{?!3:%{2}}*\
 %{nil}
 
+# get dependencies from library that foo.so points to, very convenient if
+# one wants to ie. generate requires/suggests against libraries that are
+# dlopen()'ed.
+# First argument specifies library name without .so extension and lib prefix,
+# while the optional second argument can be used to specify different library
+# path than %%{_libdir}
+%dlopen_req() %{shrink:\
+%([ -e %{?!2:%{_libdir}}%{?2}/lib%{1}.so ] && \
+rpm -qf --fileprovide $(readlink -f %{?!2:%{_libdir}}%{?2}/lib%{1}.so) 2>/dev/null | \
+grep $(readlink -f %{?!2:%{_libdir}}%{?2}/lib%{1}.so) | cut -f2 || echo %{name})}
+
 %{load:/etc/rpm/macros.d/*.macros}
