diff -up rpm-4.18.90/lib/rpmrc.c.6~ rpm-4.18.90/lib/rpmrc.c
--- rpm-4.18.90/lib/rpmrc.c.6~	2023-05-03 16:27:04.481294761 +0200
+++ rpm-4.18.90/lib/rpmrc.c	2023-05-03 16:28:25.493156602 +0200
@@ -469,6 +469,7 @@ static void setDefaults(void)
 				confdir, "/platform/%{_target}/macros", ":",
 				confdir, "/fileattrs/*.attr", ":",
   				confdir, "/" RPM_VENDOR "/macros", ":",
+				SYSCONFDIR "/rpm/macros.d/*.macros", ":",
 				SYSCONFDIR "/rpm/macros.*", ":",
 				SYSCONFDIR "/rpm/macros", ":",
 				SYSCONFDIR "/rpm/%{_target}/macros", ":",
