--- rpm-5.4.10/cpu-os-macros/armv6j-linux/macros.omv~	2014-06-03 15:07:08.800913825 +0200
+++ rpm-5.4.10/cpu-os-macros/armv6j-linux/macros	2014-06-03 15:08:11.734020955 +0200
@@ -5,7 +5,7 @@
 #
 %_arch			arm
 %_os			linux
-%_gnu			-gnueabi
+%_gnu			-gnueabihf
 %optflags		 %{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fPIC -g -march=armv6j -mfpu=vfp -mfloat-abi=hard}%{!?__common_cflags_with_ssp:-O2}
 
 #==============================================================================
--- rpm-5.4.10/cpu-os-macros/armv7hl-linux/macros.omv~	2014-06-03 15:06:43.829266714 +0200
+++ rpm-5.4.10/cpu-os-macros/armv7hl-linux/macros	2014-06-03 15:06:52.088150091 +0200
@@ -5,7 +5,7 @@
 #
 %_arch			arm
 %_os			linux
-%_gnu			-gnueabi
+%_gnu			-gnueabihf
 %optflags		 %{?__common_cflags_with_ssp:%{__common_cflags_with_ssp} -fPIC -g -march=armv7-a -mfloat-abi=hard -mfpu=vfpv3-d16}%{!?__common_cflags_with_ssp:-O2}
 
 #==============================================================================
