--- rpm-5.4.10/macros/mandriva.in.omv~	2014-01-02 15:51:20.777668021 +0100
+++ rpm-5.4.10/macros/mandriva.in	2014-01-02 15:51:55.871001815 +0100
@@ -176,10 +176,6 @@ export CFLAGS="%{optflags} -fPIE"; expor
 #	signing and verification.
 #
 
-# We disable the gpg key automatic querying 
-%_hkp_keyserver_query   %nil
-
-
 #	Verify digest/signature flags for various rpm modes:
 #	0x30300 (_RPMVSF_NODIGESTS)    --nohdrchk      if set, don't check digest(s)
 #	0xc0c00 (_RPMVSF_NOSIGNATURES) --nosignature   if set, don't check signature(s)
