--- rpm-5.4.10/macros/mandriva.in.trim_ldflags~	2013-01-23 02:46:52.709850951 +0100
+++ rpm-5.4.10/macros/mandriva.in	2013-01-23 02:50:31.925006795 +0100
@@ -342,7 +342,7 @@ GCONF_CONFIG_SOURCE=`%{_gconftool_bin} -
 
 %__libtoolize_configure %{?__libtoolize:(cd $CONFIGURE_TOP; [ ! -f configure.in -a ! -f configure.ac ] || %{__libtoolize} --copy --force)}
 
-%ldflags %{?!_disable_ld_as_needed: -Wl,--as-needed}%{?!_disable_ld_no_undefined: -Wl,--no-undefined}%{?!_disable_ld_relro: -Wl,-z,relro}%{?!_disable_ld_O1: -Wl,-O1}%{?!_disable_ld_build_id: -Wl,--build-id}%{?!_disable_ld_enable_new_dtags: -Wl,--enable-new-dtags}%{?!_disable_hash_style_gnu: -Wl,--hash-style=gnu}%{?_hardened_flags: %_hardened_flags}
+%ldflags %{?!_disable_ld_no_undefined: -Wl,--no-undefined}%{?_hardened_flags}
 
 %setup_compile_flags \
   CFLAGS="${CFLAGS:-%optflags}" ; export CFLAGS ; \
