--- rpm-5.4.8/configure.ac.urpm~	2012-04-25 19:27:07.929167065 +0200
+++ rpm-5.4.8/configure.ac	2012-04-25 19:29:04.996867674 +0200
@@ -1198,8 +1198,8 @@ AC_ARG_WITH(perl-urpm, AS_HELP_STRING([-
 	    AC_SUBST(PERL_URPM_INSTALLDIRS)
         WITH_PERL_URPM_SUBDIR=perl-URPM
         WITH_PERL_URPM_SUBPACKAGE=1
-        AC_CONFIG_FILES([perl-URPM/Makefile.PL])
-        AC_CONFIG_FILES([perl-URPM/Makefile])
+        AC_CONFIG_FILES([scripts/URPM-Makefile.PL])
+        AC_CONFIG_FILES([scripts/URPM-Makefile])
     fi
 ])
 AC_SUBST(WITH_PERL_URPM_SUBDIR)
