--- rpm-5.4.10/macros/mandriva.in.legacy_macros~	2013-01-23 02:59:18.692504551 +0100
+++ rpm-5.4.10/macros/mandriva.in	2013-01-23 03:11:35.342830799 +0100
@@ -34,8 +34,6 @@
 
 # This will die as soon as remaining usage has been phased out...
 %mkrel(c:)		%{-c: 0.%{-c*}.}%{1}%{?subrel:.%subrel}
-%manbo_mkrel()		%mkrel
-%multiarch()		%{1}
 
 # This will enable the use of distepoch and disttag in stead of polluting
 # %release with such.
@@ -157,14 +155,14 @@ end\
 # cf http://wiki.mandriva.com/en/Development/Packaging/Problems#format_not_a_string_literal_and_no_format_arguments
 %Werror_cflags -Wformat -Werror=format-security
 
-%_ssp_cflags -fstack-protector --param=ssp-buffer-size=4%{?_serverbuild_flags: %_serverbuild_flags}
+%_ssp_cflags -fstack-protector --param=ssp-buffer-size=4 %{?_serverbuild_flags}
 %__common_cflags -O2 %{debugcflags} -pipe %{Werror_cflags} %{?_fortify_cflags}
 %__common_cflags_with_ssp %{__common_cflags} %{?_ssp_cflags}
 
 # Servers opt flags.
 # Also set the env variables for backward compatibility (#32050).
 %serverbuild %define _serverbuild_flags -fstack-protector-all \
-export CFLAGS="%optflags -fPIE"; export CXXFLAGS="%optflags -fPIE"; export RPM_OPT_FLAGS="%optflags -fPIE" \
+export CFLAGS="%{optflags} -fPIE"; export CXXFLAGS="%{optflags} -fPIE"; export RPM_OPT_FLAGS="%{optflags} -fPIE" \
 %{nil}
 
 # Hardened Servers opt flags.
@@ -179,7 +177,7 @@ export CFLAGS="%optflags -fPIE"; export
 #
 
 # We disable the gpg key automatic querying 
-%_hkp_keyserver_query   %nil
+%_hkp_keyserver_query   %{nil}
 
 
 #	Verify digest/signature flags for various rpm modes:
@@ -207,10 +205,10 @@ export CFLAGS="%optflags -fPIE"; export
 
 # when %_with_git_repository is set, these macros modify the behaviour of "%prep" step:
 %_after_setup %{?_with_git_repository:GIT_URL="%{?git_url}" GIT_REPOSITORY_CACHE=%{?git_repository_cache} @USRLIBRPM@/git-repository--after-tarball}
-%_patch %{?_with_git_repository:PKG_NAME=%{name} @USRLIBRPM@/git-repository--apply-patch}%{?!_with_git_repository:%__patch -U}
+%_patch %{?_with_git_repository:PKG_NAME=%{name} @USRLIBRPM@/git-repository--apply-patch}%{?!_with_git_repository:%__patch --fuzz=%{_default_patch_fuzz} %{_default_patch_flags}}
 
 # used by "git-repository--after-tarball":
-%git_repository_cache %_topdir/%{name}.git
+%git_repository_cache %{_topdir}/%{name}.git
 
 
 # Various programs used in rpm scripts
@@ -220,9 +218,6 @@ export CFLAGS="%optflags -fPIE"; export
 %_gconftool_bin %{_bindir}/gconftool-2
 %_scrollkeeper_bin %{_bindir}/scrollkeeper-update
 
-%__service          /sbin/service
-%__chkconfig        /sbin/chkconfig
-
 #==============================================================================
 # ---- Required rpmrc macros.
 #	Macros that used to be initialized as a side effect of rpmrc parsing.
@@ -244,8 +239,7 @@ Group: %{group}\
 %_gamesbindir   %{_prefix}/%{_gamesdir}
 %_gamesdatadir  %{_datadir}/%{_gamesdir}
 
-# Menu directories
-%_menudir %{_prefix}/lib/menu
+# Icon directories
 %_iconsdir %{_datadir}/icons
 %_miconsdir %{_datadir}/icons/mini
 %_liconsdir %{_datadir}/icons/large
@@ -254,8 +248,6 @@ Group: %{group}\
 
 %_webconfdir %{_sysconfdir}/httpd/conf
 %_webappconfdir %_webconfdir/webapps.d
-# compatibility
-%webappconfdir %_webappconfdir
 
 #==============================================================================
 # ---- Build configuration macros.
@@ -274,9 +266,6 @@ Group: %{group}\
 # Use internal dependency generator rather than external helpers?
 %_use_internal_dependency_generator	1
 
-# Update Menu
-%_update_menus_bin %{_bindir}/update-menus
-
 # Update Window Managers session.
 %_fndsession_bin %{_sbindir}/fndSession
 %make_session %{nil}
