diff -up rpm-4.18.90/scripts/brp-strip.13~ rpm-4.18.90/scripts/brp-strip
--- rpm-4.18.90/scripts/brp-strip.13~	2023-05-03 16:30:19.104363329 +0200
+++ rpm-4.18.90/scripts/brp-strip	2023-05-03 16:31:46.666189595 +0200
@@ -33,6 +33,8 @@ strip_elf_binaries()
 
   find "$RPM_BUILD_ROOT" -type f \
     ! -regex "${RPM_BUILD_ROOT}/*usr/lib/debug.*" \
+    ! -regex "${RPM_BUILD_ROOT}/lib/firmware.*" \
+    ! -regex "${RPM_BUILD_ROOT}/usr/lib/firmware.*" \
     ! -name "*.go" -links "${nlinks}" -print0 | \
     xargs -0 -r -P${nprocs} -n${MAX_ARGS} sh -c "file \"\$@\" | \
     sed -n -e 's/^\(.*\):[ 	]*ELF.*, not stripped.*/\1/p' | \
