diff -Naur rpm-5.4.17/macros/macros.in rpm-5.4.17.tpg/macros/macros.in
--- rpm-5.4.17/macros/macros.in	2016-05-08 19:04:41.000000000 +0000
+++ rpm-5.4.17.tpg/macros/macros.in	2017-01-29 08:13:38.191995004 +0000
@@ -1059,55 +1059,55 @@
 
 #------------------------------------------------------------------------
 # cmake(...) configuration
-#%%{load:%{_usrlibrpm}/macros.d/cmake}
+%{load:%{_usrlibrpm}/macros.d/cmake}
 
 #------------------------------------------------------------------------
 # gstreamer(...) configuration.
-#%%{load:%{_usrlibrpm}/macros.d/gstreamer}
+%{load:%{_usrlibrpm}/macros.d/gstreamer}
 
 #------------------------------------------------------------------------
 # haskell(...) configuration.
-#%%{load:%{_usrlibrpm}/macros.d/haskell}
+%{load:%{_usrlibrpm}/macros.d/haskell}
 
 #------------------------------------------------------------------------
 # perl(...) configuration
-#%%{load:%{_usrlibrpm}/macros.d/perl}
+%{load:%{_usrlibrpm}/macros.d/perl}
 
 #------------------------------------------------------------------------
 # python(...) configuration.
-#%%{load:%{_usrlibrpm}/macros.d/python}
+%{load:%{_usrlibrpm}/macros.d/python}
 
 #------------------------------------------------------------------------
 # php(...) configuration.
-#%%{load:%{_usrlibrpm}/macros.d/php}
+%{load:%{_usrlibrpm}/macros.d/php}
 
 #------------------------------------------------------------------------
 # java(...) configuration.
-#%%{load:%{_usrlibrpm}/macros.d/java}
+%{load:%{_usrlibrpm}/macros.d/java}
 
 #------------------------------------------------------------------------
 # libtool(...) configuration.
-#%%{load:%{_usrlibrpm}/macros.d/libtool}
+%{load:%{_usrlibrpm}/macros.d/libtool}
 
 #------------------------------------------------------------------------
 # pkgconfig(...) configuration.
-#%%{load:%{_usrlibrpm}/macros.d/pkgconfig}
+%{load:%{_usrlibrpm}/macros.d/pkgconfig}
 
 #------------------------------------------------------------------------
 # mono(...) configuration.
-#%%{load:%{_usrlibrpm}/macros.d/mono}
+%{load:%{_usrlibrpm}/macros.d/mono}
 
 #------------------------------------------------------------------------
 # ruby(...) configuration.
-#%%{load:%{_usrlibrpm}/macros.d/ruby}
+%{load:%{_usrlibrpm}/macros.d/ruby}
 
 #------------------------------------------------------------------------
 # tcl(...) configuration.
-#%%{load:%{_usrlibrpm}/macros.d/tcl}
+%{load:%{_usrlibrpm}/macros.d/tcl}
 
 #------------------------------------------------------------------------
 # typelib(...) configuration.
-#%%{load:%{_usrlibrpm}/macros.d/typelib}
+%{load:%{_usrlibrpm}/macros.d/typelib}
 
 # \endverbatim
 #*/
