diff -up rpm-4.18.0-alpha2/tools/rpmbuild.cc.3~ rpm-4.18.0-alpha2/tools/rpmbuild.cc
--- rpm-4.18.0-alpha2/tools/rpmbuild.cc.3~	2022-05-04 11:59:13.454484278 +0200
+++ rpm-4.18.0-alpha2/tools/rpmbuild.cc	2022-06-24 16:18:06.665014523 +0200
@@ -653,12 +653,12 @@ int main(int argc, char *argv[])
     case 'i':
 	ba->buildAmount |= RPMBUILD_INSTALL;
 	ba->buildAmount |= RPMBUILD_CHECK;
-	if ((buildChar == 'i') && shortCircuit)
+	if (/*(buildChar == 'i') && */ shortCircuit)
 	    break;
 	/* fallthrough */
     case 'c':
 	ba->buildAmount |= RPMBUILD_BUILD;
-	if ((buildChar == 'c') && shortCircuit)
+	if (/*(buildChar == 'c') &&*/ shortCircuit)
 	    break;
 	/* fallthrough */
     case 'f':
