--- rpm-5.4.15/macros/mandriva.in.EVRD~	2014-12-11 19:50:30.122998720 +0100
+++ rpm-5.4.15/macros/mandriva.in	2014-12-11 20:11:29.869137214 +0100
@@ -46,7 +46,7 @@
 
 # Dynamic EVRD tags macro for use with versioned provides/requires in place of
 # '%{version}-%{release}', ie. to automatically add distepoch or epoch if present
-%EVRD			%{?epoch:%{epoch}:}%{?version:%{version}}%{?release:-%{release}}%{?distepoch::%{distepoch}}
+%EVRD			%{?epoch:%{epoch}:}%{?version:%{version}}%{?release:-%{release}%{?distepoch::%{distepoch}}} 
 
 %rename() \
 Obsoletes:	%{1} < %{?2}%{!?2:%{EVRD}} \
