--- rpm-5.4.13/rpmio/rpmpython.c.rpmpythonmod~	2013-09-13 06:12:15.810882169 +0200
+++ rpm-5.4.13/rpmio/rpmpython.c	2013-09-13 06:54:51.588169901 +0200
@@ -66,9 +66,9 @@ static rpmpython rpmpythonGetPool(/*@nul
 /*@unchecked@*/
 #if defined(MODULE_EMBED)
 static const char * _rpmpythonI_init = "\
-import sys\n\
-from cStringIO import StringIO\n\
-sys.stdout = StringIO()\n\
+import sys;\
+from cStringIO import StringIO;\
+sys.stdout = StringIO();\
 ";
 #endif
 
--- rpm-5.4.13/tests/Makefile.am.rpmpythonmod~	2013-09-13 08:15:53.479544111 +0200
+++ rpm-5.4.13/tests/Makefile.am	2013-09-13 08:26:44.664406981 +0200
@@ -324,6 +324,12 @@ check-macros:
 	@-${rpm} -E '%{squirrel:print("--> squirrel: Eat Nuts!")}'
 	@-${rpm} -E '%{perl:print "-->     perl: Artistic Scribbles!"}'
 	@-${rpm} -E '%{python:print "-->   python: Snake Eggs!",}'
+	@-if [ "$(shell ${rpm} -E '%{python:True}')" != "%{python:True}" ]; then \
+	    if [ $(shell ${rpm} -D 'foo %{python:import sys;sys.stdout.write("bar")}' -E 'foo%{foo}') != "foobar" ]; then \
+	      echo python busted; \
+	      exit 1; \
+	    fi \
+	  fi
 	@-${rpm} -E '%{ficl:.( -->     ficl: Das OpenBoot!)}'
 
 check-srpm:
