--- rpm-5.4.15/build/files.c.no_ghost~	2015-02-20 12:30:37.509291705 +0100
+++ rpm-5.4.15/build/files.c	2015-02-20 12:31:57.555145096 +0100
@@ -1931,6 +1931,10 @@ static int addFile(FileList fl, const ch
 	    statp->st_atime = now;
 	    statp->st_mtime = now;
 	    statp->st_ctime = now;
+	} else if(fl->currentFlags & RPMFILE_GHOST) {
+		rpmlog(RPMLOG_DEBUG, _("Ghost file skipped: %s\n"), diskURL);
+		rc = RPMRC_OK;
+		goto addtolist;
 	} else if (Lstat(diskURL, statp)) {
 	    if (fl->currentFlags & RPMFILE_OPTIONAL) {
 		rpmlog(RPMLOG_WARNING, _("Optional file not found: %s\n"), diskURL);
@@ -1955,6 +1959,7 @@ static int addFile(FileList fl, const ch
     fileUid = statp->st_uid;
     fileGid = statp->st_gid;
 
+addtolist:
     if (S_ISDIR(fileMode) && fl->cur_ar.ar_dmodestr) {
 	fileMode &= S_IFMT;
 	fileMode |= fl->cur_ar.ar_dmode;
@@ -1978,7 +1983,7 @@ static int addFile(FileList fl, const ch
 	fileUname = getUname(getuid());
     if (fileGname == NULL)
 	fileGname = getGname(getgid());
-    
+
     /* Add to the file list */
     if (fl->fileListRecsUsed == fl->fileListRecsAlloced) {
 	fl->fileListRecsAlloced += 128;
@@ -2209,7 +2214,7 @@ static rpmRC processBinaryFile(/*@unused
 	    goto exit;
 	}
     }
-    
+
     /* Copy file name or glob pattern removing multiple "/" chars. */
     /*
      * Note: rpmGetPath should guarantee a "canonical" path. That means
@@ -2234,7 +2239,14 @@ static rpmRC processBinaryFile(/*@unused
 	}
 
 	xx = rpmGlob(diskURL, &argc, &argv);
-	if (xx == 0 && argc >= 1) {
+/*    if (fl->currentFlags & RPMFILE_GHOST) {
+	    rpmlog(RPMLOG_DEBUG, _("Skipping ghost file: %s\n"),
+			    diskURL);
+	    rc = RPMRC_OK;
+	    goto exit;
+    }*/
+
+	if (fl->currentFlags & RPMFILE_GHOST || (xx == 0 && argc >= 1)) {
 	    for (i = 0; i < argc; i++) {
 		rc = addFile(fl, argv[i], NULL);
 		argv[i] = _free(argv[i]);
