diff -up rpm-4.17.0/macros.in.omv~ rpm-4.17.0/macros.in
--- rpm-4.17.0/macros.in.omv~	2022-06-18 20:33:24.051343109 +0200
+++ rpm-4.17.0/macros.in	2022-06-18 20:34:01.159572459 +0200
@@ -137,7 +137,7 @@
 %_builddir		%{_topdir}/BUILD
 
 #	The interpreter used for build scriptlets.
-%_buildshell		/bin/sh
+%_buildshell		%{_bindir}/sh
 
 #	The location of the rpm database file(s).
 %_dbpath		%{_var}/lib/rpm
@@ -735,7 +735,7 @@ package or when debugging this package.\
 #	Global defaults used for building scriptlet templates.
 #
 
-%___build_shell		%{?_buildshell:%{_buildshell}}%{!?_buildshell:/bin/sh}
+%___build_shell		%{?_buildshell:%{_buildshell}}%{!?_buildshell:%{_bindir}/sh}
 %___build_args		-e
 %___build_cmd		%{?_sudo:%{_sudo} }%{?_remsh:%{_remsh} %{_remhost} }%{?_remsudo:%{_remsudo} }%{?_remchroot:%{_remchroot} %{_remroot} }%{___build_shell} %{___build_args}
 %___build_pre	\
