--- rpm-5.4.5/lib/rpmds.c.16005~	2012-02-24 14:13:40.677094532 +0100
+++ rpm-5.4.5/lib/rpmds.c	2012-02-24 14:17:17.278202493 +0100
@@ -2913,7 +2913,7 @@ static char * sonameDep(/*@returned@*/ c
 	tmp = stpcpy(tmp, "devel(");
     }
 #if !defined(__alpha__) && !defined(__sun)
-    if (!isElf64) {
+    if (isElf64) {
 	/* XXX: eehhk, would've been nice with consistency, mandriva legacy... :| */
 	if (!devel && s[strlen(s)-1] != ')')
 	(void) stpcpy( stpcpy(tmp, s), "()(64bit)");
