--- rpm-5.4.4/macros/macros.rpmbuild.in.builddir~	2011-12-20 15:36:20.813852241 +0100
+++ rpm-5.4.4/macros/macros.rpmbuild.in	2011-12-20 15:37:43.385806877 +0100
@@ -326,8 +326,9 @@ echo "Patch #%{__patch_number} (%{basena
 %___build_args		-e
 %___build_cmd		%{?_sudo:%{_sudo} }%{?_remsh:%{_remsh} %{_remhost} }%{?_remsudo:%{_remsudo} }%{?_remchroot:%{_remchroot} %{_remroot} }%{___build_shell} %{___build_args}
 %___build_pre	\
+  RPM_BUILD_DIR=\"%{u2p:%{_builddir}}\"\
   RPM_OPT_FLAGS=\"%{optflags}\"\
-  export RPM_OPT_FLAGS\
+  export RPM_BUILD_DIR RPM_OPT_FLAGS\
   %{?buildroot:RPM_BUILD_ROOT=\"%{u2p:%{buildroot}}\"\
   export RPM_BUILD_ROOT}\
   %{?_javaclasspath:CLASSPATH=\"%{_javaclasspath}\"\
--- rpm-5.4.4/scripts/find-debuginfo.sh.builddir~	2011-12-20 15:36:31.962846125 +0100
+++ rpm-5.4.4/scripts/find-debuginfo.sh	2011-12-20 15:36:49.492836503 +0100
@@ -32,7 +32,6 @@ strip_r=false
 # Barf on missing build IDs.
 strict=false
 
-RPM_BUILD_DIR="`pwd`"
 BUILDDIR=.
 out=debugfiles.list
 nout=0
