--- rpm-5.4.14/macros/mandriva.in.dlopen_req~	2014-06-06 20:52:02.212667806 +0200
+++ rpm-5.4.14/macros/mandriva.in	2014-06-14 06:39:04.475853577 +0200
@@ -489,4 +489,14 @@ The %{1} library, a part of %{name}\
 %{_libdir}/lib%{1}.so.%{2}*\
 %{nil}
 
+# get dependencies from library that foo.so points to, very convenient if one wanna
+# ie. generate requires/suggests against libraries that are dlopen()'ed
+# first argument specifies library name without .so extension and lib prefix,
+# while the optional second # argument can be used to specify different library
+# path than %%{_libdir}
+%dlopen_req() %{shrink:\
+%([ -e %{?!2:%{_libdir}}%{?2}/lib%{1}.so ] && \
+rpm -qf --fileprovide $(readlink -f %{?!2:%{_libdir}}%{?2}/lib%{1}.so) 2>/dev/null | \
+grep $(readlink -f %{?!2:%{_libdir}}%{?2}/lib%{1}.so) | cut -f2 || echo %{name})}
+
 %{load:/etc/rpm/macros.d/*.macros}
