--- rpm-5.4.14/lib/psm.c.fixembtypo~	2014-04-22 08:16:03.955591728 +0200
+++ rpm-5.4.14/lib/psm.c	2014-04-22 08:16:06.064493862 +0200
@@ -28,7 +28,7 @@
 #if defined(WITH_LUA) || defined(WITH_AUGEAS) || defined(WITH_FICL) || defined(WITH_GPSEE) || defined(WITH_PERLEMBED) || defined(WITH_PYTHONEMBED) || defined(WITH_RUBYEMBED) || defined(WITH_SEMANAGE) || defined(WITH_SQLITE) || defined(WITH_SQUIRREL) || defined(WITH_TCL)
 #define	_WITH_EMBEDDED
 #else
-#undef _WITH_ENBEDDED
+#undef _WITH_EMBEDDED
 #endif
 
 #include <rpmtag.h>
