--- rpm-5.4.5/lib/rpmfc.c.python_coloring~	2012-03-05 23:13:57.442354701 +0100
+++ rpm-5.4.5/lib/rpmfc.c	2012-03-05 23:13:59.610357114 +0100
@@ -641,6 +641,7 @@ static struct rpmfcTokens_s rpmfcTokens[
   /* XXX "python 2.3 byte-compiled" */
   { " /usr/bin/python",		RPMFC_PYTHON|RPMFC_INCLUDE },
   { "python ",			RPMFC_PYTHON|RPMFC_INCLUDE },
+  { "Python script",		RPMFC_PYTHON|RPMFC_INCLUDE },
 
   { "libtool library ",		RPMFC_LIBTOOL|RPMFC_INCLUDE },
   { "pkgconfig ",		RPMFC_PKGCONFIG|RPMFC_INCLUDE },
