--- rpm-5.4.15/macros/mandriva.in.ltupdate~	2015-07-12 20:06:27.736365674 +0200
+++ rpm-5.4.15/macros/mandriva.in	2015-07-12 20:12:57.189863679 +0200
@@ -358,6 +358,19 @@ GCONF_CONFIG_SOURCE=`%{_gconftool_bin} -
   find . -name config.guess -o -name config.sub | while read i ; do \
            [ -f /usr/share/libtool/config/"$(basename \"$i\")" ] && /bin/rm -f "$i" && /bin/cp -fv /usr/share/libtool/config/"$(basename \"$i\")" "$i" ; \
   done ; \
+  %{!?_disable_rebuild_configure:if [ -e configure.ac -a -e Makefile.am ]; then \
+    find . -name configure.ac |xargs dirname |while read D; do \
+      pushd $D; \
+      if grep -qE '(LT_INIT|LIBTOOL)' configure.ac >/dev/null; then \
+        libtoolize --force ; \
+      fi ; \
+      aclocal $((find . -name "*.m4" 2>/dev/null |grep -vE 'ac(local|include).m4' | xargs dirname) | grep -v '^\.$' |sort |uniq |cut -d/ -f2- |while read R; do [ -e $R/configure.ac ] || echo -n "-I$R "; done) ; \
+      automake -a ; \
+      autoconf ; \
+      popd ; \
+    done ; \
+  fi ;} \
+  %{nil}
 
 %before_configure \
   %config_update \
