--- rpm-5.4.12/misc/librpmmisc.vers.rpmluaprint~	2013-08-08 18:23:58.367842414 +0200
+++ rpm-5.4.12/misc/librpmmisc.vers	2013-08-08 18:20:25.019996866 +0200
@@ -36,6 +36,7 @@ LIBRPMMISC_0
     lua_gettop;
     lua_isnumber;
     lua_isstring;
+    lua_setglobal;
     luaL_argerror;
     luaL_checklstring;
     luaL_error;
--- rpm-5.4.12/rpmio/rpmlua.c.rpmluaprint~	2013-08-08 18:23:06.755942297 +0200
+++ rpm-5.4.12/rpmio/rpmlua.c	2013-08-08 18:23:32.325390087 +0200
@@ -207,8 +207,8 @@ rpmlua rpmluaNew(void)
 #else
     lua_pushglobaltable(L);
 #endif
-    lua_pushliteral(L, "print");
     lua_pushcfunction(L, rpm_print);
+    lua_setglobal(L, "print");
 
 #if defined(LUA_GLOBALSINDEX)
     lua_rawset(L, LUA_GLOBALSINDEX);
