--- rpm-5.4.15/macros/mandriva.in.no_comp_debug~	2015-07-07 04:23:19.194099795 +0200
+++ rpm-5.4.15/macros/mandriva.in	2015-07-07 04:22:37.356117557 +0200
@@ -149,7 +149,7 @@ end\
 # - %{mklibname test 1 -d 0 -s} => libtest1_0-static-devel
 %mklibname(ds)  %{_lib}%{1}%{!-d:%(if echo %{1} |rev |cut -b1 |grep -q '[0-9]'; then echo -n _; fi)}%{?2:%{2}}%{?3:_%{3}}%{-s:-static}%{-d:-devel}
 
-%debugcflags %{?_enable_debug_packages:-Wa,--compress-debug-sections -gdwarf-4 -Wstrict-aliasing=2}
+%debugcflags %{?_enable_debug_packages:-gdwarf-4 -Wstrict-aliasing=2}
 
 # common compilation flags
 
