--- rpm-5.4.10/Makefile.am.cmakedeps~	2014-02-02 23:15:43.491973554 +0100
+++ rpm-5.4.10/Makefile.am	2014-02-02 23:15:56.498640394 +0100
@@ -116,7 +116,7 @@ pkgbin_SCRIPTS = install-sh mkinstalldir
 
 pkgcfgdir =	$(pkgdatadir)/macros.d
 pkgcfg_DATA = \
-	macros/cmake macros/haskell macros/gstreamer macros/java macros/kernel \
+	macros/cmake macros/cmakedeps macros/haskell macros/gstreamer macros/java macros/kernel \
 	macros/libtool macros/mandriva macros/mono macros/perl macros/pkgconfig \
 	macros/php macros/python macros/ruby macros/selinux macros/tcl
 
--- rpm-5.4.10/configure.ac.cmakedeps~	2014-02-02 23:16:14.885307299 +0100
+++ rpm-5.4.10/configure.ac	2014-02-02 23:16:27.701974133 +0100
@@ -2570,7 +2570,7 @@ AC_CONFIG_FILES([ po/Makefile.in
     scripts/Makefile
     rpmdb/DB_CONFIG
     macros/macros macros/macros.rpmbuild
-    macros/cmake macros/gstreamer macros/haskell macros/java macros/kernel macros/libtool
+    macros/cmake macros/cmakedeps macros/gstreamer macros/haskell macros/java macros/kernel macros/libtool
     macros/mandriva macros/suse macros/fedora macros/mono macros/perl macros/pkgconfig macros/php
     macros/python macros/ruby macros/selinux macros/tcl
     doc/Makefile
--- rpm-5.4.10/lib/rpmfc.c.cmakedeps~	2014-02-02 23:17:44.391975140 +0100
+++ rpm-5.4.10/lib/rpmfc.c	2014-02-02 23:21:35.091978169 +0100
@@ -646,6 +646,7 @@ static struct rpmfcTokens_s rpmfcTokens[
 
   { "libtool library ",		RPMFC_LIBTOOL|RPMFC_INCLUDE },
   { "pkgconfig ",		RPMFC_PKGCONFIG|RPMFC_INCLUDE },
+  { "cmake ",			RPMFC_CMAKE|RPMFC_INCLUDE },
 
   { "Bourne ",			RPMFC_BOURNE|RPMFC_INCLUDE },
   { "Bourne-Again ",		RPMFC_BOURNE|RPMFC_INCLUDE },
@@ -946,6 +947,10 @@ static int rpmfcSCRIPT(rpmfc fc)
 #endif
 	    xx = rpmfcHelper(fc, 'R', "pkgconfig");
     } else
+    if (fc->fcolor->vals[fc->ix] & RPMFC_CMAKE) {
+	xx = rpmfcHelper(fc, 'P', "cmake");
+	xx = rpmfcHelper(fc, 'R', "cmake");
+    } else
     if (fc->fcolor->vals[fc->ix] & RPMFC_BOURNE) {
 #ifdef	NOTYET
 	xx = rpmfcHelper(fc, 'P', "executable");
@@ -1140,7 +1145,7 @@ typedef struct rpmfcApplyTbl_s {
 /*@unchecked@*/
 static struct rpmfcApplyTbl_s rpmfcApplyTable[] = {
     { rpmfcELF,		RPMFC_ELF },
-    { rpmfcSCRIPT,	(RPMFC_SCRIPT|RPMFC_FONT|RPMFC_HASKELL|RPMFC_RUBY|RPMFC_PERL|RPMFC_PYTHON|RPMFC_LIBTOOL|RPMFC_PKGCONFIG|RPMFC_BOURNE|RPMFC_JAVA|RPMFC_PHP|RPMFC_MONO|RPMFC_TYPELIB) },
+    { rpmfcSCRIPT,	(RPMFC_SCRIPT|RPMFC_FONT|RPMFC_HASKELL|RPMFC_RUBY|RPMFC_PERL|RPMFC_PYTHON|RPMFC_LIBTOOL|RPMFC_CMAKE|RPMFC_PKGCONFIG|RPMFC_BOURNE|RPMFC_JAVA|RPMFC_PHP|RPMFC_MONO|RPMFC_TYPELIB) },
 #if defined(RPM_VENDOR_MANDRIVA)
     { rpmfcSYMLINK,	RPMFC_SYMLINK },
 #endif
@@ -1448,6 +1453,10 @@ assert(s != NULL && *s == '/');
 	    else if (_suffix(s, ".pc"))
 		ftype = "pkgconfig file";
 
+	    /* XXX all files with extension ".pc" are cmake for now. */
+	    else if (_suffix(s, ".cmake"))
+		ftype = "cmake file";
+
 	    /* XXX all files with extension ".php" are PHP for now. */
 	    else if (_suffix(s, ".php"))
 		ftype = "PHP script text";
--- rpm-5.4.10/lib/rpmfc.h.cmakedeps~	2014-02-02 23:18:28.198642381 +0100
+++ rpm-5.4.10/lib/rpmfc.h	2014-02-02 23:19:36.865309951 +0100
@@ -24,7 +24,7 @@ typedef enum FCOLOR_e {
 #define	RPMFC_ELF	(RPMFC_ELF32|RPMFC_ELF64|RPMFC_ELFMIPSN32)
 	/* (1 << 3) leaks into package headers, reserved */
 
-	/* bit 4 unused */
+    RPMFC_CMAKE			= (1 <<  4),
     RPMFC_TYPELIB		= (1 <<  5),
     RPMFC_HASKELL		= (1 <<  6),
     RPMFC_RUBY			= (1 <<  7),
--- rpm-5.4.10/macros/cmakedeps.in.cmakedeps~	2014-02-02 23:15:10.088639781 +0100
+++ rpm-5.4.10/macros/cmakedeps.in	2014-02-02 23:14:55.738639595 +0100
@@ -0,0 +1,11 @@
+# Pkgconfig specific macro definitions.
+# To make use of these macros insert the following line into your spec file:
+# %{load:%{_usrlibrpm}/macros.d/cmake}
+#
+
+# Path to scripts to autogenerate pkgconfig package dependencies,
+#
+# Note: Used iff _use_internal_dependency_generator is non-zero. The
+# helpers are also used by %{_rpmhome}/rpmdeps {--provides|--requires}.
+%__cmake_provides	%{_rpmhome}/cmakedeps.sh --provides
+%__cmake_requires	%{_rpmhome}/cmakedeps.sh --requires
--- rpm-5.4.10/macros/macros.in.cmakedeps~	2014-02-02 23:16:52.895307798 +0100
+++ rpm-5.4.10/macros/macros.in	2014-02-02 23:17:31.425308304 +0100
@@ -1030,6 +1030,10 @@ $_arbitrary_tags_tests	Foo:Bar
 %{load:%{_usrlibrpm}/macros.d/cmake}
 
 #------------------------------------------------------------------------
+# cmake(...) dependencies.
+%{load:%{_usrlibrpm}/macros.d/cmakedeps}
+
+#------------------------------------------------------------------------
 # gstreamer(...) configuration.
 %{load:%{_usrlibrpm}/macros.d/gstreamer}
 
--- rpm-5.4.10/scripts/Makefile.am.cmakedeps~	2014-02-02 23:13:26.501971756 +0100
+++ rpm-5.4.10/scripts/Makefile.am	2014-02-02 23:13:58.131972172 +0100
@@ -27,7 +27,7 @@ EXTRA_DIST = api-sanity-autotest.pl api-
 	multiarch-dispatch.h mkmultiarch gem_helper.rb \
 	URPM-Makefile.PL.in URPM-Makefile.am \
 	nix_meta \
-	osgideps.pl pkgconfigdeps.sh \
+	osgideps.pl pkgconfigdeps.sh cmakedeps.sh \
 	perldeps.pl perl.prov perl.req pythondeps.sh pythoneggs.py \
 	php.prov php.req pom2spec psq_pkgng_yaml rpm2cpio \
 	rpmdb_loadcvt rpmdiff rpmdiff.cgi \
@@ -60,7 +60,7 @@ pkgdata_SCRIPTS = \
 	kmod-deps.sh gstreamer.sh javadeps.sh libtooldeps.sh \
 	mono-find-provides mono-find-requires \
 	check-multiarch-files mkmultiarch \
-	osgideps.pl pkgconfigdeps.sh \
+	osgideps.pl pkgconfigdeps.sh cmakedeps.sh \
 	perldeps.pl perl.prov perl.req pythondeps.sh pythoneggs.py \
 	php.prov php.req \
 	rpm2cpio rpmdb_loadcvt \
--- rpm-5.4.10/scripts/cmakedeps.sh.cmakedeps~	2014-02-02 23:13:18.921971657 +0100
+++ rpm-5.4.10/scripts/cmakedeps.sh	2014-02-02 23:13:12.635304907 +0100
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+case $1 in
+-P|--provides)
+	(while read filename ; do
+		case "${filename}" in
+		*-config.cmake)
+			N="`basename \"${filename}\" -config.cmake`"
+			echo "cmake($N)"
+			;;
+		*Config.cmake)
+			N="`basename \"${filename}\" Config.cmake`"
+			echo "cmake($N)"
+			;;
+		*/Find*.cmake)
+			N="`basename \"${filename}\" .cmake |cut -b5-`"
+			echo "cmake($N)"
+			;;
+		esac
+	done) | sort | uniq
+	;;
+-R|--requires)
+	(while read filename ; do
+		case "${filename}" in
+		*.cmake)
+			if grep -qiE '^\s*find_package\s*\([^)]*required' "${filename}"; then
+				grep -iE '^\s*find_package\s*\([^)]*required' ${filename} |sed -e 's,.*(,,;s,^\s*,,;s,\s.*,,' |while read N; do
+					echo "$N" |grep -q '\$' || echo "cmake($N)"
+				done
+			fi
+			if grep -qiE '^\s*find_dependency\s*\([^)]*' "${filename}"; then
+				grep -iE '^\s*find_dependency\s*\([^)]*' ${filename} |sed -e 's,.*(,,;s,^\s*,,;s,\s.*,,' |while read N; do
+					echo "$N" |grep -q '\$' || echo "cmake($N)"
+				done
+			fi
+			;;
+		esac
+	done) | sort | uniq
+	;;
+esac
+exit 0
