--- rpm-5.4.14/lib/verify.c.ghost_verify~	2012-06-06 22:50:41.000000000 +0200
+++ rpm-5.4.14/lib/verify.c	2014-03-16 04:35:03.952941366 +0100
@@ -589,7 +589,7 @@ uint32_t fc = rpmfiFC(fi);
 
 	/* If not verifying %ghost, skip ghost files. */
 	/* XXX the broken!!! logic disables %ghost queries always. */
-	if (!(FF_ISSET(qva->qva_fflags, GHOST) && FF_ISSET(fflags, GHOST)))
+	if (FF_ISSET(qva->qva_fflags, GHOST) && FF_ISSET(fflags, GHOST))
 	    continue;
 
 	/* Gather per-file data into a carrier. */
