--- rpm-5.4.15/macros/mandriva.in.lib_soname~	2014-12-12 14:54:57.193504867 +0100
+++ rpm-5.4.15/macros/mandriva.in	2014-12-12 14:56:28.765589456 +0100
@@ -514,4 +514,17 @@ The %{1} library, a part of %{name}\
 rpm -qf --fileprovide $(readlink -f %{?!2:%{_libdir}}%{?2}/lib%{1}.so) 2>/dev/null | \
 grep $(readlink -f %{?!2:%{_libdir}}%{?2}/lib%{1}.so) | cut -f2 || echo %{name})}
 
+# get soname of libfoo.so
+# usage:
+# %%{lib_soname intl} expands to libintl.so.8
+#
+# default library search path is %%{_libdir}, which you can override with an
+# optional second argument:
+# %%{lib_soname intl %%{uclibc_root}%%{_libdir}}
+
+%lib_soname() %(\
+[ -e %{?!2:%{_libdir}}%{?2}/lib%{1}.so ] && \
+objdump -p %{?!2:%{_libdir}}%{?2}/lib%{1}.so | grep -e SONAME | sed -e 's#.*\\\(lib.*\\\)\$#\\\1#g'\
+)
+
 %{load:/etc/rpm/macros.d/*.macros}
