--- rpm-5.4.15/lib/rpmfc.c.perl-magic~	2015-01-03 15:45:12.881462099 +0100
+++ rpm-5.4.15/lib/rpmfc.c	2015-01-03 15:46:33.190329916 +0100
@@ -633,6 +633,7 @@ static struct rpmfcTokens_s rpmfcTokens[
   { "GNU Info",			RPMFC_MANPAGE|RPMFC_INCLUDE },
 
   { "perl script text",		RPMFC_PERL|RPMFC_INCLUDE },
+  { "Perl script",		RPMFC_PERL|RPMFC_INCLUDE },
   { "Perl5 module source text", RPMFC_PERL|RPMFC_MODULE|RPMFC_INCLUDE },
 
   { "PHP script text",		RPMFC_PHP|RPMFC_INCLUDE },
