--- rpm-5.4.4/rpmpopt.in.clean~	2011-12-05 20:52:18.544053402 +0100
+++ rpm-5.4.4/rpmpopt.in	2011-12-05 20:53:51.520325674 +0100
@@ -454,5 +454,7 @@ rpmbuild alias --lsb --noautoreq --noaut
 
 rpmbuild	alias --withoutcheck		--define 'check exit 0' \
 	--POPTdesc=$"disable %check stanza for build" 
+rpmbuild	alias --withoutclean		--define '__spec_clean_body %{nil}' \
+	--POPTdesc=$"disable %clean stanza for build" 
 # \endverbatim
 #*/
