--- rpm-5.4.10/rpmdb/rpmdb.c.xrealloc~	2013-03-19 10:57:50.188039986 +0100
+++ rpm-5.4.10/rpmdb/rpmdb.c	2013-03-26 15:16:47.114499354 +0100
@@ -2660,7 +2660,8 @@ assert(keylen == sizeof(hdrNum));
 		}
 		if(set && set->count != size) {
 		    set->count = size;
-		    set->recs = xrealloc(set->recs, size * sizeof(*set->recs));
+		    if (size)
+			set->recs = xrealloc(set->recs, size * sizeof(*set->recs));
 		}
 
 		xx = dbiCclose(pdbi, pdbc, 0);
