--- rpm-5.4.15/python/rpmmodule.c.rpmvsf~	2015-02-25 23:25:33.808579555 +0100
+++ rpm-5.4.15/python/rpmmodule.c	2015-02-25 23:26:18.284512570 +0100
@@ -612,6 +612,8 @@ void init_rpm(void)
     REGISTER_ENUM(RPMVSF_NODSA);
     REGISTER_ENUM(RPMVSF_NORSA);
     REGISTER_ENUM(RPMVSF_NOECDSA);
+    REGISTER_ENUM(_RPMVSF_NODIGESTS);
+    REGISTER_ENUM(_RPMVSF_NOSIGNATURES);
 
     REGISTER_ENUM(TR_ADDED);
     REGISTER_ENUM(TR_REMOVED);
--- rpm-5.4.15/rpmio/rpmiotypes.h.rpmvsf~	2015-02-25 23:24:36.503666264 +0100
+++ rpm-5.4.15/rpmio/rpmiotypes.h	2015-02-25 23:26:31.950492038 +0100
@@ -130,6 +130,18 @@ typedef enum pgpVSFlags_e {
     /* bit(s) 21-31 unused */
 } pgpVSFlags;
 
+#define	_RPMVSF_NODIGESTS	\
+  ( RPMVSF_NOSHA1HEADER |	\
+    RPMVSF_NOMD5HEADER |	\
+    RPMVSF_NOSHA1 |		\
+    RPMVSF_NOMD5 )
+
+#define	_RPMVSF_NOSIGNATURES	\
+  ( RPMVSF_NODSAHEADER |	\
+    RPMVSF_NORSAHEADER |	\
+    RPMVSF_NODSA |		\
+    RPMVSF_NORSA )
+
 /*@-redef@*/ /* LCL: ??? */
 typedef /*@abstract@*/ const void * fnpyKey;
 /*@=redef@*/
