--- rpm-5.4.5/lib/rpmfc.c.env_color~	2012-03-05 21:27:16.396663509 +0100
+++ rpm-5.4.5/lib/rpmfc.c	2012-03-05 21:28:12.112703555 +0100
@@ -868,7 +868,7 @@ static int rpmfcSCRIPT(rpmfc fc)
 
 	if (!_filter_values
 	 || (!fc->skipReq
-	  && !rpmfcMatchRegexps(fc->Rmires, fc->Rnmire, s, 'R')))
+	  && !rpmfcMatchRegexps(fc->Rmires, fc->Rnmire, s, 'R'))) {
 	if (is_executable &&
 		strncmp(bn, "bash", sizeof("bash")-1) &&
 		strcmp(bn, "env") &&
@@ -885,6 +885,10 @@ static int rpmfcSCRIPT(rpmfc fc)
 
 	    (void)rpmdsFree(ds);
 	    ds = NULL;
+	} else if (!strcmp(bn, "env") && strlen(se)) {
+	    bn = se;
+	    strsep(&se, " \t\n\r");
+	}
 	}
 
 	/* Set color based on interpreter name. */
