--- rpm-5.4.4/scripts/pkgconfigdeps.sh.oneshot~	2011-12-11 01:40:12.599487103 +0100
+++ rpm-5.4.4/scripts/pkgconfigdeps.sh	2011-12-11 01:40:14.418488230 +0100
@@ -37,12 +37,10 @@ case $1 in
     done
     ;;
 -R|--requires)
-    oneshot="pkgconfig"
     while read filename ; do
     case "${filename}" in
     *.pc)
 	if [[ "$(dirname ${filename})" =~ pkgconfig ]]; then
-	    [ -n "$oneshot" ] && echo "$oneshot"; oneshot=""
 	    # Query the dependencies of the package.
 	    DIR=`dirname ${filename}`
 	    PKG_CONFIG_PATH="$DIR:$DIR/../../share/pkgconfig"
@@ -54,7 +52,6 @@ case $1 in
 		else
 		    echo "pkgconfig($n)"
 		fi
-		oneshot=""
 	    done
 	fi
 	;;
