--- rpm-5.4.15/config.h.in.db52~	2014-08-17 14:03:48.000000000 +0200
+++ rpm-5.4.15/config.h.in	2014-12-19 12:41:37.000000000 +0100
@@ -365,11 +365,11 @@
 /* Define to 1 if you have the `cudf' library (-lcudf). */
 #undef HAVE_LIBCUDF
 
-/* Define to 1 if you have the `db-6.1' library (-ldb-6.1). */
-#undef HAVE_LIBDB_6_1
+/* Define to 1 if you have the `db-5.2' library (-ldb-5.2). */
+#undef HAVE_LIBDB_5_2
 
-/* Define to 1 if you have the `db_sql-6.1' library (-ldb_sql-6.1). */
-#undef HAVE_LIBDB_SQL_6_1
+/* Define to 1 if you have the `db_sql-5.2' library (-ldb_sql-5.2). */
+#undef HAVE_LIBDB_SQL_5_2
 
 /* Define to 1 if you have the `dl' library (-ldl). */
 #undef HAVE_LIBDL
--- rpm-5.4.15/configure.ac.db52~	2014-12-19 12:41:25.232440415 +0100
+++ rpm-5.4.15/configure.ac	2014-12-19 12:41:25.385439931 +0100
@@ -544,7 +544,7 @@ else
     MYPATH=$PATH
 fi
 
-DBXY=db61
+DBXY=db52
 
 AC_PATH_PROG(__BASH, bash, %{_bindir}/bash, $MYPATH)
 AC_PATH_PROG(__BZIP2, bzip2, %{_bindir}/bzip2, $MYPATH)
@@ -1655,7 +1655,7 @@ DBLIBSRCS=""
 CPPFLAGS="${CPPFLAGS} -I${prefix}/include/${DBXY}"
 RPM_CHECK_LIB(
     [Berkeley-DB], [db],
-    [db-6.1], [db_create], [db.h],
+    [db-5.2], [db_create], [db.h],
     [yes,external], [db3],
     [ DBLIBSRCS="$DBLIBSRCS db3.c"
       AM_CONDITIONAL(WITH_DB, [ true ])
@@ -1681,7 +1681,7 @@ RPM_CHECK_LIB(
 dnl # Sqlite 3.8.3.1 from db-6.1.19
 RPM_CHECK_LIB(
     [Berkeley-DB (+SQLite3)], [dbsql],
-    [db_sql-6.1], [sqlite3_open], [dbsql.h],
+    [db_sql-5.2], [sqlite3_open], [dbsql.h],
     [yes,external], [db3/sql],
     [
       AM_CONDITIONAL(WITH_DBSQL, [ true ])
