--- rpm-5.4.13/build/expression.c.tok_free~	2013-09-13 04:31:41.788770077 +0200
+++ rpm-5.4.13/build/expression.c	2013-09-13 04:31:43.779802059 +0200
@@ -365,10 +365,8 @@ static Value doPrimary(ParseState state)
 
     case TOK_IDENTIFIER: {
 	const char *name = state->tokenValue->data.s;
-	char *t = rpmExpand(name, NULL);
 
-	v = valueMakeString(t);
-	t = _free(t);
+	v = valueMakeString(rpmExpand(name, NULL));
 	if (rdToken(state)) {
 	    if (v) valueFree(v);
 	    return NULL;
