--- rpm-5.4.14/configure.ac.rundir~	2014-06-14 06:45:59.849702661 +0200
+++ rpm-5.4.14/configure.ac	2014-06-14 06:46:02.906397787 +0200
@@ -2274,6 +2274,14 @@ AC_SUBST(RPMCANONVENDOR)
 AC_SUBST(RPMCANONOS)
 AC_SUBST(RPMCANONGNU)
 
+RUNDIR="/run"
+AC_ARG_WITH([rundir], 
+  AS_HELP_STRING([--with-rundir=RUNDIR], [specify run-time variable directory]),
+  [RUNDIR=$withval])
+AC_DEFINE_UNQUOTED([RUNDIR],["${RUNDIR}"],[run-time variable directory])
+AC_SUBST(RUNDIR)
+
+
 dnl # RPM installation user/group and uid/gid
 : ${RPMUSER=rpm}
 : ${RPMUID=37}
--- rpm-5.4.14/macros/macros.in.rundir~	2014-06-14 06:46:49.530165313 +0200
+++ rpm-5.4.14/macros/macros.in	2014-06-14 06:47:28.260526153 +0200
@@ -929,6 +929,7 @@ $_arbitrary_tags_tests	Foo:Bar
 %_initddir              %{_sysconfdir}/rc.d/init.d
 # This is a deprecated misspelling, present for backwards compatibility.
 %_initrddir             %{_initddir}
+%_rundir                @RUNDIR@
 
 #---------------------------------------------------------------------
 #   Multilibs
