--- rpm-5.4.14/rpmio/rpmmacro.h.missing_prototype~	2014-07-04 00:43:28.606158462 +0200
+++ rpm-5.4.14/rpmio/rpmmacro.h	2014-07-04 00:43:32.868199976 +0200
@@ -200,6 +200,8 @@ void rpmInitMacros(/*@null@*/ MacroConte
 		h_errno, fileSystem, internalState @*/
 	/*@modifies mc, rpmGlobalMacroContext, fileSystem, internalState @*/;
 
+void delMacroAll(/*@null@*/ MacroContext mc, const char * n);
+
 /**
  * Destroy macro context.
  * @param mc		macro context (NULL uses global context).
