diff -up rpm-4.14.2.1/macros.in.omv~ rpm-4.14.2.1/macros.in
--- rpm-4.14.2.1/macros.in.omv~	2019-02-27 16:29:41.117185506 +0100
+++ rpm-4.14.2.1/macros.in	2019-02-27 16:29:53.114753631 +0100
@@ -529,7 +529,7 @@ package or when debugging this package.\
 #   Same as for "separate" but if the __debug_package global is set then
 #   the -debuginfo package will have a compatibility link for the main
 #   ELF /usr/lib/debug/.build-id/xx/yyy -> /usr/lib/.build-id/xx/yyy
-%_build_id_links compat
+%_build_id_links alldebug
 
 # Whether build-ids should be made unique between package version/releases
 # when generating debuginfo packages. If set to 1 this will pass
