--- rpm-5.4.10/rpmio/rpmiotypes.h.cb2~	2013-03-18 04:37:50.985920598 +0100
+++ rpm-5.4.10/rpmio/rpmiotypes.h	2013-03-18 04:37:05.729281072 +0100
@@ -140,7 +140,8 @@ typedef enum rpmCallbackType_e {
     RPMCALLBACK_CPIO_ERROR      = (1 << 14),
     RPMCALLBACK_SCRIPT_ERROR    = (1 << 15),
     RPMCALLBACK_SCRIPT_START    = (1 << 16),
-    RPMCALLBACK_SCRIPT_STOP     = (1 << 17)
+    RPMCALLBACK_SCRIPT_STOP     = (1 << 17),
+    RPMCALLBACK_INST_STOP       = (1 << 18)
 } rpmCallbackType;
 
 /**
