--- rpm-5.4.14/build/reqprov.c.strfmt_arg~	2014-06-20 01:56:20.821504892 +0200
+++ rpm-5.4.14/build/reqprov.c	2014-06-19 05:17:44.969926263 +0200
@@ -50,7 +50,7 @@ int addReqProv(/*@unused@*/ Spec spec, H
     } else {
 #if defined(RPM_VENDOR_MANDRIVA)
 	if (!strcmp(N, "/bin/sh") || !strcmp(N, "/sbin/ldconfig")) {
-	    rpmlog(RPMLOG_DEBUG, "%s will be satisfied by glibc or it's dependencies, won't add dependency to package\n");
+	    rpmlog(RPMLOG_DEBUG, "%s will be satisfied by glibc or it's dependencies, won't add dependency to package\n", N);
 	    return 0;
 	}
 #endif
