--- rpm-5.4.14/macros/mandriva.in.omv~	2014-07-30 13:15:24.291454831 +0200
+++ rpm-5.4.14/macros/mandriva.in	2014-07-30 13:16:34.935344711 +0200
@@ -145,7 +145,7 @@ end\
 # - %{mklibname test}               => lib64test, on a lib64 platform
 # - %{mklibname test 1 -d}      => libtest1-devel
 # - %{mklibname test 1 -d 0 -s} => libtest1_0-static-devel
-%mklibname(ds)  %{_lib}%{1}%{?2:%{2}}%{?3:_%{3}}%{-s:-static}%{-d:-devel}
+%mklibname(ds)  %{_lib}%{1}%{!-d:%(if echo %{1} |rev |cut -b1 |grep -q '[0-9]'; then echo -n _; fi)}%{?2:%{2}}%{?3:_%{3}}%{-s:-static}%{-d:-devel}
 
 %debugcflags %{?_enable_debug_packages:-Wa,--compress-debug-sections -gdwarf-4 -Wstrict-aliasing=2}
 
