--- rpm-5.4.9/macros/macros.in.l10ndir~	2012-05-18 11:46:41.037713866 +0200
+++ rpm-5.4.9/macros/macros.in	2012-05-18 11:46:59.425827976 +0200
@@ -286,7 +286,7 @@ package or when debugging this package.\
 %__gitdir	%{__varlibwdj}/git
 %__svndir	%{__varlibwdj}/svn
 %__sqldir	%{__varlibwdj}/sql
-%__l10ndir	%{__varlibwdj}/l10n
+#__l10ndir	%{__varlibwdj}/l10n
 
 #	Elements necessary to construct a mongodb:// URI configurably
 #
