--- rpm-5.4.10/tools/debugedit.c.debug_path~	2012-12-27 23:23:40.648261468 +0100
+++ rpm-5.4.10/tools/debugedit.c	2012-12-27 23:23:55.928456269 +0100
@@ -1558,6 +1558,14 @@ main (int argc, char *argv[])
 	}
     }
 
+  /* Get rid of //, ./, ../, symlinks, relative paths, etc */
+  p = malloc(PATH_MAX);
+  if(realpath(base_dir,p)!=NULL){
+    free(base_dir);
+    base_dir=p;
+  } else
+      free(p);
+
   /* Make sure there are trailing slashes in dirs */
   if (base_dir != NULL && base_dir[strlen (base_dir)-1] != '/')
     {
