--- rpm-5.4.9/macros/mandriva.in.install_info~	2012-05-30 21:11:09.893734550 +0200
+++ rpm-5.4.9/macros/mandriva.in	2012-05-30 21:11:12.643700170 +0200
@@ -141,15 +141,6 @@ end\
 %_extension .xz
 
 
-%__install_info /sbin/install-info
-
-%_install_info() if [[ -f %{_infodir}/%{1}%{_extension} ]]; then %{__install_info} %{_infodir}/%{1}%{_extension} --dir=%{_infodir}/dir; fi \
-%{nil}
-# (nb: keep the "alone on its line ';'")
-%_remove_install_info() if [ "$1" = "0" ]; then if [[ -f %{_infodir}/%{1}%{_extension} ]]; then %{__install_info} %{_infodir}/%{1}%{_extension} --dir=%{_infodir}/dir --remove ; fi; fi \
-%{nil}
-
-
 # Macro: %{mklibname <name> [<major> [<minor>]] [-s] [-d]}
 # - %{mklibname test}               => lib64test, on a lib64 platform
 # - %{mklibname test 1 -d}      => libtest1-devel
