diff -up rpm-4.14.1/build/files.c.5004 rpm-4.14.1/build/files.c
--- rpm-4.14.1/build/files.c.5004	2018-01-16 09:06:56.474740606 +0000
+++ rpm-4.14.1/build/files.c	2018-03-24 19:53:37.180349696 +0000
@@ -858,6 +858,7 @@ static VFA_t const virtualAttrs[] = {
     { "%pubkey",	RPMFILE_PUBKEY },
     { "%missingok",	RPMFILE_MISSINGOK },
     { "%artifact",	RPMFILE_ARTIFACT },
+    { "%optional",	RPMFILE_OPTIONAL },
     { NULL, 0 }
 };
 
@@ -1385,7 +1386,7 @@ static rpmRC addFile(FileList fl, const
 	    int lvl = RPMLOG_ERR;
 	    const char *msg = fl->cur.isDir ? _("Directory not found: %s\n") :
 					      _("File not found: %s\n");
-	    if (fl->cur.attrFlags & RPMFILE_EXCLUDE) {
+	    if (fl->cur.attrFlags & RPMFILE_EXCLUDE || fl->cur.attrFlags & RPMFILE_OPTIONAL) {
 		lvl = RPMLOG_WARNING;
 		rc = RPMRC_OK;
 	    }
@@ -2341,6 +2342,8 @@ static void processSpecialDir(rpmSpec sp
 		free(newfile);
 	    }
 	    argvFree(globFiles);
+	} else if (fl->cur.attrFlags & RPMFILE_OPTIONAL) {
+	    rpmlog(RPMLOG_WARNING, _("Optional file not found by glob: %s\n"), eorigfile);
 	} else {
 	    rpmlog(RPMLOG_ERR, _("File not found by glob: %s\n"), eorigfile);
 	    fl->processingFailed = 1;
diff -up rpm-4.14.1/lib/rpmfiles.h.5004 rpm-4.14.1/lib/rpmfiles.h
--- rpm-4.14.1/lib/rpmfiles.h.5004	2018-03-24 19:53:19.196032815 +0000
+++ rpm-4.14.1/lib/rpmfiles.h	2018-03-24 19:54:17.317055338 +0000
@@ -61,6 +61,7 @@ enum rpmfileAttrs_e {
     /* bits 9-10 unused */
     RPMFILE_PUBKEY	= (1 << 11),	/*!< from %%pubkey */
     RPMFILE_ARTIFACT	= (1 << 12),	/*!< from %%artifact */
+    RPMFILE_OPTIONAL	= (1 << 13),	/*!< from %%optional */
 };
 
 typedef rpmFlags rpmfileAttrs;
