--- rpm-5.4.10/lib/rpmds.c.buildid_deps~	2013-01-23 03:39:17.232845754 +0100
+++ rpm-5.4.10/lib/rpmds.c	2013-01-23 03:38:37.428673897 +0100
@@ -3297,7 +3297,7 @@ fprintf(stderr, "*** rpmdsELF(%s, %d, %p
 	    continue;
 	    /*@notreached@*/ /*@switchbreak@*/ break;
 	case SHT_NOTE:
-#if defined(HAVE_GELF_GETNOTE)	/* XXX OpenIndiana & older elfutils haven't. */
+#if defined(HAVE_GELF_GETNOTE) && !defined(RPM_VENDOR_MANDRIVA)	/* XXX OpenIndiana & older elfutils haven't. */
 	    if (!(shdr->sh_flags & SHF_ALLOC))
 		continue;
 	    data = NULL;
