--- rpm-5.4.10/macros/mandriva.in~	2013-02-22 10:55:23.246579032 +0100
+++ rpm-5.4.10/macros/mandriva.in	2013-02-22 10:55:24.679603992 +0100
@@ -364,6 +364,8 @@ GCONF_CONFIG_SOURCE=`%{_gconftool_bin} -
   CONFIGURE_XPATH="--x-includes=%{_prefix}/include --x-libraries=%{_prefix}/%{_lib}" \
   $CONFIGURE_TOP/configure $CROSSCOMPILE \\\
   	--disable-silent-rules \\\
+	--disable-dependency-tracking \\\
+	--disable-rpath \\\
 	--program-prefix=%{?_program_prefix} \\\
  	--prefix=%{_prefix} \\\
 	--exec-prefix=%{_exec_prefix} \\\
@@ -387,6 +389,8 @@ GCONF_CONFIG_SOURCE=`%{_gconftool_bin} -
   CONFIGURE_XPATH="--x-includes=%{_prefix}/include --x-libraries=%{_prefix}/%{_lib}" \
   $CONFIGURE_TOP/configure $CROSSCOMPILE \\\
   	--disable-silent-rules \\\
+	--disable-dependency-tracking \\\
+	--disable-rpath \\\
   	--program-prefix=%{?_program_prefix} \\\
  	--prefix=%{_prefix} \\\
 	--exec-prefix=%{_exec_prefix} \\\
