--- rpm-5.4.15/configure.ac.multiarch~	2014-12-05 21:16:00.464767498 +0100
+++ rpm-5.4.15/configure.ac	2014-12-05 21:16:00.475600487 +0100
@@ -2211,6 +2211,16 @@ RPM_CHECK_LIB(
     [no,external:none], [],
     [AC_MSG_WARN([CPUInfo support is under development])], [])
 
+dnl # Mandriva Linux derived multiarch
+AC_ARG_WITH([multiarch],
+	    AS_HELP_STRING([--with-multiarch], [install multiarch support]),
+	    [WITH_MULTIARCH=yes],
+	    [WITH_MULTIARCH=no])
+AM_CONDITIONAL(
+    [WITH_MULTIARCH],
+    [ test ".$WITH_MULTIARCH" = .yes ])
+AC_SUBST([WITH_MULTIARCH])
+
 dnl ##
 dnl ## ==== INSTALLATION PARAMETERS ====
 dnl ##
--- rpm-5.4.15/scripts/Makefile.am.multiarch~	2014-12-05 21:16:00.313105662 +0100
+++ rpm-5.4.15/scripts/Makefile.am	2014-12-05 22:00:44.863192630 +0100
@@ -47,12 +47,13 @@ EXTRA_DIST = api-sanity-autotest.pl api-
 all:
 
 noinst_SCRIPTS =
-
-noinst_SCRIPTS += multiarch-dispatch multiarch-platform
 bin_SCRIPTS = gendiff
+if WITH_MULTIARCH
+bin_SCRIPTS += multiarch-dispatch multiarch-platform
+else
+noinst_SCRIPTS += multiarch-dispatch multiarch-platform
+endif
 
-
-noinst_SCRIPTS += check-multiarch-files mkmultiarch
 pkgdatadir =		@USRLIBRPM@
 pkgdata_SCRIPTS = \
 	brp-compress brp-python-bytecompile brp-java-gcjcompile \
@@ -74,6 +75,13 @@ pkgdata_SCRIPTS = \
 	vcheck vpkg-provides.sh vpkg-provides2.sh
 
 
+if WITH_MULTIARCH
+pkgdata_SCRIPTS += check-multiarch-files mkmultiarch
+else
+noinst_SCRIPTS  += check-multiarch-files mkmultiarch
+endif
+
+
 pkglibbindir =		@USRLIBRPM@/bin
 pkglibbin_SCRIPTS = api-sanity-autotest.pl api-sanity-checker.pl mgo pom2spec
 
@@ -96,7 +104,11 @@ pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = rpm.pc
 
 pkgincdir = $(includedir)
+if WITH_MULTIARCH
+pkginc_HEADERS = multiarch-dispatch.h
+else
 noinst_HEADERS = multiarch-dispatch.h
+endif
 
 install-exec-hook:
 if WITH_PATH_VERSIONED
