--- rpm-5.4.15/lib/psm.c.sstate_check~	2015-04-14 18:28:43.266816987 +0200
+++ rpm-5.4.15/lib/psm.c	2015-04-14 18:28:48.776804589 +0200
@@ -473,6 +473,7 @@ static pid_t psmWait(rpmpsm psm)
 {
     const rpmts ts = psm->ts;
     rpmtime_t msecs;
+    int ix = tag2slx(psm->scriptTag);
 
     (void) rpmsqWait(&psm->sq);
     msecs = psm->sq.op.usecs/1000;
@@ -484,8 +485,8 @@ static pid_t psmWait(rpmpsm psm)
 	(unsigned)psm->sq.reaped, psm->sq.status,
 	(unsigned)msecs/1000, (unsigned)msecs%1000);
 
-    if (psm->sstates != NULL)
-    {	rpmuint32_t * ssp = psm->sstates + tag2slx(psm->scriptTag);
+    if (ix >= 0 && ix < RPMSCRIPT_MAX)
+    {	rpmuint32_t * ssp = psm->sstates + ix;
 	*ssp &= ~0xffff;
 	*ssp |= (psm->sq.status & 0xffff);
 	*ssp |= RPMSCRIPT_STATE_REAPED;
@@ -646,9 +647,10 @@ static rpmRC runEmbeddedScript(rpmpsm ps
     rpmuint32_t * ssp = NULL;
     int inChroot = enterChroot(psm, &pwdFdno, &rootFdno);
     size_t len = 4 + (matches ? argvCount(matches) : 0);
+    int ix = tag2slx(psm->scriptTag);
 
-    if (psm->sstates != NULL)
-	ssp = psm->sstates + tag2slx(psm->scriptTag);
+    if (ix >= 0 && ix < RPMSCRIPT_MAX)
+	ssp = psm->sstates + ix;
     if (ssp != NULL)
 	*ssp |= (RPMSCRIPT_STATE_EMBEDDED|RPMSCRIPT_STATE_EXEC);
 
@@ -858,7 +860,7 @@ static rpmRC runScript(rpmpsm psm, Heade
     int xx;
     int i;
 
-    if (psm->sstates != NULL && ix >= 0 && ix < RPMSCRIPT_MAX)
+    if (ix >= 0 && ix < RPMSCRIPT_MAX)
 	ssp = psm->sstates + ix;
     if (ssp != NULL)
 	*ssp = RPMSCRIPT_STATE_UNKNOWN;
