--- rpm-5.4.14/scripts/rubygems.rb.no_ruby_version~	2014-03-16 11:47:35.797098586 +0100
+++ rpm-5.4.14/scripts/rubygems.rb	2014-03-16 11:42:57.758190869 +0100
@@ -68,7 +68,7 @@ end
 
 if requires or abi_provide
   abidep = "ruby(abi)"
-  if ruby_versioned
+  if ruby_versioned and RbConfig::CONFIG["ruby_version"] != ""
     abidep += " = %s" % RbConfig::CONFIG["ruby_version"]
   end
   print abidep + "\n"
