--- rpm-5.4.12/rpmio/rpmlua.c.rpmluaext~	2013-08-06 15:31:19.850418373 +0200
+++ rpm-5.4.12/rpmio/rpmlua.c	2013-08-06 15:45:04.007587658 +0200
@@ -1235,7 +1235,7 @@ static int luaopen_rpm(lua_State *L)
     lua_pushglobaltable(L);
 #endif
     luaL_newlib(L, rpmlib);
-    return 0;
+    return 1;
 }
 #endif	/* WITH_LUA */
 
--- rpm-5.4.12/tests/Makefile.am.rpmluaext~	2013-08-06 15:52:53.941594400 +0200
+++ rpm-5.4.12/tests/Makefile.am	2013-08-06 15:52:42.851402687 +0200
@@ -974,6 +974,7 @@ check-convert:
 check-lua:
 	@echo "=== $@ ==="
 	@-${rpm} -E '%{lua:print("-->      lua: Hard Rocks!")}'
+	@-${rpm} -E '%{lua:print(rpm.expand("-->      lua: rpm macro expansion works on rpm %{_rpmversion}!"))
 	@-${rpm} -e lua-test
 	@-${rpm} -U lua-test/lua-test-*.noarch.rpm
 	@-${rpm} -e lua-test
