--- rpm-5.4.10/macros/mandriva.in~	2012-12-26 09:37:59.322548015 +0100
+++ rpm-5.4.10/macros/mandriva.in	2012-12-26 11:03:21.758675061 +0100
@@ -236,7 +236,7 @@ Summary: %{summary}\
 Group: %{group}\
 %description __restore__
 
-%_arch_tag_suffix (%_lib)
+%_arch_tag_suffix %([ "%{?_lib}" = "lib64" ] && echo "()(64bit)")
 %arch_tagged() %{1}%{_arch_tag_suffix} %{?2:%{2} %{3}%{?!3:%{error:undefined 3rd argument in arch_tagged}}}
 
 # Games macros
