--- rpm-5.4.7/scripts/mono-find-requires.newline~	2012-04-25 11:18:24.291758077 +0200
+++ rpm-5.4.7/scripts/mono-find-requires	2012-04-25 11:19:20.977097581 +0200
@@ -174,7 +174,7 @@ PROVIDES=$(
 # TODO: make this use the mono-find-provides script, to share code
 
 # Filter out dups from both lists
-REQUIRES=$(echo "$REQUIRES $config_REQUIRES"  | sort | uniq)
+REQUIRES=$(echo "$REQUIRES${config_REQUIRES: $config_REQUIRES}"  | sort | uniq)
 PROVIDES=$(echo "$PROVIDES" | sort | uniq)
 
 #
