diff -up rpm-4.14.2-rc1/scripts/find-debuginfo.sh.4000~ rpm-4.14.2-rc1/scripts/find-debuginfo.sh
--- rpm-4.14.2-rc1/scripts/find-debuginfo.sh.4000~	2018-06-29 12:07:24.640708925 +0200
+++ rpm-4.14.2-rc1/scripts/find-debuginfo.sh	2018-08-04 15:22:05.804565058 +0200
@@ -237,8 +237,9 @@ strip_to_debug()
   application/x-executable*) g=-g ;;
   application/x-pie-executable*) g=-g ;;
   esac
-  eu-strip --remove-comment $r $g ${keep_remove_args} -f "$1" "$2" || exit
-  chmod 444 "$1" || exit
+  [ -n "$EXCLUDE_FULL_REGEXP" ] && grep -E -q "$EXCLUDE_FULL_REGEXP" <<< "$2" && g=-g
+  eu-strip --remove-comment $r $g ${keep_remove_args} $([ -n "$DISABLE_DEBUG" ] || echo -f "$1") "$2" || exit
+  [ -n "$DISABLE_DEBUG" ] || chmod 444 "$1" || exit
 }
 
 add_minidebug()
@@ -349,11 +350,21 @@ while read nlinks inum f; do
   echo "$nlinks $inum $f" >>"$temp/primary"
 done
 
+[[ -n "$EXCLUDE_FROM_STRIP" ]] && \
+EXCLUDE_REGEXP=`perl -e 'print "(", join("|", @ARGV), ")"' $EXCLUDE_FROM_STRIP`
+[[ -n "$EXCLUDE_FROM_FULL_STRIP" ]] && \
+EXCLUDE_FULL_REGEXP=`perl -e 'print "(", join("|", @ARGV), ")"' $EXCLUDE_FROM_FULL_STRIP`
+
+echo $EXCLUDE_REGEXP
 # Strip ELF binaries
 do_file()
 {
   local nlinks=$1 inum=$2 f=$3 id link linked
 
+  [[ -n "$EXCLUDE_REGEXP" ]] && grep -E -q "$EXCLUDE_REGEXP" <<< "$f" && \
+  return
+  [ -n "$DISABLE_DEBUG" ] && strip_to_debug "" "$f" && return
+
   get_debugfn "$f"
   [ -f "${debugfn}" ] && return
 
@@ -560,7 +571,7 @@ fi
 
 if [ -d "${RPM_BUILD_ROOT}/usr/lib" -o -d "${RPM_BUILD_ROOT}/usr/src" ]; then
   ((nout > 0)) ||
-  test ! -d "${RPM_BUILD_ROOT}/usr/lib" ||
+  test ! -d "${RPM_BUILD_ROOT}/usr/lib/debug" ||
   (cd "${RPM_BUILD_ROOT}/usr/lib"; find debug -type d) |
   sed 's,^,%dir /usr/lib/,' >> "$LISTFILE"
 
