--- rpm-5.4.10/macros/ruby.in.ruby_utf8~	2012-12-27 22:56:04.080134683 +0100
+++ rpm-5.4.10/macros/ruby.in	2012-12-27 22:55:11.132459560 +0100
@@ -27,11 +27,13 @@
 %__gem_helper %{_usrlibrpm}/gem_helper.rb
 
 %gem_build(f:j:) \
+	export LC_CTYPE=UTF-8 \
 	%__gem_helper build \\\
 	%{-f:-f%{-f*}} \\\
 	%{!-j:%{_smp_mflags}}%{-j:-j%{-j*}}
 
 %gem_install(i:n:C) \
+	export LC_CTYPE=UTF-8 \
 	DESTDIR=${DESTDIR:-%{buildroot}} \\\
 	%__gem_helper install \\\
 	--env-shebang --rdoc --ri --force --ignore-dependencies \\\
