--- rpm-5.4.15/configure.ac.platform_detect~	2014-12-18 18:17:13.995050484 +0100
+++ rpm-5.4.15/configure.ac	2014-12-18 18:17:14.018378193 +0100
@@ -2211,6 +2211,16 @@ RPM_CHECK_LIB(
     [no,external:none], [],
     [AC_MSG_WARN([CPUInfo support is under development])], [])
 
+dnl # Automatic /etc/rpm/platform detection
+AC_ARG_WITH([platform_detect],
+	    AS_HELP_STRING([--enable-platform-detect], [enable automatic platform detection]),
+	    [WITH_PLATFORM_DETECT=yes],
+	    [WITH_PLATFORM_DETECT=no])
+AM_CONDITIONAL(
+    [WITH_PLATFORM_DETECT],
+    [ test ".$WITH_PLATFORM_DETECT" = .yes ])
+AC_SUBST([WITH_PLATFORM_DETECT])
+
 dnl # Mandriva Linux derived multiarch
 AC_ARG_WITH([multiarch],
 	    AS_HELP_STRING([--with-multiarch], [install multiarch support]),
--- rpm-5.4.15/lib/rpmrc.c.platform_detect~	2014-12-18 18:14:16.931046682 +0100
+++ rpm-5.4.15/lib/rpmrc.c	2014-12-18 18:14:20.189184583 +0100
@@ -518,7 +518,7 @@ exit:
 }
 /*@=onlytrans@*/
 
-#if defined(WITH_CPUINFO) && defined(WITH_SYCK)
+#if defined(WITH_PLATFORM_DETECT) && defined(WITH_SYCK)
 static inline int rpmCpuinfoMatch(const char * feature, const char * EVR, rpmds cpuinfo)
 {
     rpmds cpufeature = rpmdsSingle(RPMTAG_REQUIRENAME, feature, EVR, RPMSENSE_PROBE);
