--- rpm-5.4.5/lib/rpmfc.c.interpret_deps~	2012-02-29 21:36:46.309581229 +0100
+++ rpm-5.4.5/lib/rpmfc.c	2012-02-29 21:52:44.099058625 +0100
@@ -843,10 +843,19 @@ static int rpmfcSCRIPT(rpmfc fc)
 	*se = '\0';
 	se++;
 
+/*@-moduncon@*/
+	bn = basename(s);
+
 	if (!_filter_values
 	 || (!fc->skipReq
 	  && !rpmfcMatchRegexps(fc->Rmires, fc->Rnmire, s, 'R')))
-	if (is_executable) {
+	if (is_executable &&
+		strncmp(bn, "bash", sizeof("bash")-1) &&
+		strcmp(bn, "env") &&
+		strncmp(bn, "perl", sizeof("perl")-1) &&
+		strncmp(bn, "python", sizeof("python")-1) &&
+		strncmp(bn, "ruby", sizeof("ruby")-1) &&
+		strcmp(bn, "sh")) {
 	    /* Add to package requires. */
 	    ds = rpmdsSingle(RPMTAG_REQUIRENAME, s, "", RPMSENSE_FIND_REQUIRES);
 	    xx = rpmdsMerge(&fc->requires, ds);
@@ -860,8 +869,6 @@ static int rpmfcSCRIPT(rpmfc fc)
 
 	/* Set color based on interpreter name. */
 	/* XXX magic token should have already done this?!? */
-/*@-moduncon@*/
-	bn = basename(s);
 /*@=moduncon@*/
 	if (!strcmp(bn, "perl"))
 	    fc->fcolor->vals[fc->ix] |= RPMFC_PERL;
