--- rpm-5.4.15/macros/mandriva.in.fortify~	2014-12-05 22:41:03.626479677 +0100
+++ rpm-5.4.15/macros/mandriva.in	2014-12-05 22:41:06.114242848 +0100
@@ -152,7 +152,7 @@ end\
 
 # common compilation flags
 
-%_fortify_cflags -Wp,-D_FORTIFY_SOURCE=2
+%_fortify_cflags -D_FORTIFY_SOURCE=2
 
 # cf http://wiki.mandriva.com/en/Development/Packaging/Problems#format_not_a_string_literal_and_no_format_arguments
 %Werror_cflags -Wformat -Werror=format-security
