--- rpm-5.4.7/macros/macros.in.debuginfo~	2012-03-16 18:15:56.639500578 +0100
+++ rpm-5.4.7/macros/macros.in	2012-03-16 18:15:58.278502754 +0100
@@ -154,15 +154,15 @@
 %debug_package \
 %ifnarch noarch\
 %global __debug_package 1\
-%package debug\
+%package debuginfo\
 Summary: Debug information for package %{name}\
 Group: Development/Debug\
 AutoReqProv: 0\
-%description debug\
+%description debuginfo\
 This package provides debug information for package %{name}.\
 Debug information is useful when developing applications that use this\
 package or when debugging this package.\
-%files debug -f debugfiles.list\
+%files debuginfo -f debugfiles.list\
 %defattr(-,root,root)\
 %endif\
 %{nil}
