--- rpm-5.4.7/build/reqprov.c.triggers_nooverlap~	2012-03-20 16:44:54.803999496 +0100
+++ rpm-5.4.7/build/reqprov.c	2012-03-20 16:44:57.294002536 +0100
@@ -117,6 +117,9 @@ int addReqProv(/*@unused@*/ Spec spec, H
 	     * bit further, leaving under #ifdef for now...
 	     * TODO: auto-generated deps too
 	     */
+	    if (Flags & RPMSENSE_TRIGGER)
+		continue;
+
 	    if (flagtag && versions != NULL) {
 		int overlap = 0, res = 0;
 		rpmds old = rpmdsSingle(flagtag, names[len], versions[len] ? versions[len] : "", flags[len]);
