--- rpm-5.4.15/build/rpmbuild.h.missing_types~	2014-09-05 06:52:58.148927604 +0200
+++ rpm-5.4.15/build/rpmbuild.h	2014-09-05 06:52:58.162261070 +0200
@@ -7,6 +7,7 @@
  */
 
 #include <rpmiotypes.h>
+#include <rpmio.h>	/* XXX FD_t typedef */
 #include <rpmmacro.h>
 #include <rpmtypes.h>
 #include <rpmtag.h>
--- rpm-5.4.15/build/rpmspec.h.missing_types~	2014-09-05 06:52:58.082260273 +0200
+++ rpm-5.4.15/build/rpmspec.h	2014-09-05 06:52:58.165594436 +0200
@@ -6,9 +6,14 @@
  *  The Spec and Package data structures used during build.
  */
 
+#include <sys/types.h>	/* XXX size_t typedef */
+#include <rpmtypes.h>
+#include <rpmiotypes.h>
 #include <rpmevr.h>
 #include <rpmfi.h>	/* XXX rpmfi typedef */
 #include <rpmcli.h>	/* XXX QVA_t typedef */
+#include <rpmtag.h>
+#include <rpmio.h>	/* XXX FD_t typedef */
 
 /** \ingroup rpmbuild
  */
--- rpm-5.4.15/lib/rpm46compat.h.missing_types~	2010-10-03 08:12:32.000000000 +0200
+++ rpm-5.4.15/lib/rpm46compat.h	2014-09-05 06:52:58.162261070 +0200
@@ -7,6 +7,7 @@
 #ifndef	H_RPM46COMPAT
 #define	H_RPM46COMPAT		1
 
+#include <stdint.h>	/* XXX unint32_t typedef */
 #include <rpm/rpm4compat.h>
 #include <rpmbuild.h>
 
--- rpm-5.4.15/lib/rpm4compat.h.missing_types~	2011-01-09 11:02:56.000000000 +0100
+++ rpm-5.4.15/lib/rpm4compat.h	2014-09-05 06:52:58.162261070 +0200
@@ -42,6 +42,7 @@
 #include <unistd.h>
 #include <stdio.h>
 #include <errno.h>
+#include <stdint.h>	/* XXX unint32_t typedef */
 
 #define WITH_DB
 #define	_RPMDB_INTERNAL
--- rpm-5.4.15/lib/rpmcli.h.missing_types~	2014-09-05 06:52:58.082260273 +0200
+++ rpm-5.4.15/lib/rpmcli.h	2014-09-05 06:52:58.162261070 +0200
@@ -5,6 +5,7 @@
  * \file lib/rpmcli.h
  */
 
+#include <stdint.h>	/* XXX unint32_t typedef */
 #include <popt.h>
 #include <rpmmacro.h>
 #include <rpmtypes.h>
--- rpm-5.4.15/lib/rpmds.h.missing_types~	2012-06-04 17:10:11.000000000 +0200
+++ rpm-5.4.15/lib/rpmds.h	2014-09-05 06:52:58.162261070 +0200
@@ -6,7 +6,12 @@
  * Structure(s) used for dependency tag sets.
  */
 
+#include <stdio.h>	/* XXX FILE * typedef */
+#include <sys/types.h>	/* XXX size_t typedef */
+#include <rpmtypes.h>
+#include <rpmiotypes.h>
 #include <rpmevr.h>
+#include <rpmtag.h>
 #define	_RPMNS_INTERNAL
 #include <rpmns.h>
 #include <rpmps.h>
--- rpm-5.4.15/lib/rpmfi.h.missing_types~	2014-07-20 00:55:17.000000000 +0200
+++ rpm-5.4.15/lib/rpmfi.h	2014-09-05 06:52:58.165594436 +0200
@@ -1,6 +1,11 @@
 #ifndef H_RPMFI
 #define H_RPMFI
 
+#include <sys/types.h>	/* XXX size_t typedef */
+#include <rpmtypes.h>
+#include <rpmiotypes.h>
+#include <rpmtag.h>
+
 /** \ingroup rpmfi
  * \file lib/rpmfi.h
  * Structure(s) used for file info tag sets.
--- rpm-5.4.15/lib/rpmgi.h.missing_types~	2011-03-18 20:41:12.000000000 +0100
+++ rpm-5.4.15/lib/rpmgi.h	2014-09-05 06:52:58.165594436 +0200
@@ -8,9 +8,12 @@
 #include <fts.h>
 #include <argv.h>
 #include <rpmtypes.h>
+#include <rpmiotypes.h>
 #include <rpmds.h>
 #include <rpmte.h>
 #include <rpmts.h>
+#include <rpmtag.h>
+#include <rpmio.h>	/* XXX FD_t typedef */
 
 /**
  */
--- rpm-5.4.15/lib/rpmps.h.missing_types~	2012-06-04 17:10:11.000000000 +0200
+++ rpm-5.4.15/lib/rpmps.h	2014-09-05 06:52:58.165594436 +0200
@@ -1,6 +1,9 @@
 #ifndef H_RPMPS
 #define H_RPMPS
 
+#include <stdio.h>	/* XXX FILE * typedef */
+#include <rpmiotypes.h>
+
 /** \ingroup rpmps
  * \file lib/rpmps.h
  * Structures and prototypes used for an "rpmps" problem set.
--- rpm-5.4.15/lib/rpmrc.h.missing_types~	2008-08-01 14:33:30.000000000 +0200
+++ rpm-5.4.15/lib/rpmrc.h	2014-09-05 06:52:58.165594436 +0200
@@ -1,6 +1,8 @@
 #ifndef H_RPMRC
 #define	H_RPMRC
 
+#include <stdio.h>	/* XXX FILE * typedef */
+
 /**
  * \file lib/rpmrc.h
  */
--- rpm-5.4.15/lib/rpmte.h.missing_types~	2012-04-19 19:26:07.000000000 +0200
+++ rpm-5.4.15/lib/rpmte.h	2014-09-05 06:52:58.165594436 +0200
@@ -5,7 +5,13 @@
  * \file lib/rpmte.h
  * Structures used for an "rpmte" transaction element.
  */
+#include <stdint.h>	/* XXX unint32_t typedef */
+#include <sys/types.h>	/* XXX size_t typedef */
+#include <rpmtypes.h>
+#include <rpmiotypes.h>
 #include <rpmfi.h>
+#include <rpmtag.h>
+#include <rpmio.h>	/* XXX FD_t typedef */
 
 /**
  */
--- rpm-5.4.15/lib/rpmts.h.missing_types~	2012-06-04 17:10:11.000000000 +0200
+++ rpm-5.4.15/lib/rpmts.h	2014-09-05 06:52:58.165594436 +0200
@@ -6,9 +6,14 @@
  * Structures and prototypes used for an "rpmts" transaction set.
  */
 
-#include "rpmps.h"
-#include "rpmsw.h"
+#include <stdio.h>	/* XXX FILE * typedef */
+#include <stdint.h>	/* XXX unint32_t typedef */
+#include <rpmtypes.h>
+#include <rpmiotypes.h>
+#include <rpmps.h>
+#include <rpmsw.h>
 #include <rpmpgp.h>		/* XXX pgpVSFlags */
+#include <rpmtag.h>
 #if defined(_RPMTS_INTERNAL)
 #include <rpmbag.h>
 #endif
--- rpm-5.4.15/lib/rpmversion.h.missing_types~	2014-08-17 14:02:21.000000000 +0200
+++ rpm-5.4.15/lib/rpmversion.h	2014-09-05 06:52:58.165594436 +0200
@@ -1,6 +1,8 @@
 #ifndef __RPMVERSION_H__
 #define __RPMVERSION_H__
 
+#include <stdint.h>	/* XXX unint32_t typedef */
+
 #ifdef __cplusplus
 extern "C" {
 #endif
--- rpm-5.4.15/rpmconstant/rpmconstant.h.missing_types~	2010-12-01 23:50:01.000000000 +0100
+++ rpm-5.4.15/rpmconstant/rpmconstant.h	2014-09-05 06:52:58.162261070 +0200
@@ -18,6 +18,8 @@
  *
  */
 
+#include <stdint.h>	/* XXX unint32_t typedef */
+
 #include <rpmio.h>
 #include <rpmiotypes.h>		/* XXX fnpyKey */
 #include <rpmpgp.h>
--- rpm-5.4.15/rpmdb/pkgio.h.missing_types~	2010-03-06 17:20:46.000000000 +0100
+++ rpm-5.4.15/rpmdb/pkgio.h	2014-09-05 06:52:58.162261070 +0200
@@ -6,6 +6,8 @@
  * Methods to handle package elements.
  */
 
+#include <stdint.h>	/* XXX unint32_t typedef */
+#include <rpmtypes.h>
 #include <rpmio.h>	/* XXX FD_t typedef */
 #include <rpmpgp.h>	/* XXX pgpDig typedef */
 #include <rpmtag.h>	/* XXX Header typedef */
--- rpm-5.4.15/rpmdb/rpmdb.h.missing_types~	2014-09-05 06:52:58.078926907 +0200
+++ rpm-5.4.15/rpmdb/rpmdb.h	2014-09-05 06:52:58.162261070 +0200
@@ -7,6 +7,7 @@
  * Access RPM indices using Berkeley DB interface(s).
  */
 
+#include <stdint.h>	/* XXX unint32_t typedef */
 #include <assert.h>
 #include <mire.h>
 
--- rpm-5.4.15/rpmdb/rpmevr.h.missing_types~	2014-09-05 06:52:58.165594436 +0200
+++ rpm-5.4.15/rpmdb/rpmevr.h	2014-09-05 06:53:44.422721490 +0200
@@ -6,6 +6,8 @@
  * Structure(s) and routine(s) used for EVR parsing and comparison.
  */
 
+#include <stdint.h>	/* XXX uint32_t typedef */
+
 #ifdef __cplusplus
 extern "C" {
 #endif
--- rpm-5.4.15/rpmdb/rpmns.h.missing_types~	2014-09-05 06:52:58.165594436 +0200
+++ rpm-5.4.15/rpmdb/rpmns.h	2014-09-05 06:54:14.726356424 +0200
@@ -1,6 +1,9 @@
 #ifndef H_RPMNS
 #define H_RPMNS
 
+#include <sys/types.h>	/* XXX size_t typedef */
+#include <rpmiotypes.h>
+
 /** \ingroup rpmds
  * \file rpmdb/rpmns.h
  * Structure(s) and routine(s) used for classifying and parsing names.
--- rpm-5.4.15/rpmdb/rpmtag.h.missing_types~	2014-09-05 06:52:58.055593342 +0200
+++ rpm-5.4.15/rpmdb/rpmtag.h	2014-09-05 06:52:58.165594436 +0200
@@ -5,6 +5,7 @@
  * \file rpmdb/rpmtag.h
  */
 
+#include <stdint.h>	/* XXX unint32_t typedef */
 #include <rpmiotypes.h>
 #include <rpmsw.h>
 
--- rpm-5.4.15/rpmio/argv.h.missing_types~	2013-07-15 20:44:03.000000000 +0200
+++ rpm-5.4.15/rpmio/argv.h	2014-09-05 06:52:58.162261070 +0200
@@ -1,6 +1,8 @@
 #ifndef _H_ARGV_
 #define	_H_ARGV_
 
+#include <stdio.h>	/* XXX FILE * typedef */
+
 /** \ingroup rpmio
  * \file rpmio/argv.h
  */
--- rpm-5.4.15/rpmio/rpmbf.h.missing_types~	2014-07-19 23:55:25.000000000 +0200
+++ rpm-5.4.15/rpmio/rpmbf.h	2014-09-05 06:52:58.162261070 +0200
@@ -1,6 +1,8 @@
 #ifndef	H_RPMBF
 #define	H_RPMBF
 
+#include <sys/types.h>	/* XXX size_t typedef */
+
 /** \ingroup rpmio
  * \file rpmio/rpmbf.h
  */
--- rpm-5.4.15/rpmio/rpmiotypes.h.missing_types~	2014-09-05 06:52:58.095593740 +0200
+++ rpm-5.4.15/rpmio/rpmiotypes.h	2014-09-05 06:52:58.165594436 +0200
@@ -1,6 +1,9 @@
 #ifndef _H_RPMIOTYPES_
 #define	_H_RPMIOTYPES_
 
+#include <stdlib.h>	/* XXX NULL typedef */
+#include <sys/types.h>	/* XXX size_t typedef */
+
 /** \ingroup rpmio
  * \file rpmio/rpmiotypes.h
  */
--- rpm-5.4.15/rpmio/rpmmacro.h.missing_types~	2014-06-16 21:42:44.000000000 +0200
+++ rpm-5.4.15/rpmio/rpmmacro.h	2014-09-05 06:52:58.165594436 +0200
@@ -1,6 +1,9 @@
 #ifndef _H_MACRO_
 #define	_H_MACRO_
 
+#include <stdio.h>	/* XXX FILE * typedef */
+#include <sys/types.h>	/* XXX size_t typedef */
+
 /** \ingroup rpmio
  * \file rpmio/rpmmacro.h
  */
--- rpm-5.4.15/rpmio/rpmsw.h.missing_types~	2013-07-15 20:44:03.000000000 +0200
+++ rpm-5.4.15/rpmio/rpmsw.h	2014-09-05 06:52:58.165594436 +0200
@@ -1,6 +1,9 @@
 #ifndef	H_RPMSW
 #define	H_RPMSW
 
+#include <stdio.h>	/* XXX FILE * typedef */
+#include <sys/time.h>	/* XXX struct timeval typedef */
+
 /** \ingroup rpmio
  * \file rpmio/rpmsw.h
  */
--- rpm-5.4.15/rpmio/yarn.h.missing_types~	2012-04-16 05:21:22.000000000 +0200
+++ rpm-5.4.15/rpmio/yarn.h	2014-09-05 06:52:58.165594436 +0200
@@ -111,6 +111,9 @@
         handler will exit (set to NULL by default for no action)
  */
 
+#include <sys/types.h>	/* XXX size_t typedef */
+#include <rpmiotypes.h>	/* XXX yarnLock typedef */
+
 #ifdef __cplusplus
 extern "C" {
 #endif
