diff -Naur rpm-5.4.15/macros/mandriva.in rpm-5.4.15.tpg/macros/mandriva.in
--- rpm-5.4.15/macros/mandriva.in	2015-04-13 19:53:18.000000000 +0000
+++ rpm-5.4.15.tpg/macros/mandriva.in	2015-04-13 20:00:32.857122881 +0000
@@ -37,8 +37,8 @@
 
 # This will enable the use of distepoch and disttag in stead of polluting
 # %release with such.
-%evr_tuple_select	1234
-%evr_tuple_order	EVRD
+%evr_tuple_select	12345
+%evr_tuple_order	EVTRD
 %disttag		%{!?distsuffix:omv}%{?distsuffix}
 %distepoch		%(sed -e 's#.*release\\ \\(\\S*\\).*#\\1#' /etc/release)
 
