From 94b8de0e338eec0d485654a06b5e384e008a4a8c Mon Sep 17 00:00:00 2001
From: Neal Gompa <ngompa13@gmail.com>
Date: Thu, 19 Dec 2019 23:57:57 -0500
Subject: [PATCH] Clarify ARM arch macro and add %aarch64 arch macro

---
 macros.in | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/macros.in b/macros.in
index f1ae7f8c1..34afcd2b5 100644
--- a/macros.in
+++ b/macros.in
@@ -1122,9 +1122,13 @@ package or when debugging this package.\
 %x86_64	x86_64 znver1 amd64 em64t
 
 #------------------------------------------------------------------------------
-# arch macro for all supported ARM processors
+# arch macro for all supported 32-bit ARM processors
 %arm	armv3l armv4b armv4l armv4tl armv5tl armv5tel armv5tejl armv6l armv6hl armv7l armv7hl armv7hnl armv8l armv8hl armv8hnl armv8hcnl
 
+#------------------------------------------------------------------------------
+# arch macro for all supported 64-bit ARM processors
+%aarch64	aarch64
+
 #------------------------------------------------------------------------------
 # arch macro for 32-bit MIPS processors
 %mips32	mips mipsel mipsr6 mipsr6el
-- 
2.23.0

