--- rpm-5.4.14/lua/linit.c.fixintlua~	2014-03-16 05:57:09.845977334 +0100
+++ rpm-5.4.14/lua/linit.c	2014-03-16 06:11:58.830401582 +0100
@@ -41,7 +41,7 @@ static const luaL_Reg loadedlibs[] = {
   {LUA_DBLIBNAME, luaopen_debug},
   {"rex_posix", luaopen_rex_posix},
   {"rex_pcre", luaopen_rex_pcre},
-  {"posix", luaopen_posix},
+  {"posix", luaopen_posix_c},
   {NULL, NULL}
 };
 
--- rpm-5.4.14/lua/local/lposix.c.fixintlua~	2014-03-16 05:57:09.865976533 +0100
+++ rpm-5.4.14/lua/local/lposix.c	2014-03-16 06:11:58.910398379 +0100
@@ -3938,6 +3938,10 @@ LUALIB_API int luaopen_posix (lua_State
 #undef	NODEV
 #endif
 #if !defined(__APPLE__)
+// sys/param.h defines NODEV as ((dev_t) -1) which confuses MENTRY macro
+#ifdef NODEV
+#undef NODEV
+#endif
 	MENTRY( NODEV		);
 #endif
 	MENTRY( NOENT		);
--- rpm-5.4.14/lua/local/lposix.h.fixintlua~	2004-03-23 06:09:14.000000000 +0100
+++ rpm-5.4.14/lua/local/lposix.h	2014-03-16 06:11:58.960396378 +0100
@@ -1,7 +1,9 @@
 #ifndef LPOSIX_H
 #define LPOSIX_H
 
-int luaopen_posix (lua_State *L)
+#include <lua.h>
+
+int luaopen_posix_c (lua_State *L)
 	/*@modifies L @*/;
 
 #endif
