--- rpm-5.4.15/lib/rpmfc.c.desktop~	2014-09-05 03:20:43.219551626 +0200
+++ rpm-5.4.15/lib/rpmfc.c	2014-09-05 03:20:43.222884994 +0200
@@ -1011,6 +1011,17 @@ static int rpmfcSCRIPT(rpmfc fc)
 		xx = rpmfcHelper(fc, 'R', "kernel");
 	    }
 	mire = mireFree(mire);
+    }
+    if ((fc->fcolor->vals[fc->ix] & RPMFC_SCRIPT)) {
+	miRE mire = mireNew(RPMMIRE_REGEX, RPMTAG_FILEPATHS);
+	if (!mireRegcomp(mire, "^.*/usr/share/applications/.*\\.desktop$"))
+	    if (mireRegexec(mire, fc->fn[fc->ix], (size_t) 0) >= 0) {
+		fc->fcolor->vals[fc->ix] |= (RPMFC_MODULE|RPMFC_SCRIPT);
+		xx = rpmfcHelper(fc, 'P', "desktop");
+		/* XXX: currently of no use, but for the sake of consistency... */
+		xx = rpmfcHelper(fc, 'R', "desktop");
+	    }
+	mire = mireFree(mire);
 #endif
     }
 
@@ -1226,6 +1237,12 @@ assert(fc->fn != NULL);
 		    if (mireRegexec(mire, fc->fn[fc->ix], (size_t) 0) >= 0)
 			fc->fcolor->vals[fc->ix] |= (RPMFC_MODULE|RPMFC_SCRIPT);
 		mire = mireFree(mire);
+		mire = mireNew(RPMMIRE_REGEX, RPMTAG_FILEPATHS);
+		/* buttfuckin' retarded, but whatever.. it'll work at least! ;) */
+		if (!mireRegcomp(mire, "^.*/usr/share/applications/.*\\.desktop$"))
+		    if (mireRegexec(mire, fc->fn[fc->ix], (size_t) 0) >= 0)
+			fc->fcolor->vals[fc->ix] |= RPMFC_SCRIPT;
+		mire = mireFree(mire);
 #endif
 	    }
 	}
--- rpm-5.4.15/macros/macros.rpmbuild.in.desktop~	2014-09-05 03:20:43.216218257 +0200
+++ rpm-5.4.15/macros/macros.rpmbuild.in	2014-09-05 03:20:43.222884994 +0200
@@ -711,5 +711,8 @@ done \
 %__font_provides        %{_rpmhome}/fontconfig.prov
 #%__font_requires        %{_rpmhome}/fontconfig.req
 
+%__desktop_provides        %{_rpmhome}/desktop-file.prov
+#%__desktop_requires        %{_rpmhome}/desktop-file.prov
+
 # \endverbatim
 #*/
--- rpm-5.4.15/scripts/desktop-file.prov.desktop~	2014-09-05 03:20:43.222884994 +0200
+++ rpm-5.4.15/scripts/desktop-file.prov	2014-09-05 03:20:43.222884994 +0200
@@ -0,0 +1,23 @@
+#!/bin/sh
+#
+# Transform desktop mimetype info into RPM mimehandler(type) provides
+#
+# Author: Richard Hughes <richard@hughsie.com>
+# Based on other provides scripts from RPM
+
+OLD_IFS="$IFS"
+while read instfile ; do
+	case "$instfile" in
+	*.desktop)
+		if ! grep -q '^Type=Application$' "$instfile"; then continue; fi
+		if ! grep -q '^Exec=' "$instfile"; then continue; fi
+		mime=`grep '^MimeType=' "$instfile" | cut -d'=' -f2`
+		IFS=';'
+		for type in $mime ; do
+			echo 'mimehandler('$type')'
+		done
+		;;
+	esac
+done
+IFS=$OLD_IFS
+
--- rpm-5.4.15/scripts/Makefile.am.desktop~	2014-09-05 03:20:43.216218257 +0200
+++ rpm-5.4.15/scripts/Makefile.am	2014-09-05 03:21:20.573278516 +0200
@@ -10,7 +10,7 @@ EXTRA_DIST = api-sanity-autotest.pl api-
 	brp-strip brp-strip-comment-note brp-nobuildrootpath \
 	brp-strip-shared brp-strip-static-archive brp-sparc64-linux \
 	brp-implant-ident-static brp-java-repack-jars \
-	check-files cross-build dbconvert.sh \
+	check-files cross-build dbconvert.sh desktop-file.prov \
 	deb_Packages deb_Sources \
 	deb_control deb_md5sums deb_postinst deb_postrm deb_preinst deb_prerm \
 	executabledeps.sh \
@@ -58,7 +58,7 @@ pkgdata_SCRIPTS = \
 	brp-compress brp-python-bytecompile brp-java-gcjcompile \
 	brp-strip brp-strip-comment-note brp-nobuildrootpath \
 	brp-strip-shared brp-strip-static-archive brp-sparc64-linux \
-	check-files cross-build dbconvert.sh executabledeps.sh \
+	check-files cross-build dbconvert.sh desktop-file.prov executabledeps.sh \
 	find-debuginfo.sh find-lang.sh find-prov.pl find-req.pl \
 	find-provides.perl \
 	find-requires.perl fontconfig.prov gem_helper.rb getpo.sh haskelldeps.sh http.req \
