--- rpm-5.4.15/scripts/pkgconfigdeps.sh.omv~	2016-02-27 00:29:56.684992828 +0100
+++ rpm-5.4.15/scripts/pkgconfigdeps.sh	2016-02-27 00:30:42.800174496 +0100
@@ -45,7 +45,7 @@ case $1 in
 	    DIR=`dirname ${filename}`
 	    PKG_CONFIG_PATH="$DIR:$DIR/../../share/pkgconfig"
 	    export PKG_CONFIG_PATH
-	    $pkgconfig --print-requires "$filename" | while read n r v ; do
+	    $pkgconfig --print-requires --print-requires-private "$filename" | while read n r v ; do
 		[ -n "$n" ] || continue
 		if  [ -n "$r" ] && [ -n "$v" ]; then
 		    echo "pkgconfig($n) $r $v"
