--- rpm-5.4.7/lib/rpmds.c.gnu_hash~	2012-03-10 17:35:53.830770413 +0100
+++ rpm-5.4.7/lib/rpmds.c	2012-03-10 17:35:53.837770420 +0100
@@ -3455,6 +3455,7 @@ assert(s != NULL);
     }
     /*@=uniondef @*/
 
+#if !defined(RPM_VENDOR_MANDRIVA)
     /* For DSOs which use the .gnu_hash section and don't have a .hash
      * section, we need to ensure that we have a new enough glibc. */
     if (gotGNUHASH && !gotHASH) {
@@ -3464,6 +3465,7 @@ assert(s != NULL);
 	(void)rpmdsFree(ds);
 	ds = NULL;
     }
+#endif
 
     /* For DSO's, provide the basename of the file if DT_SONAME not found. */
     if (!skipP && isDSO && !gotDEBUG && !gotSONAME) {
--- rpm-5.4.7/macros/mandriva.in.gnu_hash~	2012-03-10 17:35:53.000000000 +0100
+++ rpm-5.4.7/macros/mandriva.in	2012-03-10 17:39:05.871934996 +0100
@@ -358,7 +358,7 @@ GCONF_CONFIG_SOURCE=`%{_gconftool_bin} -
 
 %__libtoolize_configure %{?__libtoolize:(cd $CONFIGURE_TOP; [ ! -f configure.in -a ! -f configure.ac ] || %{__libtoolize} --copy --force)}
 
-%ldflags %{?!_disable_ld_as_needed: -Wl,--as-needed}%{?!_disable_ld_no_undefined: -Wl,--no-undefined}%{?!_disable_ld_relro: -Wl,-z,relro}%{?!_disable_ld_O1: -Wl,-O1}%{?!_disable_ld_build_id: -Wl,--build-id}%{?!_disable_ld_enable_new_dtags: -Wl,--enable-new-dtags}%{?_hardened_flags: %_hardened_flags}
+%ldflags %{?!_disable_ld_as_needed: -Wl,--as-needed}%{?!_disable_ld_no_undefined: -Wl,--no-undefined}%{?!_disable_ld_relro: -Wl,-z,relro}%{?!_disable_ld_O1: -Wl,-O1}%{?!_disable_ld_build_id: -Wl,--build-id}%{?!_disable_ld_enable_new_dtags: -Wl,--enable-new-dtags}%{?!_disable_hash_style_gnu: -Wl,--hash-style=gnu}%{?_hardened_flags: %_hardened_flags}
 
 %setup_compile_flags \
   CFLAGS="${CFLAGS:-%optflags}" ; export CFLAGS ; \
