diff -up rpm-4.19.90/scripts/brp-strip.11~ rpm-4.19.90/scripts/brp-strip
--- rpm-4.19.90/scripts/brp-strip.11~	2024-04-07 13:24:26.744068425 +0200
+++ rpm-4.19.90/scripts/brp-strip	2024-04-07 13:26:01.757048782 +0200
@@ -34,6 +34,8 @@ strip_elf_binaries()
 
   find "$RPM_BUILD_ROOT" -type f \
     ! -regex "${RPM_BUILD_ROOT}/*usr/lib/debug.*" \
+    ! -regex "${RPM_BUILD_ROOT}/lib/firmware.*" \
+    ! -regex "${RPM_BUILD_ROOT}/usr/lib/firmware.*" \
     ! -name "*.py" ! -name "*.js" ! -name "*.rb" \
     ! -name "*.go" -links "${nlinks}" -print0 | \
     xargs -0 -r -P${nprocs} -n${MAX_ARGS} sh -c "file \"\$@\" | \
