--- rpm-5.4.14/rpmdb/header.c.noassert~	2014-06-25 19:22:35.521902207 +0200
+++ rpm-5.4.14/rpmdb/header.c	2014-06-25 19:22:37.381920455 +0200
@@ -1076,7 +1076,7 @@ assert(h != NULL);
 		rpmuint32_t * stei = (rpmuint32_t *)
 			memcpy(alloca(nb), dataStart + off, nb);
 		rdl = (rpmuint32_t)-ntohl(stei[2]);	/* negative offset */
-assert((rpmint32_t)rdl >= 0);	/* XXX insurance */
+//assert((rpmint32_t)rdl >= 0);	/* XXX insurance */
 		ril = (rpmuint32_t)(rdl/sizeof(*pe));
 		if (hdrchkTags(ril) || hdrchkData(rdl))
 		    goto errxit;
