--- rpm-5.4.9/lib/rpmds.c.depoch_rpmlib~	2012-05-15 02:59:01.000000000 +0200
+++ rpm-5.4.9/lib/rpmds.c	2012-05-15 03:00:11.147448068 +0200
@@ -1565,6 +1565,11 @@ static struct rpmlibProvides_s rpmlibPro
     { "rpmlib(SetVersions)",		"4.0.4-alt98",
         (evrFlags)(RPMSENSE_RPMLIB|RPMSENSE_EQUAL),
     N_("dependencies support set/subset versions.") },
+#if defined(RPM_VENDOR_MANDRIVA)
+    { "rpmlib(DistEpoch)",		"5.4.7-1",
+	(RPMSENSE_RPMLIB|RPMSENSE_EQUAL),
+    N_("package has distepoch.") },
+#endif
     { NULL,				NULL, (evrFlags)0,	NULL }
 };
 
