--- rpm-5.4.14/rpmio/macro.c.offbyone~	2014-03-08 11:21:20.952059811 +0100
+++ rpm-5.4.14/rpmio/macro.c	2014-03-08 11:21:29.822059906 +0100
@@ -1880,7 +1880,7 @@ expandMacro(MacroBuf mb)
 	if (STREQ("lua", f, fn)) {
 		rpmlua lua = rpmluaGetGlobalState();
 		rpmlua olua = (rpmlua) memcpy(alloca(sizeof(*olua)), lua, sizeof(*olua));
-		char *scriptbuf = (char *)xmalloc(gn);
+		char *scriptbuf = (char *)alloca(gn+1);
 		const char *printbuf;
 
 		/* Reset the stateful output buffer before recursing down. */
@@ -1910,7 +1910,6 @@ expandMacro(MacroBuf mb)
 		lua->printbufsize = olua->printbufsize;
 		lua->printbufused = olua->printbufused;
 
-		free(scriptbuf);
 		s = se;
 		continue;
 	}
