--- rpm-5.4.10/lib/rpmfc.c.fontdep_sure~	2013-04-16 02:17:42.052532219 +0200
+++ rpm-5.4.10/lib/rpmfc.c	2013-04-16 02:18:44.003630165 +0200
@@ -971,7 +971,7 @@ static int rpmfcSCRIPT(rpmfc fc)
 	if (is_executable || (fc->fcolor->vals[fc->ix] & RPMFC_MODULE))
 	    xx = rpmfcHelper(fc, 'R', "ruby");
     } else
-    if (fc->fcolor->vals[fc->ix] & RPMFC_FONT) {
+    if ((fc->fcolor->vals[fc->ix] == (RPMFC_FONT|RPMFC_INCLUDE))) {
        xx = rpmfcHelper(fc, 'P', "font");
        /* XXX: currently of no use, but for the sake of consistency... */
        xx = rpmfcHelper(fc, 'R', "font");
