diff -up rpm-4.19.90/build/build.c.omv~ rpm-4.19.90/build/build.c
--- rpm-4.19.90/build/build.c.omv~	2024-05-08 18:42:11.829507751 +0200
+++ rpm-4.19.90/build/build.c	2024-05-08 18:47:51.744982052 +0200
@@ -407,7 +407,11 @@ static int buildSpec(rpmts ts, BTA_t bui
 	    (rc = doCheckBuildRequires(ts, spec, test)))
 		goto exit;
 
-	if ((what & RPMBUILD_MKBUILDDIR) && (rc = doBuildDir(spec, test, sbp)))
+	/* mkbuilddir and prep really should go together -- mock uses
+	   --noprep to avoid having to uncompress stuff again after it
+	   has been unpacked for generate_buildrequires, but mkbuilddir
+	   actually wipes out the previous effort. */
+	if ((what & RPMBUILD_MKBUILDDIR) && (what & RPMBUILD_PREP) && (rc = doBuildDir(spec, test, sbp)))
 		goto exit;
 
 	if ((what & RPMBUILD_PREP) &&
