--- rpm-5.4.4/rpmdb/rpmdb.c.repackage~	2011-11-24 16:28:22.436460713 +0100
+++ rpm-5.4.4/rpmdb/rpmdb.c	2011-11-24 16:28:30.107464575 +0100
@@ -2758,7 +2758,7 @@ int rpmdbAdd(rpmdb db, int iid, Header h
 if (_rpmdb_debug)
 fprintf(stderr, "--> %s(%p, %u, %p, %p) h# %u\n", __FUNCTION__, db, (unsigned)iid, h, ts, (unsigned)hdrNum);
 
-assert(headerIsEntry(h, RPMTAG_REMOVETID) == 0);	/* XXX sanity */
+//assert(headerIsEntry(h, RPMTAG_REMOVETID) == 0);	/* XXX sanity */
 
     /* Add the install transaction id. */
     if (iid != 0 && iid != -1) {
