--- rpm-5.4.10/macros/macros.rpmbuild.in.clangdefault~	2014-05-27 22:56:48.430676717 +0200
+++ rpm-5.4.10/macros/macros.rpmbuild.in	2014-05-27 22:56:48.440676592 +0200
@@ -11,9 +11,9 @@
 #
 %__ar			@AR@
 %__as			@AS@
-%__cc			@CC@
-%__cpp			@CPP@
-%__cxx			@CXX@
+%__cc			/usr/bin/clang
+%__cpp			/usr/bin/clang -E
+%__cxx			/usr/bin/clang++
 %__ld			@__LD@
 %__nm			@__NM@
 %__objcopy		@__OBJCOPY@
--- rpm-5.4.10/macros/mandriva.in.clangdefault~	2014-05-27 22:56:48.439676604 +0200
+++ rpm-5.4.10/macros/mandriva.in	2014-05-27 22:57:52.967869891 +0200
@@ -145,7 +145,7 @@ end\
 # - %{mklibname test 1 -d 0 -s} => libtest1_0-static-devel
 %mklibname(ds)  %{_lib}%{1}%{?2:%{2}}%{?3:_%{3}}%{-s:-static}%{-d:-devel}
 
-%debugcflags %{?_enable_debug_packages:-Wa,--compress-debug-sections -gdwarf-4 -fvar-tracking-assignments -frecord-gcc-switches -Wstrict-aliasing=2}
+%debugcflags %{?_enable_debug_packages:-Wa,--compress-debug-sections -gdwarf-4 -Wstrict-aliasing=2}
 
 # common compilation flags
 
@@ -330,6 +330,8 @@ GCONF_CONFIG_SOURCE=`%{_gconftool_bin} -
 %ldflags %{?!_disable_ld_no_undefined: -Wl,--no-undefined} %{?_hardened_flags}
 
 %setup_compile_flags \
+  CC="${CC:-%__cc}" ; export CC ; \
+  CXX="${CXX:-%__cxx}" ; export CXX ; \
   CFLAGS="${CFLAGS:-%optflags}" ; export CFLAGS ; \
   CXXFLAGS="${CXXFLAGS:-%optflags}" ; export CXXFLAGS ; \
   FFLAGS="${FFLAGS:-%optflags}" ; export FFLAGS ; \
