--- rpm-5.4.10/macros/mandriva.in.rename~	2013-02-24 23:17:05.947081929 +0100
+++ rpm-5.4.10/macros/mandriva.in	2013-02-25 00:33:33.342514014 +0100
@@ -47,8 +47,8 @@
 %EVRD			%{?epoch:%{epoch}:}%{?version:%{version}}%{?release:-%{release}}%{?distepoch::%{distepoch}}
 
 %rename() \
-Obsoletes: %{1} < %{EVRD} \
-Provides: %{1} = %{EVRD}
+Obsoletes:	%{1} < %{?2}%{!?2:%{EVRD}} \
+Provides:	%{1} = %{?2}%{!?2:%{EVRD}}
 
 
 %_default_patch_flags	-s -U
