Index: rpm/lib/psm.c
RCS File: /v/rpm/cvs/rpm/lib/psm.c,v
rcsdiff -q -kk '-r2.399.2.5' '-r2.399.2.6' -u '/v/rpm/cvs/rpm/lib/psm.c,v' 2>/dev/null
--- rpm/lib/psm.c	2012/04/19 17:26:06	2.399.2.5
+++ rpm/lib/psm.c	2012/12/03 19:48:55	2.399.2.6
@@ -2755,7 +2755,7 @@
 	    psm->scriptTag = RPMTAG_POSTIN;
 	    psm->progTag = RPMTAG_POSTINPROG;
 	    psm->sense = RPMSENSE_TRIGGERIN;
-	    psm->countCorrection = 0;
+	    psm->countCorrection = 1;
 
 	    if (!(rpmtsFlags(ts) & RPMTRANS_FLAG_NOPOST)) {
 		rc = (rpmRC) rpmpsmNext(psm, PSM_SCRIPT);
