--- rpm-5.4.15/configure.ac.ac_syntax~	2015-01-03 16:03:56.402589444 +0100
+++ rpm-5.4.15/configure.ac	2015-01-03 16:05:57.456383851 +0100
@@ -8,15 +8,15 @@ dnl ##
 
 AC_PREREQ([2.63])
 m4_define([PACKAGE_BUGREPORT_DEFAULT], [rpm-devel@rpm5.org])
-AC_INIT([rpm],[5.4.15],[PACKAGE_BUGREPORT_DEFAULT])
+AC_INIT(rpm, 5.4.15)
+AC_CANONICAL_TARGET
+AM_INIT_AUTOMAKE([foreign tar-ustar subdir-objects color-tests dist-xz])
 PACKAGE_TIMESTAMP="2014-08-13" dnl # YYYY[-MM[-DD[ HH[:MM]]]]
 AC_MSG_TITLE([RPM Package Manager (RPM)], [$PACKAGE_VERSION])
 
 AC_MSG_HEADER([INITIALIZATION])
 AC_CONFIG_SRCDIR([rpmqv.c])
 AC_CONFIG_HEADERS([config.h])
-AC_CANONICAL_TARGET
-AM_INIT_AUTOMAKE([foreign tar-ustar subdir-objects color-tests])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 AC_REQUIRE_AUX_FILE([install-sh])
 AC_REQUIRE_AUX_FILE([mkinstalldirs])
