--- rpm-5.4.12/rpmio/rpmpython.c.modload~	2013-08-08 21:39:28.605164699 +0200
+++ rpm-5.4.12/rpmio/rpmpython.c	2013-08-08 21:39:10.913948618 +0200
@@ -66,9 +66,9 @@ static rpmpython rpmpythonGetPool(/*@nul
 /*@unchecked@*/
 #if defined(MODULE_EMBED)
 static const char * _rpmpythonI_init = "\
-import sys\n\
-from cStringIO import StringIO\n\
-sys.stdout = StringIO()\n\
+import sys;\
+from cStringIO import StringIO;\
+sys.stdout = StringIO();\
 ";
 #endif
 
