diff -up rpm-4.19.90/macros.in.omv~ rpm-4.19.90/macros.in
--- rpm-4.19.90/macros.in.omv~	2024-04-07 19:30:08.328190824 +0200
+++ rpm-4.19.90/macros.in	2024-04-07 19:30:50.048577918 +0200
@@ -432,7 +432,9 @@ Supplements:   (%{name} = %{version}-%{r
 # Should rpm try to download missing sources at build-time?
 # Enabling this is dangerous as long as rpm has no means to validate
 # the integrity of the download with a digest or signature.
-%_disable_source_fetch 1
+# But it is also necessary to download the file in order to verify
+# the signature.
+#_disable_source_fetch 1
 
 #
 # Program to call for each successfully built and written binary package.
