--- rpm-5.4.7/lib/rpmfc.c.free~	2012-03-10 19:45:12.592164430 +0100
+++ rpm-5.4.7/lib/rpmfc.c	2012-03-10 19:45:15.353166385 +0100
@@ -901,7 +901,7 @@ static int rpmfcSCRIPT(rpmfc fc)
     if (fc->fcolor->vals[fc->ix] & RPMFC_PERL) {
     	defaultdocdir = rpmExpand("%{?_defaultdocdir}", NULL);
     	if (defaultdocdir == NULL || *defaultdocdir == '\0') 
-            defaultdocdir = "/usr/share/doc";
+            defaultdocdir = strdup("/usr/share/doc");
 
 	if (strncmp(fn, defaultdocdir, strlen(defaultdocdir))) {
 	    if (fc->fcolor->vals[fc->ix] & RPMFC_MODULE)
