--- rpm-5.4.14/macros/ruby.in.ruby_macros~	2014-03-16 11:29:03.201449905 +0100
+++ rpm-5.4.14/macros/ruby.in	2014-03-16 11:29:06.391322789 +0100
@@ -21,8 +21,8 @@
 %ruby_vendordir		%(%{__ruby} -rrbconfig -e 'print RbConfig::CONFIG["vendordir"]')
 %ruby_vendorarchdir	%(%{__ruby} -rrbconfig -e 'print RbConfig::CONFIG["vendorarchdir"]')
 %ruby_vendorlibdir	%(%{__ruby} -rrbconfig -e 'print RbConfig::CONFIG["vendorlibdir"]')
-%ruby_gemdir		%(%{__ruby} -rrbconfig -e 'print RbConfig::CONFIG["rubylibdir"].sub(RbConfig::CONFIG["ruby_version"], "gems/#{RbConfig::CONFIG["ruby_version"]}")')
-%ruby_ridir		%(%{__ruby} -rrbconfig -e 'print File.join(RbConfig::CONFIG["datadir"], "ri", RbConfig::CONFIG["ruby_version"])')
+%ruby_gemdir		%(%{__ruby} -rrbconfig -e 'print RbConfig::CONFIG["rubygemsdir"]')
+%ruby_ridir		%(%{__ruby} -rrbconfig -e 'print RbConfig::CONFIG["ridir"])')
 
 %__gem_helper %{_usrlibrpm}/gem_helper.rb
 
