--- rpm-5.4.14/build/files.c.unpkg_dirdups~	2014-05-27 20:43:47.882407176 +0200
+++ rpm-5.4.14/build/files.c	2014-05-27 20:58:15.173017444 +0200
@@ -2961,7 +2961,7 @@ static int fiIntersect(/*@null@*/ rpmfi
 	    if (!dups)
 		dups = rpmiobNew(0);
 	    dups = rpmiobAppend(dups, "\t", 0);
-	    dups = rpmiobAppend(dups, fn1+buildrootL, 1);
+	    dups = rpmiobAppend(dups, fn1, 1);
 	    n++;
 	}
     }
@@ -3085,7 +3085,7 @@ static int pkgUnpackagedSubdirs(Package
 	if (p != NULL)
 	while ((p = strchr(p + 1, '/'))) {
 	    *p = '\0';
-	    if (packagedDir(pkg, fn)) {
+	    if (packagedDir(pkg, fn) || (strlen(fn) <= buildrootL)) {
 		*p = '/';
 		/*@innercontinue@*/ continue;
 	    }
@@ -3117,7 +3117,7 @@ static int pkgUnpackagedSubdirs(Package
 
 	for (i = 0; i < n; i++) {
 	    list = rpmiobAppend(list, "\t", 0);
-	    list = rpmiobAppend(list, unpackaged[i]+buildrootL, 1);
+	    list = rpmiobAppend(list, buildrootL+unpackaged[i], 1);
 	    unpackaged[i] = _free(unpackaged[i]);
 	}
 	unpackaged = _free(unpackaged);
