--- rpm-5.4.14/lib/rpmfc.c.remove_all_deptags~	2014-07-02 22:39:15.830295151 +0200
+++ rpm-5.4.14/lib/rpmfc.c	2014-07-02 22:39:51.477287343 +0200
@@ -1954,7 +1954,19 @@ static unsigned removeSillyDeps(Header h
 	}
     }
 
-    if (removed) {
+    if (rnum == removed) {
+	he->tag = RPMTAG_REQUIRENAME;
+	xx = headerDel(h, he, 0);
+	deps = _free(deps);
+
+	he->tag = RPMTAG_REQUIREVERSION;
+	xx = headerDel(h, he, 0);
+	versions = _free(versions);
+
+	he->tag = RPMTAG_REQUIREFLAGS;
+	xx = headerDel(h, he, 0);
+	flags = _free(flags);
+    } else if (removed) {
 	he->tag = RPMTAG_REQUIRENAME;
 	he->t = RPM_STRING_ARRAY_TYPE;
 	he->p.argv = deps;
