--- rpm-5.4.10/macros/mandriva.in.disttag~	2012-10-04 16:36:15.896596812 +0200
+++ rpm-5.4.10/macros/mandriva.in	2012-10-15 23:29:54.191813564 +0200
@@ -1,7 +1,7 @@
-%distribution		Mandriva Linux
-%vendor			Mandriva
+%distribution		Moondrake GNU/Linux
+%vendor			Moondrake
 %bugurl			http://qa.mandriva.com
-%disturl		http://mandriva.org
+%disturl		http://moondrake.org
 
 # This macro will disable the transaction lock on /var/lib/rpm/__db.*.
 # This lock is of no use to us and will also result in errors when trying to
@@ -41,7 +41,7 @@
 # %release with such.
 %evr_tuple_select	1234
 %evr_tuple_order	EVRD
-%disttag		%{!?distsuffix:mdv}%{?distsuffix}
+%disttag		%{!?distsuffix:mdk}%{?distsuffix}
 %distepoch		%(sed -e 's#.*release\\ \\(\\S*\\).*#\\1#' /etc/release)
 
 # Dynamic EVRD tags macro for use with versioned provides/requires in place of
