--- rpm-5.4.14/macros/gstreamer.in.gst~	2011-04-01 07:16:21.000000000 +0200
+++ rpm-5.4.14/macros/gstreamer.in	2014-03-16 04:53:41.428249136 +0100
@@ -2,7 +2,7 @@
 # To make use of these macros insert the following line into your spec file:
 # %{load:%{_usrlibrpm}/macros.d/gstreamer}
 
-%__gst_inspect		@__GST_INSPECT
+%__gst_inspect		@__GST_INSPECT@
 
 # Path to scripts to autogenerate gstreamer package dependencies,
 #
