--- rpm-5.4.10/rpmdb/tagname.c~	2012-04-25 22:34:49.000000000 +0200
+++ rpm-5.4.10/rpmdb/tagname.c	2012-09-07 23:10:04.768327450 +0200
@@ -508,6 +508,7 @@
 if (!(he->tag == 62))
 if (!(he->tag == 261 || he->tag == 269))
 if (!(he->tag == 1000 || he->tag == 1004 || he->tag == 1007))
+if (!(he->tag == 1029))
 if (!(he->tag == 1086 || he->tag == 1087))
 if (he->t != (tagType(he->tag) & 0xffff))
 fprintf(stderr, "==> warning: tag %u type(0x%x) != implicit type(0x%x)\n", (unsigned) he->tag, he->t, tagType(he->tag));
