--- rpm-5.4.15/lib/psm.c.implicit_func_decl~	2015-04-14 18:26:58.308053079 +0200
+++ rpm-5.4.15/lib/psm.c	2015-04-14 18:27:11.337023782 +0200
@@ -7,17 +7,20 @@
 
 #define	_MIRE_INTERNAL	/* XXX mireApply doesn't tell which pattern matched. */
 #define	_RPMLUA_INTERNAL
+#define _RPMDB_INTERNAL
 
 #include <rpmio_internal.h>	/* XXX FDSTAT_READ */
 #include <rpmcb.h>		/* XXX fnpyKey */
 #include <rpmsx.h>
 #include <rpmmacro.h>
 #include <rpmurl.h>
+#include <rpmdb.h>
 
 #include <rpmaug.h>
 #include <rpmficl.h>
 #include <rpmjni.h>
 #include <rpmjs.h>
+#include <rpmlio.h>
 #include <rpmlua.h>
 #include <rpmmrb.h>
 #include <rpmperl.h>
