--- rpm-5.4.12/rpmconstant/rpmconstant.c.locale~	2013-06-26 21:30:48.000000000 +0200
+++ rpm-5.4.12/rpmconstant/rpmconstant.c	2013-07-30 16:21:46.070730344 +0200
@@ -89,7 +89,7 @@ int rpmconstInitToContext(rpmconst c, co
     int rc = 0;
     if (context)	/* XXX coverity #1035889 */
     for (ptr = lccontext; *ptr != 0; ptr++)
-        *ptr = (char)tolower(*ptr);
+        *ptr = (char)xtolower(*ptr);
     rpmconstInitL(c);
     while (rpmconstNextL(c)) {
         if (!strcmp(lccontext, rpmconstContext(c))) {
@@ -108,7 +108,7 @@ int rpmconstNameMatch(rpmconst c, const
     char * ucname = strdup(name);
     
     for (uc = ucname; *uc != 0; uc++)
-        *uc = (char)toupper(*uc);
+        *uc = (char)xtoupper(*uc);
     
     if (!prefixed) prefixed = ALLCASE_PREFIX;
     if (prefixed & WITH_PREFIX)
