--- rpm-5.4.14/build/files.c.fixfdigests~	2014-03-15 05:26:49.147965139 +0100
+++ rpm-5.4.14/build/files.c	2014-03-16 04:12:22.417407913 +0100
@@ -394,7 +394,6 @@ static rpmRC parseForVerify(char * buf,
 		if (strcmp(p, vfa->attribute))
 		    /*@innercontinue@*/ continue;
 		verifyFlags |= vfa->flag;
-		    verifyFlags &= ~RPMVERIFY_FDIGEST;
 		/*@innerbreak@*/ break;
 	    }
 	    if (vfa->attribute)
