--- rpm-5.4.9/rpmdb/tagname.c.i18n_str~	2012-05-18 07:27:53.216728091 +0200
+++ rpm-5.4.9/rpmdb/tagname.c	2012-05-18 08:08:19.766839174 +0200
@@ -493,7 +493,11 @@ void tagTypeValidate(HE_t he)
 {
     /* XXX Re-map RPM_I18NSTRING_TYPE -> RPM_STRING_TYPE */
     if (he->t == RPM_I18NSTRING_TYPE)
+#if !defined(SUPPORT_I18NSTRING_TYPE)
 	he->t = RPM_STRING_TYPE;
+#else
+	return;
+#endif
 
     /* XXX Arbitrary tags are always strings. */
     if ((he->tag & 0x40000000)
