--- rpm-5.4.15/macros/mandriva.in.no_conf_target_host_build~	2014-12-06 05:39:29.646939761 +0100
+++ rpm-5.4.15/macros/mandriva.in	2014-12-06 05:39:33.420266085 +0100
@@ -340,14 +340,14 @@ GCONF_CONFIG_SOURCE=`%{_gconftool_bin} -
 %if %{cross_compiling} \
     PKG_CONFIG_PATH=/usr/%{_target_platform}/sys-root%{_libdir}/pkgconfig:/usr/%{_target_platform}/sys-root%{_datadir}/pkgconfig:%{_libdir}/pkgconfig:%{_datadir}/pkgconfig:$PKG_CONFIG_PATH; export PKG_CONFIG_PATH; \
 %ifnarch noarch \
-    CROSSCOMPILE="--target=%{_target_platform} --host=%{_target_platform} --build=%{_build}" ; \
+    CROSSCOMPILE="%{?!noconftarget:--target=%{_target_platform}} %{?!noconfhost:--host=%{_target_platform}} %{?!noconfbuild:--build=%{_build}}" ; \
 %endif \
 %else \
 %ifnarch noarch \
-    CROSSCOMPILE="--target=%{_target_platform} --build=%{_target_platform}" ; \
+    CROSSCOMPILE="%{?!noconftarget:--target=%{_target_platform}} %{?!noconfbuild:--build=%{_target_platform}}" ; \
 %endif \
 %endif \
-  %{?cross:CROSSCOMPILE="--target=%{cross} --host=%{_target_platform} --build=%{_build}" ; } \
+  %{?cross:CROSSCOMPILE="%{?!noconftarget:--target=%{cross}} %{?!noconfhost:--host=%{_target_platform}} %{?!noconfbuild:--build=%{_build}}" ; } \
   export CROSSCOMPILE ; \
   %{nil}
 
