diff -up rpm-4.18.90/CMakeLists.txt.omv~ rpm-4.18.90/CMakeLists.txt
--- rpm-4.18.90/CMakeLists.txt.omv~	2023-05-04 22:21:02.762144768 +0200
+++ rpm-4.18.90/CMakeLists.txt	2023-05-04 22:21:24.947330746 +0200
@@ -72,12 +72,13 @@ function(makemacros)
 	set(RUNDIR /run)
 
 	set(acutils
-		awk ar as cpp cxx
+		awk ar as cpp
 	)
 	foreach (V ${acutils})
 		string(TOUPPER ${V} n)
 		set(${n} /usr/bin/${V})
 	endforeach()
+	set(cxx /usr/bin/c++)
 	set(extutils
 		7zip bzip2 cat chmod chown cp curl file gpg grep gzip id cc ln
 		install lrzip lzip xz make mkdir mv patch rm sed tar unzip
