--- rpm-5.4.10/macros/mandriva.in.omv~	2014-05-27 18:55:13.954882600 +0200
+++ rpm-5.4.10/macros/mandriva.in	2014-05-27 19:02:06.629723455 +0200
@@ -363,7 +363,7 @@ GCONF_CONFIG_SOURCE=`%{_gconftool_bin} -
 # This is an improved version of %configure (from PLD team).
 %configure \
   %before_configure ; \
-  %{?!_disable_libtoolize:%{?__libtoolize_configure:%{__libtoolize_configure};}} \
+  %{?_enable_libtoolize:%{?__libtoolize_configure:%{__libtoolize_configure};}} \
   [ -f $CONFIGURE_TOP/configure.in -o -f $CONFIGURE_TOP/configure.ac ] && \
   CONFIGURE_XPATH="--x-includes=%{_prefix}/include --x-libraries=%{_prefix}/%{_lib}" \
   $CONFIGURE_TOP/configure $CROSSCOMPILE \\\
@@ -388,31 +388,8 @@ GCONF_CONFIG_SOURCE=`%{_gconftool_bin} -
     $CONFIGURE_XPATH
 
 %configure2_5x \
-  %before_configure ; \
-  %{?_enable_libtoolize:%{?__libtoolize_configure:%{__libtoolize_configure};}} \
-  [ -f $CONFIGURE_TOP/configure.in -o -f $CONFIGURE_TOP/configure.ac ] && \
-  CONFIGURE_XPATH="--x-includes=%{_prefix}/include --x-libraries=%{_prefix}/%{_lib}" \
-  $CONFIGURE_TOP/configure $CROSSCOMPILE \\\
-	--disable-static \\\
-  	--disable-silent-rules \\\
-	--disable-dependency-tracking \\\
-	--disable-rpath \\\
-  	--program-prefix=%{?_program_prefix} \\\
- 	--prefix=%{_prefix} \\\
-	--exec-prefix=%{_exec_prefix} \\\
-	--bindir=%{_bindir} \\\
-	--sbindir=%{_sbindir} \\\
-	--sysconfdir=%{_sysconfdir} \\\
-	--datadir=%{_datadir} \\\
-	--includedir=%{_includedir} \\\
-	--libdir=%{_libdir} \\\
-	--libexecdir=%{_libexecdir} \\\
-	--localstatedir=%{_localstatedir} \\\
-	--sharedstatedir=%{_sharedstatedir} \\\
-	--mandir=%{_mandir} \\\
-	--infodir=%{_infodir} \\\
-    $CONFIGURE_XPATH
-
+  echo "%%configure2_5x is deprecated, please update the spec to use %%configure instead"; \
+  %configure
 
 %old_makeinstall \
     make \\\
