--- rpm-5.4.14/macros/mandriva.in.twiddle~	2014-07-27 17:56:00.904730114 +0200
+++ rpm-5.4.14/macros/mandriva.in	2014-07-27 17:57:17.588611155 +0200
@@ -37,8 +37,8 @@
 
 # This will enable the use of distepoch and disttag in stead of polluting
 # %release with such.
-%evr_tuple_select	1234
-%evr_tuple_order	EVRD
+%evr_tuple_select	12345
+%evr_tuple_order	EVTRD
 %disttag		%{!?distsuffix:omv}%{?distsuffix}
 %distepoch		%(sed -e 's#.*release\\ \\(\\S*\\).*#\\1#' /etc/release)
 
