--- rpm-5.4.5/lib/rpmfc.c.php_dep_gen~	2012-02-24 13:41:47.442140908 +0100
+++ rpm-5.4.5/lib/rpmfc.c	2012-02-24 13:41:49.933142149 +0100
@@ -919,7 +919,9 @@ static int rpmfcSCRIPT(rpmfc fc)
     } else
     if (fc->fcolor->vals[fc->ix] & RPMFC_PHP) {
 	xx = rpmfcHelper(fc, 'P', "php");
+#ifdef	NOTYET
 	if (is_executable)
+#endif
 	    xx = rpmfcHelper(fc, 'R', "php");
     } else
     if (fc->fcolor->vals[fc->ix] & RPMFC_MONO) {
