--- rpm-5.4.14/configure.ac.lua~	2014-02-12 20:10:02.534403788 +0100
+++ rpm-5.4.14/configure.ac	2014-02-12 20:10:14.434403711 +0100
@@ -1976,10 +1976,12 @@ dnl           *-*-linux* ) USE_LUA_SHADO
 dnl       esac
           AC_DEFINE(WITH_LUA_INTERNAL, 1, [Define if building with internal Lua])
           AC_CONFIG_FILES([lua/Makefile])
+	  AM_CONDITIONAL(WITH_LUA_INTERNAL, [true])
       else
           AC_CHECK_LIB(m, fabs)
           AC_CHECK_LIB(dl, dlopen)
           AC_CHECK_FUNC(dlopen)
+	  AM_CONDITIONAL(WITH_LUA_INTERNAL, [false])
       fi
     ], [])
 AC_SUBST(WITH_LUA_SUBDIR_DEF)
--- rpm-5.4.14/rpmio/Makefile.am.lua~	2014-02-12 20:10:22.064403662 +0100
+++ rpm-5.4.14/rpmio/Makefile.am	2014-02-12 20:11:06.284403379 +0100
@@ -215,6 +215,10 @@ if ENABLE_BUILD_MAXEXTLIBDEP
 librpmio_la_LDFLAGS += $(LDFLAGS)
 #librpmio_la_LIBADD  += $(LIBS)
 endif
+if WITH_LUA_INTERNAL
+librpmio_la_LIBADD += \
+	$(top_builddir)/lua/liblua.la
+endif
 #librpmio.la: $(librpmio_la_OBJECTS) $(librpmio_la_DEPENDENCIES) 
 #	$(librpmio_la_LINK) -rpath $(usrlibdir) $(librpmio_la_OBJECTS) $(librpmio_la_LIBADD)
 
