--- rpm-5.4.10/perl/Makefile.PL.in.omv~	2014-06-12 18:39:00.521964904 +0200
+++ rpm-5.4.10/perl/Makefile.PL.in	2014-06-12 18:39:28.102620099 +0200
@@ -64,6 +64,8 @@ WriteMakefile(
 	      'OBJECT'	=> join(' ', @objects),
 	      'VERSION' => '@VERSION@',
 	      'MAKEFILE'=> 'Makefile.perl',
+	      'CC'	=> '@CC@',
+	      'CXX'	=> '@CXX@',
 	      'CCFLAGS' => join(' ', '@CFLAGS@', '-Wno-unused-variable', '-Wno-missing-prototypes', '-Wno-implicit', '-Wno-strict-prototypes'),
 	      'dynamic_lib'	=> { OTHERLDFLAGS => join(' ', @ldaddp, @ldadd, @ldlibs) },
 	      'OPTIMIZE'=> '-g',
