diff -up rpm-6.0.0/build/files.cc.14~ rpm-6.0.0/build/files.cc
--- rpm-6.0.0/build/files.cc.14~	2025-09-22 14:09:24.000000000 +0200
+++ rpm-6.0.0/build/files.cc	2025-11-09 16:53:07.592031818 +0100
@@ -868,6 +868,7 @@ static VFA_t const virtualAttrs[] = {
     { "%pubkey",	RPMFILE_PUBKEY },
     { "%missingok",	RPMFILE_MISSINGOK },
     { "%artifact",	RPMFILE_ARTIFACT },
+    { "%optional",	RPMFILE_OPTIONAL },
     { NULL, 0 }
 };
 
@@ -1459,7 +1460,7 @@ static rpmRC addFile(FileList fl, const
 	    int ignore = 0;
 	    const char *msg = fl->cur.isDir ? _("Directory not found: %s\n") :
 					      _("File not found: %s\n");
-	    if (fl->cur.attrFlags & RPMFILE_EXCLUDE)
+	    if (fl->cur.attrFlags & RPMFILE_EXCLUDE || fl->cur.attrFlags & RPMFILE_OPTIONAL)
 		ignore = 1;
 	    if (fl->cur.attrFlags & RPMFILE_DOC) {
 		int strict_doc =
diff -up rpm-6.0.0/include/rpm/rpmfiles.h.14~ rpm-6.0.0/include/rpm/rpmfiles.h
--- rpm-6.0.0/include/rpm/rpmfiles.h.14~	2025-09-22 14:09:24.000000000 +0200
+++ rpm-6.0.0/include/rpm/rpmfiles.h	2025-11-09 16:53:07.592187472 +0100
@@ -59,6 +59,7 @@ enum rpmfileAttrs_e {
     /* bits 9-10 unused */
     RPMFILE_PUBKEY	= (1 << 11),	/*!< from %%pubkey */
     RPMFILE_ARTIFACT	= (1 << 12),	/*!< from %%artifact */
+    RPMFILE_OPTIONAL	= (1 << 13),	/*!< from %%optional */
 };
 
 typedef rpmFlags rpmfileAttrs;
diff -up rpm-6.0.0/tests/data/SPECS/optional.spec.14~ rpm-6.0.0/tests/data/SPECS/optional.spec
--- rpm-6.0.0/tests/data/SPECS/optional.spec.14~	2025-11-09 16:53:07.592250411 +0100
+++ rpm-6.0.0/tests/data/SPECS/optional.spec	2025-11-09 16:53:07.592250411 +0100
@@ -0,0 +1,38 @@
+Summary: optional -- test for the %%optional tag
+Name: optional
+Version: 1.0
+Release: 1
+Group: Utilities
+License: GPL
+BuildArch: noarch
+
+%description
+Test for files with the %%optional tag
+
+%prep
+
+%build
+
+%install
+mkdir -p %{buildroot}/dir-exists \
+	%{buildroot}/dirglob-exists-1 \
+	%{buildroot}/dirglob-exists-2
+touch %{buildroot}/file-exists \
+	%{buildroot}/glob-exists-1 \
+	%{buildroot}/glob-exists-2
+ln -s file-exists %{buildroot}/symlink-exists
+ln -s / %{buildroot}/symlink-to-dir-exists
+
+%files
+%defattr(-,root,root)
+%optional /file-exists
+%optional /file-does-not-exist
+%optional /glob-exists-*
+%optional /glob-does-not-exist-*
+%optional /dir-exists
+%optional /dir-does-not-exist
+%optional /dirglob-exists-*
+%optional /symlink-exists
+%optional /symlink-does-not-exist
+%optional /symlink-to-dir-exists
+%optional /symlink-to-dir-does-not-exist
diff -up rpm-6.0.0/tests/rpmbuild.at.14~ rpm-6.0.0/tests/rpmbuild.at
--- rpm-6.0.0/tests/rpmbuild.at.14~	2025-09-22 14:09:24.000000000 +0200
+++ rpm-6.0.0/tests/rpmbuild.at	2025-11-09 16:59:42.020966894 +0100
@@ -2863,6 +2863,34 @@ runroot rpmbuild --quiet -bb \
 RPMTEST_CLEANUP
 
 # ------------------------------
+# Check functionality of %optional tag
+AT_SETUP([rpmbuild optional tag])
+AT_KEYWORDS([build] [optional])
+AT_CHECK([[
+rm -rf ${TOPDIR}
+
+runroot rpmbuild \
+  -bb --quiet /data/SPECS/optional.spec
+
+runroot rpm -qp --qf \
+  "\n[%{filemodes:perms} %-8{fileusername} %-8{filegroupname} %{filenames}\n]"\
+  /build/RPMS/noarch/optional-1.0*.noarch.rpm
+]],
+[0],
+[
+drwxr-xr-x /dir-exists
+drwxr-xr-x /dirglob-exists-1
+drwxr-xr-x /dirglob-exists-2
+-rw-r--r-- /file-exists
+-rw-r--r-- /glob-exists-1
+-rw-r--r-- /glob-exists-2
+lrwxrwxrwx /symlink-exists
+lrwxrwxrwx /symlink-to-dir-exists
+],
+[])
+AT_CLEANUP
+
+# ------------------------------
 # Check dynamic build requires
 RPMTEST_SETUP_RW([dynamic build requires rpmbuild -bs])
 AT_KEYWORDS([build])
