--- rpm-5.4.10/macros/macros.rpmbuild.in.omv~	2014-07-05 16:25:29.194517856 +0200
+++ rpm-5.4.10/macros/macros.rpmbuild.in	2014-07-05 16:25:33.435464837 +0200
@@ -310,7 +310,7 @@ echo "Patch #%{__patch_number} (%{basena
 
 #
 # Define a generic value for optflags. Normally overridden by per-target macros.
-%optflags		-O2
+%optflags		-Oz
 
 #
 # Define per-arch and per-os defaults. Normally overridden by per-target macros.
--- rpm-5.4.10/macros/mandriva.in.omv~	2014-07-05 16:25:07.218792590 +0200
+++ rpm-5.4.10/macros/mandriva.in	2014-07-05 16:25:16.683674263 +0200
@@ -157,7 +157,7 @@ end\
 %Werror_cflags -Wformat -Werror=format-security
 
 %_ssp_cflags -fstack-protector --param=ssp-buffer-size=4 %{?_serverbuild_flags}
-%__common_cflags -O2 %{debugcflags} -pipe %{Werror_cflags} %{?_fortify_cflags}
+%__common_cflags -Oz %{debugcflags} -pipe %{Werror_cflags} %{?_fortify_cflags}
 %__common_cflags_with_ssp %{__common_cflags} %{?_ssp_cflags}
 
 # Servers opt flags.
