Index: rpm/tools/debugedit.c
RCS File: /v/rpm/cvs/rpm/tools/debugedit.c,v
rcsdiff -q -kk '-r2.23.2.10' '-r2.23.2.11' -u '/v/rpm/cvs/rpm/tools/debugedit.c,v' 2>/dev/null
--- rpm/tools/debugedit.c	2013/08/08 15:53:36	2.23.2.10
+++ rpm/tools/debugedit.c	2013/08/08 16:39:05	2.23.2.11
@@ -1632,11 +1632,17 @@
 	  /* TODO: Handle stabs */
 	  if (strcmp (name, ".stab") == 0)
 	    {
-	      fprintf (stderr, "Stabs debuginfo not supported: %s\n", file);
-	      exit (1);
+	      fprintf (stderr, "Warning: Stabs debuginfo currently not supported: %s\n", file);
+	      break;
 	    }
+		/* The core DWARF data  containing DIEs */
 	  if (strcmp (name, ".debug_info") == 0)
 	    edit_dwarf2 (dso);
+		else 
+			 {
+	      fprintf (stderr, "Failed to produce debuginfo: debug format not supported: %s\n", name);
+				exit(1);
+			 }
 
 	  break;
 	case SHT_NOTE:
