Index: rpm/rpmio/rpmlog.c
RCS File: /v/rpm/cvs/rpm/rpmio/rpmlog.c,v
rcsdiff -q -kk '-r2.46.4.4' '-r2.46.4.5' -u '/v/rpm/cvs/rpm/rpmio/rpmlog.c,v' 2>/dev/null
--- rpmio/rpmlog.c	2013/07/13 22:53:01	2.46.4.4
+++ rpmio/rpmlog.c	2013/10/25 18:46:39	2.46.4.5
@@ -284,6 +284,7 @@
 
 void _rpmlog (int code, const char *fmt, ...)
 {
+    int saved_errno = errno;
     va_list ap;
 
     va_start(ap, fmt);
@@ -291,4 +292,5 @@
     vrpmlog(code, fmt, ap);
     /*@=internalglobs@*/
     va_end(ap);
+    errno = saved_errno;
 }
